<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Means Separation and Data Presentation | Data Science for Agricultural Professionals</title>
  <meta name="description" content="Practical statistics for those involved in agronomy and related agricultural sciences." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Means Separation and Data Presentation | Data Science for Agricultural Professionals" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://mharbur.github.io/data-science-for-agricultural-professionals/" />
  
  <meta property="og:description" content="Practical statistics for those involved in agronomy and related agricultural sciences." />
  <meta name="github-repo" content="https://github.com/mharbur/data-science-for-agricultural-professionals" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Means Separation and Data Presentation | Data Science for Agricultural Professionals" />
  
  <meta name="twitter:description" content="Practical statistics for those involved in agronomy and related agricultural sciences." />
  

<meta name="author" content="Marin L. Harbur" />


<meta name="date" content="2021-09-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="multiple-treatment-designs.html"/>
<link rel="next" href="messy-and-missing-data.html"/>
<script src="libs/header-attrs-2.7/header-attrs.js"></script>
<script src="libs/jquery-3.5.1/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<script src="libs/plotly-binding-4.9.3/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-1.57.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-1.57.1/plotly-latest.min.js"></script>
<link href="libs/leaflet-1.3.1/leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-1.3.1/leaflet.js"></script>
<link href="libs/leafletfix-1.0.0/leafletfix.css" rel="stylesheet" />
<script src="libs/proj4-2.6.2/proj4.min.js"></script>
<script src="libs/Proj4Leaflet-1.0.1/proj4leaflet.js"></script>
<link href="libs/rstudio_leaflet-1.3.1/rstudio_leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-binding-2.0.4.1/leaflet.js"></script>
<script src="libs/leaflet-providers-1.9.0/leaflet-providers_1.9.0.js"></script>
<script src="libs/leaflet-providers-plugin-2.0.4.1/leaflet-providers-plugin.js"></script>
<script src="libs/stars-1/data_stars_starsf123b5.txt"></script>
<script src="libs/joda-0.0.1/joda.js"></script>
<script src="libs/joda-0.0.1/addImageQuery-bindings.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Data Science for Agricultural Professionals</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#welcome"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#r-language"><i class="fa fa-check"></i>R-language</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="population-statistics.html"><a href="population-statistics.html"><i class="fa fa-check"></i><b>1</b> Population Statistics</a>
<ul>
<li class="chapter" data-level="1.1" data-path="population-statistics.html"><a href="population-statistics.html#populations"><i class="fa fa-check"></i><b>1.1</b> Populations</a></li>
<li class="chapter" data-level="1.2" data-path="population-statistics.html"><a href="population-statistics.html#case-study-yield-map"><i class="fa fa-check"></i><b>1.2</b> Case Study: Yield Map</a></li>
<li class="chapter" data-level="1.3" data-path="population-statistics.html"><a href="population-statistics.html#distributions"><i class="fa fa-check"></i><b>1.3</b> Distributions</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="population-statistics.html"><a href="population-statistics.html#histograms"><i class="fa fa-check"></i><b>1.3.1</b> Histograms</a></li>
<li class="chapter" data-level="1.3.2" data-path="population-statistics.html"><a href="population-statistics.html#percentiles"><i class="fa fa-check"></i><b>1.3.2</b> Percentiles</a></li>
<li class="chapter" data-level="1.3.3" data-path="population-statistics.html"><a href="population-statistics.html#normal-distribution-model"><i class="fa fa-check"></i><b>1.3.3</b> Normal Distribution Model</a></li>
<li class="chapter" data-level="1.3.4" data-path="population-statistics.html"><a href="population-statistics.html#measures-of-center"><i class="fa fa-check"></i><b>1.3.4</b> Measures of Center</a></li>
<li class="chapter" data-level="1.3.5" data-path="population-statistics.html"><a href="population-statistics.html#measures-of-dispersion"><i class="fa fa-check"></i><b>1.3.5</b> Measures of Dispersion</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="distributions-and-probability.html"><a href="distributions-and-probability.html"><i class="fa fa-check"></i><b>2</b> Distributions and Probability</a>
<ul>
<li class="chapter" data-level="2.1" data-path="distributions-and-probability.html"><a href="distributions-and-probability.html#case-study"><i class="fa fa-check"></i><b>2.1</b> Case Study</a></li>
<li class="chapter" data-level="2.2" data-path="distributions-and-probability.html"><a href="distributions-and-probability.html#the-normal-distribution-model"><i class="fa fa-check"></i><b>2.2</b> The Normal Distribution Model</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="distributions-and-probability.html"><a href="distributions-and-probability.html#the-bell-curve"><i class="fa fa-check"></i><b>2.2.1</b> The Bell Curve</a></li>
<li class="chapter" data-level="2.2.2" data-path="distributions-and-probability.html"><a href="distributions-and-probability.html#distribution-and-probability"><i class="fa fa-check"></i><b>2.2.2</b> Distribution and Probability</a></li>
<li class="chapter" data-level="2.2.3" data-path="distributions-and-probability.html"><a href="distributions-and-probability.html#probability-and-the-normal-distribution-curve"><i class="fa fa-check"></i><b>2.2.3</b> Probability and the Normal Distribution Curve</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="distributions-and-probability.html"><a href="distributions-and-probability.html#the-z-distribution"><i class="fa fa-check"></i><b>2.3</b> The Z-Distribution</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="distributions-and-probability.html"><a href="distributions-and-probability.html#important-numbers-95-and-5"><i class="fa fa-check"></i><b>2.3.1</b> Important Numbers: 95% and 5%</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="sample-statistics.html"><a href="sample-statistics.html"><i class="fa fa-check"></i><b>3</b> Sample Statistics</a>
<ul>
<li class="chapter" data-level="3.1" data-path="sample-statistics.html"><a href="sample-statistics.html#samples"><i class="fa fa-check"></i><b>3.1</b> Samples</a></li>
<li class="chapter" data-level="3.2" data-path="sample-statistics.html"><a href="sample-statistics.html#case-study-1"><i class="fa fa-check"></i><b>3.2</b> Case Study</a></li>
<li class="chapter" data-level="3.3" data-path="sample-statistics.html"><a href="sample-statistics.html#distribution-of-sample-means"><i class="fa fa-check"></i><b>3.3</b> Distribution of Sample Means</a></li>
<li class="chapter" data-level="3.4" data-path="sample-statistics.html"><a href="sample-statistics.html#central-limit-theorem"><i class="fa fa-check"></i><b>3.4</b> Central Limit Theorem</a></li>
<li class="chapter" data-level="3.5" data-path="sample-statistics.html"><a href="sample-statistics.html#standard-error"><i class="fa fa-check"></i><b>3.5</b> Standard Error</a></li>
<li class="chapter" data-level="3.6" data-path="sample-statistics.html"><a href="sample-statistics.html#degrees-of-freedom"><i class="fa fa-check"></i><b>3.6</b> Degrees of Freedom</a></li>
<li class="chapter" data-level="3.7" data-path="sample-statistics.html"><a href="sample-statistics.html#the-t-distribution"><i class="fa fa-check"></i><b>3.7</b> The t-Distribution</a></li>
<li class="chapter" data-level="3.8" data-path="sample-statistics.html"><a href="sample-statistics.html#confidence-interval"><i class="fa fa-check"></i><b>3.8</b> Confidence Interval</a></li>
<li class="chapter" data-level="3.9" data-path="sample-statistics.html"><a href="sample-statistics.html#confidence-interval-and-probability"><i class="fa fa-check"></i><b>3.9</b> Confidence Interval and Probability</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="two-treatment-comparisons.html"><a href="two-treatment-comparisons.html"><i class="fa fa-check"></i><b>4</b> Two-Treatment Comparisons</a>
<ul>
<li class="chapter" data-level="4.1" data-path="two-treatment-comparisons.html"><a href="two-treatment-comparisons.html#side-by-side-trials"><i class="fa fa-check"></i><b>4.1</b> Side-by-Side Trials</a></li>
<li class="chapter" data-level="4.2" data-path="two-treatment-comparisons.html"><a href="two-treatment-comparisons.html#blocked-design"><i class="fa fa-check"></i><b>4.2</b> Blocked Design</a></li>
<li class="chapter" data-level="4.3" data-path="two-treatment-comparisons.html"><a href="two-treatment-comparisons.html#case-study-2"><i class="fa fa-check"></i><b>4.3</b> Case Study</a></li>
<li class="chapter" data-level="4.4" data-path="two-treatment-comparisons.html"><a href="two-treatment-comparisons.html#confidence-interval-1"><i class="fa fa-check"></i><b>4.4</b> Confidence Interval</a></li>
<li class="chapter" data-level="4.5" data-path="two-treatment-comparisons.html"><a href="two-treatment-comparisons.html#t-test"><i class="fa fa-check"></i><b>4.5</b> T-Test</a></li>
<li class="chapter" data-level="4.6" data-path="two-treatment-comparisons.html"><a href="two-treatment-comparisons.html#conclusion"><i class="fa fa-check"></i><b>4.6</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="understanding-statistical-tests.html"><a href="understanding-statistical-tests.html"><i class="fa fa-check"></i><b>5</b> Understanding Statistical Tests</a>
<ul>
<li class="chapter" data-level="5.1" data-path="understanding-statistical-tests.html"><a href="understanding-statistical-tests.html#research-question"><i class="fa fa-check"></i><b>5.1</b> Research Question</a></li>
<li class="chapter" data-level="5.2" data-path="understanding-statistical-tests.html"><a href="understanding-statistical-tests.html#the-model"><i class="fa fa-check"></i><b>5.2</b> The Model</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="understanding-statistical-tests.html"><a href="understanding-statistical-tests.html#treatment-effect"><i class="fa fa-check"></i><b>5.2.1</b> Treatment Effect</a></li>
<li class="chapter" data-level="5.2.2" data-path="understanding-statistical-tests.html"><a href="understanding-statistical-tests.html#error-effect"><i class="fa fa-check"></i><b>5.2.2</b> Error Effect</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="understanding-statistical-tests.html"><a href="understanding-statistical-tests.html#hypotheses"><i class="fa fa-check"></i><b>5.3</b> Hypotheses</a></li>
<li class="chapter" data-level="5.4" data-path="understanding-statistical-tests.html"><a href="understanding-statistical-tests.html#p-value"><i class="fa fa-check"></i><b>5.4</b> P-Value</a></li>
<li class="chapter" data-level="5.5" data-path="understanding-statistical-tests.html"><a href="understanding-statistical-tests.html#the-p-value-and-errors"><i class="fa fa-check"></i><b>5.5</b> The P-Value and Errors</a></li>
<li class="chapter" data-level="5.6" data-path="understanding-statistical-tests.html"><a href="understanding-statistical-tests.html#one-sided-vs-two-sided-hypotheses"><i class="fa fa-check"></i><b>5.6</b> One-Sided vs Two-Sided Hypotheses</a></li>
<li class="chapter" data-level="5.7" data-path="understanding-statistical-tests.html"><a href="understanding-statistical-tests.html#exercise-linear-additive-model"><i class="fa fa-check"></i><b>5.7</b> Exercise: Linear Additive Model</a>
<ul>
<li class="chapter" data-level="5.7.1" data-path="understanding-statistical-tests.html"><a href="understanding-statistical-tests.html#case-study-barley-effects"><i class="fa fa-check"></i><b>5.7.1</b> Case Study: Barley Effects</a></li>
<li class="chapter" data-level="5.7.2" data-path="understanding-statistical-tests.html"><a href="understanding-statistical-tests.html#plotting-the-effects"><i class="fa fa-check"></i><b>5.7.2</b> Plotting the Effects</a></li>
<li class="chapter" data-level="5.7.3" data-path="understanding-statistical-tests.html"><a href="understanding-statistical-tests.html#examining-individual-plots"><i class="fa fa-check"></i><b>5.7.3</b> Examining Individual Plots</a></li>
<li class="chapter" data-level="5.7.4" data-path="understanding-statistical-tests.html"><a href="understanding-statistical-tests.html#practice-groudnut"><i class="fa fa-check"></i><b>5.7.4</b> Practice: Groudnut</a></li>
<li class="chapter" data-level="5.7.5" data-path="understanding-statistical-tests.html"><a href="understanding-statistical-tests.html#plotting-the-effects-1"><i class="fa fa-check"></i><b>5.7.5</b> Plotting the Effects</a></li>
</ul></li>
<li class="chapter" data-level="5.8" data-path="understanding-statistical-tests.html"><a href="understanding-statistical-tests.html#exercise-one-sided-hypotheses"><i class="fa fa-check"></i><b>5.8</b> Exercise: One-Sided Hypotheses</a>
<ul>
<li class="chapter" data-level="5.8.1" data-path="understanding-statistical-tests.html"><a href="understanding-statistical-tests.html#case-study-groundnut"><i class="fa fa-check"></i><b>5.8.1</b> Case Study: Groundnut</a></li>
<li class="chapter" data-level="5.8.2" data-path="understanding-statistical-tests.html"><a href="understanding-statistical-tests.html#one-sided-t-test"><i class="fa fa-check"></i><b>5.8.2</b> One-Sided T-Test</a></li>
<li class="chapter" data-level="5.8.3" data-path="understanding-statistical-tests.html"><a href="understanding-statistical-tests.html#practice-barley"><i class="fa fa-check"></i><b>5.8.3</b> Practice: Barley</a></li>
<li class="chapter" data-level="5.8.4" data-path="understanding-statistical-tests.html"><a href="understanding-statistical-tests.html#practice-strawberry"><i class="fa fa-check"></i><b>5.8.4</b> Practice: Strawberry</a></li>
</ul></li>
<li class="chapter" data-level="5.9" data-path="understanding-statistical-tests.html"><a href="understanding-statistical-tests.html#exercise-type-i-and-type-ii-errors"><i class="fa fa-check"></i><b>5.9</b> Exercise: Type I and Type II Errors</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html"><i class="fa fa-check"></i><b>6</b> Multiple Treatment Trials</a>
<ul>
<li class="chapter" data-level="6.1" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html#case-study-3"><i class="fa fa-check"></i><b>6.1</b> Case Study</a></li>
<li class="chapter" data-level="6.2" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html#the-linear-additive-model"><i class="fa fa-check"></i><b>6.2</b> The Linear Additive Model</a></li>
<li class="chapter" data-level="6.3" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html#analysis-of-variance"><i class="fa fa-check"></i><b>6.3</b> Analysis of Variance</a></li>
<li class="chapter" data-level="6.4" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html#the-f-statistic"><i class="fa fa-check"></i><b>6.4</b> The F statistic</a></li>
<li class="chapter" data-level="6.5" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html#the-anova-table"><i class="fa fa-check"></i><b>6.5</b> The ANOVA Table</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html#source-of-variation"><i class="fa fa-check"></i><b>6.5.1</b> Source of Variation</a></li>
<li class="chapter" data-level="6.5.2" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html#sum-of-squares-1"><i class="fa fa-check"></i><b>6.5.2</b> Sum of Squares</a></li>
<li class="chapter" data-level="6.5.3" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html#degrees-of-freedom-1"><i class="fa fa-check"></i><b>6.5.3</b> Degrees of Freedom</a></li>
<li class="chapter" data-level="6.5.4" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html#mean-square"><i class="fa fa-check"></i><b>6.5.4</b> Mean Square</a></li>
<li class="chapter" data-level="6.5.5" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html#f-value"><i class="fa fa-check"></i><b>6.5.5</b> F-Value</a></li>
<li class="chapter" data-level="6.5.6" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html#p-value-1"><i class="fa fa-check"></i><b>6.5.6</b> P-value</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html#visualizing-how-the-anova-table-relates-to-variance"><i class="fa fa-check"></i><b>6.6</b> Visualizing How the Anova Table Relates to Variance</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html#case-study-barley"><i class="fa fa-check"></i><b>6.6.1</b> Case Study: Barley</a></li>
<li class="chapter" data-level="6.6.2" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html#anova"><i class="fa fa-check"></i><b>6.6.2</b> ANOVA</a></li>
<li class="chapter" data-level="6.6.3" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html#calculating-the-coefficient-of-variation"><i class="fa fa-check"></i><b>6.6.3</b> Calculating the Coefficient of Variation</a></li>
<li class="chapter" data-level="6.6.4" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html#practice-beet-data"><i class="fa fa-check"></i><b>6.6.4</b> Practice: Beet Data</a></li>
<li class="chapter" data-level="6.6.5" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html#practice-potato-data"><i class="fa fa-check"></i><b>6.6.5</b> Practice: Potato Data</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html#exercise-treatment-means"><i class="fa fa-check"></i><b>6.7</b> Exercise: “Treatment Means”</a>
<ul>
<li class="chapter" data-level="6.7.1" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html#case-study-barley-1"><i class="fa fa-check"></i><b>6.7.1</b> Case Study: Barley</a></li>
<li class="chapter" data-level="6.7.2" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html#calculating-treatment-means"><i class="fa fa-check"></i><b>6.7.2</b> Calculating Treatment Means</a></li>
<li class="chapter" data-level="6.7.3" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html#plotting-the-means"><i class="fa fa-check"></i><b>6.7.3</b> Plotting the Means</a></li>
<li class="chapter" data-level="6.7.4" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html#practice-beet-data-1"><i class="fa fa-check"></i><b>6.7.4</b> Practice: Beet Data</a></li>
<li class="chapter" data-level="6.7.5" data-path="multiple-treatment-trials.html"><a href="multiple-treatment-trials.html#practice-potato-data-1"><i class="fa fa-check"></i><b>6.7.5</b> Practice: Potato Data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html"><i class="fa fa-check"></i><b>7</b> Multiple Treatment Designs</a>
<ul>
<li class="chapter" data-level="7.1" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#randomized-complete-block-design"><i class="fa fa-check"></i><b>7.1</b> Randomized Complete Block Design</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#case-study-randomized-complete-block-design"><i class="fa fa-check"></i><b>7.1.1</b> Case Study: Randomized Complete Block Design</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#factorial-design"><i class="fa fa-check"></i><b>7.2</b> Factorial Design</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#case-study-1-1"><i class="fa fa-check"></i><b>7.2.1</b> Case Study 1</a></li>
<li class="chapter" data-level="7.2.2" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#case-study-2-1"><i class="fa fa-check"></i><b>7.2.2</b> Case Study 2</a></li>
<li class="chapter" data-level="7.2.3" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#discussing-interactions"><i class="fa fa-check"></i><b>7.2.3</b> Discussing Interactions</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#split-plot-design"><i class="fa fa-check"></i><b>7.3</b> Split-Plot Design</a></li>
<li class="chapter" data-level="7.4" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#linear-additive-model-3"><i class="fa fa-check"></i><b>7.4</b> Linear Additive Model</a></li>
<li class="chapter" data-level="7.5" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#conclusion-1"><i class="fa fa-check"></i><b>7.5</b> Conclusion</a></li>
<li class="chapter" data-level="7.6" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#exercise-randomized-complete-block-design"><i class="fa fa-check"></i><b>7.6</b> Exercise: Randomized Complete Block Design</a>
<ul>
<li class="chapter" data-level="7.6.1" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#case-study-4"><i class="fa fa-check"></i><b>7.6.1</b> Case Study</a></li>
<li class="chapter" data-level="7.6.2" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#anova-1"><i class="fa fa-check"></i><b>7.6.2</b> ANOVA</a></li>
<li class="chapter" data-level="7.6.3" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#plotting-the-results"><i class="fa fa-check"></i><b>7.6.3</b> Plotting the Results</a></li>
<li class="chapter" data-level="7.6.4" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#practice"><i class="fa fa-check"></i><b>7.6.4</b> Practice</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#exercise-factorial-anova"><i class="fa fa-check"></i><b>7.7</b> Exercise: Factorial ANOVA</a>
<ul>
<li class="chapter" data-level="7.7.1" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#case-study-biochar"><i class="fa fa-check"></i><b>7.7.1</b> Case Study: Biochar</a></li>
<li class="chapter" data-level="7.7.2" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#anova-2"><i class="fa fa-check"></i><b>7.7.2</b> ANOVA</a></li>
<li class="chapter" data-level="7.7.3" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#interaction-plots"><i class="fa fa-check"></i><b>7.7.3</b> Interaction Plots</a></li>
<li class="chapter" data-level="7.7.4" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#testing-factors-individually"><i class="fa fa-check"></i><b>7.7.4</b> Testing Factors Individually</a></li>
<li class="chapter" data-level="7.7.5" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#bar-plots"><i class="fa fa-check"></i><b>7.7.5</b> Bar Plots</a></li>
<li class="chapter" data-level="7.7.6" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#practice-1"><i class="fa fa-check"></i><b>7.7.6</b> Practice</a></li>
</ul></li>
<li class="chapter" data-level="7.8" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#exercise-split-plot-design"><i class="fa fa-check"></i><b>7.8</b> Exercise: Split-Plot Design</a>
<ul>
<li class="chapter" data-level="7.8.1" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#case-study-corn-soybean-systems-trial"><i class="fa fa-check"></i><b>7.8.1</b> Case Study: Corn-Soybean Systems Trial</a></li>
<li class="chapter" data-level="7.8.2" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#anova-4"><i class="fa fa-check"></i><b>7.8.2</b> ANOVA</a></li>
<li class="chapter" data-level="7.8.3" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#interaction-plot"><i class="fa fa-check"></i><b>7.8.3</b> Interaction Plot</a></li>
<li class="chapter" data-level="7.8.4" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#bar-plot"><i class="fa fa-check"></i><b>7.8.4</b> Bar plot</a></li>
<li class="chapter" data-level="7.8.5" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#practice-2"><i class="fa fa-check"></i><b>7.8.5</b> Practice</a></li>
</ul></li>
<li class="chapter" data-level="7.9" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#exercise-experimental-design"><i class="fa fa-check"></i><b>7.9</b> Exercise: Experimental Design</a>
<ul>
<li class="chapter" data-level="7.9.1" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#completely-randomized-design"><i class="fa fa-check"></i><b>7.9.1</b> Completely Randomized Design</a></li>
<li class="chapter" data-level="7.9.2" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#randomized-complete-block-design-1"><i class="fa fa-check"></i><b>7.9.2</b> Randomized Complete Block Design</a></li>
<li class="chapter" data-level="7.9.3" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#factorial-design-1"><i class="fa fa-check"></i><b>7.9.3</b> Factorial Design</a></li>
<li class="chapter" data-level="7.9.4" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#split-plot-design-1"><i class="fa fa-check"></i><b>7.9.4</b> Split-Plot Design</a></li>
<li class="chapter" data-level="7.9.5" data-path="multiple-treatment-designs.html"><a href="multiple-treatment-designs.html#practice-3"><i class="fa fa-check"></i><b>7.9.5</b> Practice</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html"><i class="fa fa-check"></i><b>8</b> Means Separation and Data Presentation</a>
<ul>
<li class="chapter" data-level="8.1" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#case-study-5"><i class="fa fa-check"></i><b>8.1</b> Case Study</a></li>
<li class="chapter" data-level="8.2" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#least-significant-difference"><i class="fa fa-check"></i><b>8.2</b> Least Significant Difference</a></li>
<li class="chapter" data-level="8.3" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#lsd-output-in-r"><i class="fa fa-check"></i><b>8.3</b> LSD Output in R</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#statistics-table"><i class="fa fa-check"></i><b>8.3.1</b> Statistics Table</a></li>
<li class="chapter" data-level="8.3.2" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#means-table"><i class="fa fa-check"></i><b>8.3.2</b> Means Table</a></li>
<li class="chapter" data-level="8.3.3" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#groups-table"><i class="fa fa-check"></i><b>8.3.3</b> Groups Table</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#comparisonwise-versus-experimentwise-error"><i class="fa fa-check"></i><b>8.4</b> Comparisonwise versus Experimentwise Error</a></li>
<li class="chapter" data-level="8.5" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#tukeys-honest-significant-difference"><i class="fa fa-check"></i><b>8.5</b> Tukey’s Honest Significant Difference</a></li>
<li class="chapter" data-level="8.6" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#linear-contrast"><i class="fa fa-check"></i><b>8.6</b> Linear Contrast</a>
<ul>
<li class="chapter" data-level="8.6.1" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#coefficients"><i class="fa fa-check"></i><b>8.6.1</b> Coefficients</a></li>
<li class="chapter" data-level="8.6.2" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#linear-contrasts-with-r"><i class="fa fa-check"></i><b>8.6.2</b> Linear Contrasts with R</a></li>
</ul></li>
<li class="chapter" data-level="8.7" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#means-presentation"><i class="fa fa-check"></i><b>8.7</b> Means Presentation</a>
<ul>
<li class="chapter" data-level="8.7.1" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#means-tables"><i class="fa fa-check"></i><b>8.7.1</b> Means Tables</a></li>
<li class="chapter" data-level="8.7.2" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#plotting-means"><i class="fa fa-check"></i><b>8.7.2</b> Plotting Means</a></li>
</ul></li>
<li class="chapter" data-level="8.8" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#exercise-lsd-and-tukeys-hsd"><i class="fa fa-check"></i><b>8.8</b> Exercise: LSD and Tukey’s HSD</a>
<ul>
<li class="chapter" data-level="8.8.1" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#case-study-common-bean"><i class="fa fa-check"></i><b>8.8.1</b> Case Study: Common Bean</a></li>
<li class="chapter" data-level="8.8.2" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#least-significant-difference-1"><i class="fa fa-check"></i><b>8.8.2</b> Least Significant Difference</a></li>
<li class="chapter" data-level="8.8.3" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#tukey-hsd"><i class="fa fa-check"></i><b>8.8.3</b> Tukey HSD</a></li>
<li class="chapter" data-level="8.8.4" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#practice-apple"><i class="fa fa-check"></i><b>8.8.4</b> Practice: Apple</a></li>
<li class="chapter" data-level="8.8.5" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#practice-wheat-treatment-with-mildew"><i class="fa fa-check"></i><b>8.8.5</b> Practice: Wheat Treatment with Mildew</a></li>
</ul></li>
<li class="chapter" data-level="8.9" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#exercise-linear-contrasts"><i class="fa fa-check"></i><b>8.9</b> Exercise: Linear Contrasts</a>
<ul>
<li class="chapter" data-level="8.9.1" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#case-study-winter-canola-cultivar-trial."><i class="fa fa-check"></i><b>8.9.1</b> Case Study: Winter Canola Cultivar Trial.</a></li>
<li class="chapter" data-level="8.9.2" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#running-the-contrast"><i class="fa fa-check"></i><b>8.9.2</b> Running the Contrast</a></li>
<li class="chapter" data-level="8.9.3" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#practice-corn-nitrogen-source-and-timing"><i class="fa fa-check"></i><b>8.9.3</b> Practice: Corn Nitrogen Source and Timing</a></li>
</ul></li>
<li class="chapter" data-level="8.10" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#exercise-means-tables"><i class="fa fa-check"></i><b>8.10</b> Exercise: Means Tables</a>
<ul>
<li class="chapter" data-level="8.10.1" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#case-study-corn-nitrogen-source-and-timing"><i class="fa fa-check"></i><b>8.10.1</b> Case Study: Corn Nitrogen Source and Timing</a></li>
<li class="chapter" data-level="8.10.2" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#practice-canola"><i class="fa fa-check"></i><b>8.10.2</b> Practice: Canola</a></li>
<li class="chapter" data-level="8.10.3" data-path="means-separation-and-data-presentation.html"><a href="means-separation-and-data-presentation.html#practice-broccoli-lsd"><i class="fa fa-check"></i><b>8.10.3</b> Practice: Broccoli LSD</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html"><i class="fa fa-check"></i><b>9</b> Messy and Missing Data</a>
<ul>
<li class="chapter" data-level="9.1" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#inspecting-data-for-normal-distributions"><i class="fa fa-check"></i><b>9.1</b> Inspecting data for Normal Distributions</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#histograms-1"><i class="fa fa-check"></i><b>9.1.1</b> Histograms</a></li>
<li class="chapter" data-level="9.1.2" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#rank-percentile-plots"><i class="fa fa-check"></i><b>9.1.2</b> Rank Percentile Plots</a></li>
<li class="chapter" data-level="9.1.3" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#box-plots"><i class="fa fa-check"></i><b>9.1.3</b> Box Plots</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#inspecting-data-for-equal-variances"><i class="fa fa-check"></i><b>9.2</b> Inspecting Data for Equal Variances</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#mean-variance-plot"><i class="fa fa-check"></i><b>9.2.1</b> Mean-Variance Plot</a></li>
<li class="chapter" data-level="9.2.2" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#homogeneity-of-variance-tests"><i class="fa fa-check"></i><b>9.2.2</b> Homogeneity of Variance Tests</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#dealing-with-messy-data"><i class="fa fa-check"></i><b>9.3</b> Dealing with Messy Data</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#outliers"><i class="fa fa-check"></i><b>9.3.1</b> Outliers</a></li>
<li class="chapter" data-level="9.3.2" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#non-normal-data-and-unequal-variances"><i class="fa fa-check"></i><b>9.3.2</b> Non-normal Data and Unequal Variances</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#dealing-with-missing-data"><i class="fa fa-check"></i><b>9.4</b> Dealing with Missing Data</a></li>
<li class="chapter" data-level="9.5" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#summary"><i class="fa fa-check"></i><b>9.5</b> Summary</a></li>
<li class="chapter" data-level="9.6" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#exercise-visual-data-inspection"><i class="fa fa-check"></i><b>9.6</b> Exercise: Visual Data Inspection</a>
<ul>
<li class="chapter" data-level="9.6.1" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#case-study-1-2"><i class="fa fa-check"></i><b>9.6.1</b> Case Study 1</a></li>
<li class="chapter" data-level="9.6.2" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#histogram"><i class="fa fa-check"></i><b>9.6.2</b> Histogram</a></li>
<li class="chapter" data-level="9.6.3" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#rank-percentile"><i class="fa fa-check"></i><b>9.6.3</b> Rank-Percentile</a></li>
<li class="chapter" data-level="9.6.4" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#case-study-2-2"><i class="fa fa-check"></i><b>9.6.4</b> Case Study 2</a></li>
<li class="chapter" data-level="9.6.5" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#box-plot"><i class="fa fa-check"></i><b>9.6.5</b> Box Plot</a></li>
<li class="chapter" data-level="9.6.6" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#practice-4"><i class="fa fa-check"></i><b>9.6.6</b> Practice</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#exercise-identifying-unequal-variances"><i class="fa fa-check"></i><b>9.7</b> Exercise: Identifying Unequal Variances"</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#case-study-grape-colapsis-in-seed-corn"><i class="fa fa-check"></i><b>9.7.1</b> Case Study: Grape Colapsis in Seed Corn</a></li>
<li class="chapter" data-level="9.7.2" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#practice-pigweed-height"><i class="fa fa-check"></i><b>9.7.2</b> Practice: Pigweed Height</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#exercise-transforming-and-analyzing-data"><i class="fa fa-check"></i><b>9.8</b> Exercise: Transforming and Analyzing Data</a>
<ul>
<li class="chapter" data-level="9.8.1" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#case-study-grape-colapsis-in-seed-corn-1"><i class="fa fa-check"></i><b>9.8.1</b> Case Study: Grape Colapsis in Seed Corn</a></li>
<li class="chapter" data-level="9.8.2" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#practice-pigweed-height-1"><i class="fa fa-check"></i><b>9.8.2</b> Practice: Pigweed Height</a></li>
<li class="chapter" data-level="9.8.3" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#practice-pigweed-biomass"><i class="fa fa-check"></i><b>9.8.3</b> Practice: Pigweed Biomass</a></li>
</ul></li>
<li class="chapter" data-level="9.9" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#exercise-imputing-missing-data"><i class="fa fa-check"></i><b>9.9</b> Exercise: Imputing Missing Data</a>
<ul>
<li class="chapter" data-level="9.9.1" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#case-study-wheat-genotype-trial"><i class="fa fa-check"></i><b>9.9.1</b> Case Study: Wheat Genotype Trial</a></li>
<li class="chapter" data-level="9.9.2" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#practice-1-1"><i class="fa fa-check"></i><b>9.9.2</b> Practice 1</a></li>
<li class="chapter" data-level="9.9.3" data-path="messy-and-missing-data.html"><a href="messy-and-missing-data.html#practice-2-1"><i class="fa fa-check"></i><b>9.9.3</b> Practice 2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html"><i class="fa fa-check"></i><b>10</b> Correlation and Simple Regression</a>
<ul>
<li class="chapter" data-level="10.1" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#correlation"><i class="fa fa-check"></i><b>10.1</b> Correlation</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#case-study-cucumber"><i class="fa fa-check"></i><b>10.1.1</b> Case Study: Cucumber</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#correlation-1"><i class="fa fa-check"></i><b>10.2</b> Correlation</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#how-is-r-calcuated-optional-reading"><i class="fa fa-check"></i><b>10.2.1</b> How Is r Calcuated (optional reading)</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#regression"><i class="fa fa-check"></i><b>10.3</b> Regression</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#case-study-6"><i class="fa fa-check"></i><b>10.3.1</b> Case Study</a></li>
<li class="chapter" data-level="10.3.2" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#linear-equation"><i class="fa fa-check"></i><b>10.3.2</b> Linear Equation</a></li>
<li class="chapter" data-level="10.3.3" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#calculating-the-regression-equation"><i class="fa fa-check"></i><b>10.3.3</b> Calculating the Regression Equation</a></li>
<li class="chapter" data-level="10.3.4" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#least-squares-estimate"><i class="fa fa-check"></i><b>10.3.4</b> Least-Squares Estimate</a></li>
<li class="chapter" data-level="10.3.5" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#significance-of-coefficients"><i class="fa fa-check"></i><b>10.3.5</b> Significance of Coefficients</a></li>
<li class="chapter" data-level="10.3.6" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#analysis-of-variance-6"><i class="fa fa-check"></i><b>10.3.6</b> Analysis of Variance</a></li>
<li class="chapter" data-level="10.3.7" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#measuring-model-fit-with-r-square"><i class="fa fa-check"></i><b>10.3.7</b> Measuring Model Fit with R-Square</a></li>
<li class="chapter" data-level="10.3.8" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#checking-whether-the-linear-model-is-appropriate"><i class="fa fa-check"></i><b>10.3.8</b> Checking whether the Linear Model is Appropriate</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#extrapolation"><i class="fa fa-check"></i><b>10.4</b> Extrapolation</a></li>
<li class="chapter" data-level="10.5" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#exercise-sccatterplots-and-regression"><i class="fa fa-check"></i><b>10.5</b> Exercise: Sccatterplots and Regression</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#case-study-corn-data"><i class="fa fa-check"></i><b>10.5.1</b> Case Study: Corn Data</a></li>
<li class="chapter" data-level="10.5.2" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#practice-1-2"><i class="fa fa-check"></i><b>10.5.2</b> Practice 1</a></li>
<li class="chapter" data-level="10.5.3" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#practice-2-2"><i class="fa fa-check"></i><b>10.5.3</b> Practice 2</a></li>
<li class="chapter" data-level="10.5.4" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#practice-3-1"><i class="fa fa-check"></i><b>10.5.4</b> Practice 3</a></li>
<li class="chapter" data-level="10.5.5" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#practice-4-1"><i class="fa fa-check"></i><b>10.5.5</b> Practice 4</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#exercise-simple-linear-regression"><i class="fa fa-check"></i><b>10.6</b> Exercise: Simple Linear Regression</a>
<ul>
<li class="chapter" data-level="10.6.1" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#case-study-corn-allometry"><i class="fa fa-check"></i><b>10.6.1</b> Case Study: Corn Allometry</a></li>
<li class="chapter" data-level="10.6.2" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#fitting-the-regression-model"><i class="fa fa-check"></i><b>10.6.2</b> Fitting the regression model</a></li>
<li class="chapter" data-level="10.6.3" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#examining-the-residuals"><i class="fa fa-check"></i><b>10.6.3</b> Examining the residuals</a></li>
<li class="chapter" data-level="10.6.4" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#viewing-the-model-coefficients"><i class="fa fa-check"></i><b>10.6.4</b> Viewing the Model Coefficients</a></li>
<li class="chapter" data-level="10.6.5" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#more-measures-of-model-fit"><i class="fa fa-check"></i><b>10.6.5</b> More measures of model fit</a></li>
<li class="chapter" data-level="10.6.6" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#practice-1-3"><i class="fa fa-check"></i><b>10.6.6</b> Practice 1</a></li>
<li class="chapter" data-level="10.6.7" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#practice-2-3"><i class="fa fa-check"></i><b>10.6.7</b> Practice 2</a></li>
</ul></li>
<li class="chapter" data-level="10.7" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#exercise-making-predictions-from-regression-models."><i class="fa fa-check"></i><b>10.7</b> Exercise: Making Predictions from Regression Models.</a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#case-study-7"><i class="fa fa-check"></i><b>10.7.1</b> Case Study</a></li>
<li class="chapter" data-level="10.7.2" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#creating-and-testing-our-model"><i class="fa fa-check"></i><b>10.7.2</b> Creating and Testing our Model</a></li>
<li class="chapter" data-level="10.7.3" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#visualizing-the-regression-model"><i class="fa fa-check"></i><b>10.7.3</b> Visualizing the Regression Model</a></li>
<li class="chapter" data-level="10.7.4" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#making-predictions-with-the-regression-model"><i class="fa fa-check"></i><b>10.7.4</b> Making Predictions with the Regression Model</a></li>
<li class="chapter" data-level="10.7.5" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#practice-1-4"><i class="fa fa-check"></i><b>10.7.5</b> Practice 1</a></li>
<li class="chapter" data-level="10.7.6" data-path="correlation-and-simple-regression.html"><a href="correlation-and-simple-regression.html#practice-2-4"><i class="fa fa-check"></i><b>10.7.6</b> Practice 2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html"><i class="fa fa-check"></i><b>11</b> Nonlinear Relationships and Multiple Linear Regression</a>
<ul>
<li class="chapter" data-level="11.1" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#multiplie-linear-regression"><i class="fa fa-check"></i><b>11.1</b> Multiplie Linear Regression</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#case-study-modelling-yield-by-county"><i class="fa fa-check"></i><b>11.1.1</b> Case Study: Modelling Yield by County</a></li>
<li class="chapter" data-level="11.1.2" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#beware-of-bloated-models"><i class="fa fa-check"></i><b>11.1.2</b> Beware of Bloated Models</a></li>
<li class="chapter" data-level="11.1.3" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#methods-for-avoiding-bloated-models"><i class="fa fa-check"></i><b>11.1.3</b> Methods for Avoiding Bloated Models</a></li>
<li class="chapter" data-level="11.1.4" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#tunning-and-comparing-models"><i class="fa fa-check"></i><b>11.1.4</b> Tunning and Comparing Models</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#nonlinear-relationships"><i class="fa fa-check"></i><b>11.2</b> Nonlinear Relationships</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#fitting-nonlinear-responses-with-linear-regression"><i class="fa fa-check"></i><b>11.2.1</b> Fitting Nonlinear Responses with Linear Regression</a></li>
<li class="chapter" data-level="11.2.2" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#fitting-nonlinear-responses-with-nonlinear-regression"><i class="fa fa-check"></i><b>11.2.2</b> Fitting Nonlinear Responses with Nonlinear Regression</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#summary-1"><i class="fa fa-check"></i><b>11.3</b> Summary</a></li>
<li class="chapter" data-level="11.4" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#exercise-multiple-linear-regression"><i class="fa fa-check"></i><b>11.4</b> Exercise: Multiple Linear Regression</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#case-study-country-corn-yields"><i class="fa fa-check"></i><b>11.4.1</b> Case Study: Country Corn Yields</a></li>
<li class="chapter" data-level="11.4.2" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#fitting-the-model"><i class="fa fa-check"></i><b>11.4.2</b> Fitting the Model</a></li>
<li class="chapter" data-level="11.4.3" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#inspecting-the-model"><i class="fa fa-check"></i><b>11.4.3</b> Inspecting the Model</a></li>
<li class="chapter" data-level="11.4.4" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#practice-thompson-corn-data"><i class="fa fa-check"></i><b>11.4.4</b> Practice: Thompson Corn Data</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#exercise-avoiding-bloated-models"><i class="fa fa-check"></i><b>11.5</b> Exercise: Avoiding Bloated Models</a>
<ul>
<li class="chapter" data-level="11.5.1" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#case-study-8"><i class="fa fa-check"></i><b>11.5.1</b> Case Study</a></li>
<li class="chapter" data-level="11.5.2" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#covariance-matrix"><i class="fa fa-check"></i><b>11.5.2</b> Covariance Matrix</a></li>
<li class="chapter" data-level="11.5.3" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#partial-correlation-1"><i class="fa fa-check"></i><b>11.5.3</b> Partial Correlation</a></li>
<li class="chapter" data-level="11.5.4" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#cross-validation-1"><i class="fa fa-check"></i><b>11.5.4</b> Cross-Validation</a></li>
<li class="chapter" data-level="11.5.5" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#practice-thompson-corn-data-1"><i class="fa fa-check"></i><b>11.5.5</b> Practice: Thompson Corn Data</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#exercise-tuning-the-model"><i class="fa fa-check"></i><b>11.6</b> Exercise: Tuning the Model</a>
<ul>
<li class="chapter" data-level="11.6.1" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#case-study-9"><i class="fa fa-check"></i><b>11.6.1</b> Case Study</a></li>
<li class="chapter" data-level="11.6.2" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#tuning-the-model"><i class="fa fa-check"></i><b>11.6.2</b> Tuning the Model</a></li>
<li class="chapter" data-level="11.6.3" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#practice-5"><i class="fa fa-check"></i><b>11.6.3</b> Practice</a></li>
<li class="chapter" data-level="11.6.4" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#practice-6"><i class="fa fa-check"></i><b>11.6.4</b> Practice</a></li>
</ul></li>
<li class="chapter" data-level="11.7" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#exercise-nonlinear-regression"><i class="fa fa-check"></i><b>11.7</b> Exercise: Nonlinear Regression</a>
<ul>
<li class="chapter" data-level="11.7.1" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#case-study-1-monomolecular-data"><i class="fa fa-check"></i><b>11.7.1</b> Case Study 1: Monomolecular Data</a></li>
<li class="chapter" data-level="11.7.2" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#case-study-2-logistic-data"><i class="fa fa-check"></i><b>11.7.2</b> Case Study 2: Logistic Data</a></li>
<li class="chapter" data-level="11.7.3" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#practice-1-5"><i class="fa fa-check"></i><b>11.7.3</b> Practice 1</a></li>
<li class="chapter" data-level="11.7.4" data-path="nonlinear-relationships-and-multiple-linear-regression.html"><a href="nonlinear-relationships-and-multiple-linear-regression.html#practice-2-5"><i class="fa fa-check"></i><b>11.7.4</b> Practice 2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="spatial-statistics.html"><a href="spatial-statistics.html"><i class="fa fa-check"></i><b>12</b> Spatial Statistics</a>
<ul>
<li class="chapter" data-level="12.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#projection-general"><i class="fa fa-check"></i><b>12.1</b> Projection (General)</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#wgs-84-epsg-4236"><i class="fa fa-check"></i><b>12.1.1</b> WGS 84 (EPSG: 4236)</a></li>
<li class="chapter" data-level="12.1.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#mercator-epsg-3857"><i class="fa fa-check"></i><b>12.1.2</b> Mercator (EPSG: 3857)</a></li>
<li class="chapter" data-level="12.1.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#us-national-atlas-equal-area-epsg-2163"><i class="fa fa-check"></i><b>12.1.3</b> US National Atlas Equal Area (EPSG: 2163)</a></li>
<li class="chapter" data-level="12.1.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#utm-zone-11n-epsg-2955"><i class="fa fa-check"></i><b>12.1.4</b> UTM Zone 11N (EPSG: 2955)</a></li>
<li class="chapter" data-level="12.1.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#projection-summary"><i class="fa fa-check"></i><b>12.1.5</b> Projection Summary</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#shape-files"><i class="fa fa-check"></i><b>12.2</b> Shape Files</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#case-study-soybean-yield-in-iowa"><i class="fa fa-check"></i><b>12.2.1</b> Case Study: Soybean Yield in Iowa</a></li>
<li class="chapter" data-level="12.2.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#ssurgo"><i class="fa fa-check"></i><b>12.2.2</b> SSURGO</a></li>
<li class="chapter" data-level="12.2.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#operations-with-shapes"><i class="fa fa-check"></i><b>12.2.3</b> Operations with Shapes</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#rasters"><i class="fa fa-check"></i><b>12.3</b> Rasters</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#interpolation"><i class="fa fa-check"></i><b>12.3.1</b> Interpolation</a></li>
<li class="chapter" data-level="12.3.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#kriging"><i class="fa fa-check"></i><b>12.3.2</b> Kriging</a></li>
<li class="chapter" data-level="12.3.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#operations-on-kriged-data"><i class="fa fa-check"></i><b>12.3.3</b> Operations on Kriged Data</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="spatial-statistics.html"><a href="spatial-statistics.html#exercise-shape-files"><i class="fa fa-check"></i><b>12.4</b> Exercise: Shape Files</a>
<ul>
<li class="chapter" data-level="12.4.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#case-study-10"><i class="fa fa-check"></i><b>12.4.1</b> Case Study</a></li>
<li class="chapter" data-level="12.4.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#practice-7"><i class="fa fa-check"></i><b>12.4.2</b> Practice</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="spatial-statistics.html"><a href="spatial-statistics.html#exercise-ssurgo"><i class="fa fa-check"></i><b>12.5</b> Exercise: SSURGO</a></li>
<li class="chapter" data-level="12.6" data-path="spatial-statistics.html"><a href="spatial-statistics.html#exercise-rasters"><i class="fa fa-check"></i><b>12.6</b> Exercise: Rasters</a>
<ul>
<li class="chapter" data-level="12.6.1" data-path="spatial-statistics.html"><a href="spatial-statistics.html#case-study-soil-test-data"><i class="fa fa-check"></i><b>12.6.1</b> Case Study: Soil Test Data</a></li>
<li class="chapter" data-level="12.6.2" data-path="spatial-statistics.html"><a href="spatial-statistics.html#practice-1-6"><i class="fa fa-check"></i><b>12.6.2</b> Practice 1</a></li>
<li class="chapter" data-level="12.6.3" data-path="spatial-statistics.html"><a href="spatial-statistics.html#practice-2-6"><i class="fa fa-check"></i><b>12.6.3</b> Practice 2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="machine-learning.html"><a href="machine-learning.html"><i class="fa fa-check"></i><b>13</b> Machine Learning</a>
<ul>
<li class="chapter" data-level="13.1" data-path="machine-learning.html"><a href="machine-learning.html#machine-learning-1"><i class="fa fa-check"></i><b>13.1</b> Machine Learning</a></li>
<li class="chapter" data-level="13.2" data-path="machine-learning.html"><a href="machine-learning.html#cluster-analyses"><i class="fa fa-check"></i><b>13.2</b> Cluster Analyses</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="machine-learning.html"><a href="machine-learning.html#case-study-grouping-midwestern-environments"><i class="fa fa-check"></i><b>13.2.1</b> Case Study: Grouping Midwestern Environments</a></li>
<li class="chapter" data-level="13.2.2" data-path="machine-learning.html"><a href="machine-learning.html#scaling"><i class="fa fa-check"></i><b>13.2.2</b> Scaling</a></li>
<li class="chapter" data-level="13.2.3" data-path="machine-learning.html"><a href="machine-learning.html#clustering-animation"><i class="fa fa-check"></i><b>13.2.3</b> Clustering Animation</a></li>
<li class="chapter" data-level="13.2.4" data-path="machine-learning.html"><a href="machine-learning.html#county-cluster-analysis"><i class="fa fa-check"></i><b>13.2.4</b> County Cluster Analysis</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="machine-learning.html"><a href="machine-learning.html#k-nearest-neighbors"><i class="fa fa-check"></i><b>13.3</b> k-Nearest-Neighbors</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="machine-learning.html"><a href="machine-learning.html#case-study-guessing-county-yields-based-on-environmental-similarity"><i class="fa fa-check"></i><b>13.3.1</b> Case Study: Guessing County Yields based on Environmental Similarity</a></li>
<li class="chapter" data-level="13.3.2" data-path="machine-learning.html"><a href="machine-learning.html#scaling-1"><i class="fa fa-check"></i><b>13.3.2</b> Scaling</a></li>
<li class="chapter" data-level="13.3.3" data-path="machine-learning.html"><a href="machine-learning.html#k-nearest-neighbor-animation"><i class="fa fa-check"></i><b>13.3.3</b> k-Nearest-Neighbor Animation</a></li>
<li class="chapter" data-level="13.3.4" data-path="machine-learning.html"><a href="machine-learning.html#choosing-k"><i class="fa fa-check"></i><b>13.3.4</b> Choosing <span class="math inline">\(k\)</span></a></li>
<li class="chapter" data-level="13.3.5" data-path="machine-learning.html"><a href="machine-learning.html#model-cross-validation"><i class="fa fa-check"></i><b>13.3.5</b> Model Cross-Validation</a></li>
<li class="chapter" data-level="13.3.6" data-path="machine-learning.html"><a href="machine-learning.html#yield-prediction-with-nearest-neighbor-analysis"><i class="fa fa-check"></i><b>13.3.6</b> Yield Prediction with Nearest Neighbor Analysis</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="machine-learning.html"><a href="machine-learning.html#classification-trees"><i class="fa fa-check"></i><b>13.4</b> Classification Trees</a>
<ul>
<li class="chapter" data-level="13.4.1" data-path="machine-learning.html"><a href="machine-learning.html#features"><i class="fa fa-check"></i><b>13.4.1</b> Features</a></li>
<li class="chapter" data-level="13.4.2" data-path="machine-learning.html"><a href="machine-learning.html#quantitative-categorical-data"><i class="fa fa-check"></i><b>13.4.2</b> Quantitative (Categorical) Data</a></li>
<li class="chapter" data-level="13.4.3" data-path="machine-learning.html"><a href="machine-learning.html#quanatitiative-continuous-data"><i class="fa fa-check"></i><b>13.4.3</b> Quanatitiative (Continuous) Data</a></li>
<li class="chapter" data-level="13.4.4" data-path="machine-learning.html"><a href="machine-learning.html#overfitting-1"><i class="fa fa-check"></i><b>13.4.4</b> Overfitting</a></li>
<li class="chapter" data-level="13.4.5" data-path="machine-learning.html"><a href="machine-learning.html#cross-validation-2"><i class="fa fa-check"></i><b>13.4.5</b> Cross Validation</a></li>
<li class="chapter" data-level="13.4.6" data-path="machine-learning.html"><a href="machine-learning.html#random-forest"><i class="fa fa-check"></i><b>13.4.6</b> Random Forest</a></li>
<li class="chapter" data-level="13.4.7" data-path="machine-learning.html"><a href="machine-learning.html#feature-importance"><i class="fa fa-check"></i><b>13.4.7</b> Feature Importance</a></li>
</ul></li>
<li class="chapter" data-level="13.5" data-path="machine-learning.html"><a href="machine-learning.html#summary-2"><i class="fa fa-check"></i><b>13.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="putting-it-all-together.html"><a href="putting-it-all-together.html"><i class="fa fa-check"></i><b>14</b> Putting it all Together</a>
<ul>
<li class="chapter" data-level="14.1" data-path="putting-it-all-together.html"><a href="putting-it-all-together.html#scenario-1-yield-map-population-summary-and-z-distribution"><i class="fa fa-check"></i><b>14.1</b> Scenario 1: Yield Map (Population Summary and Z-Distribution)</a></li>
<li class="chapter" data-level="14.2" data-path="putting-it-all-together.html"><a href="putting-it-all-together.html#scenario-2-yield-estimate-sampling-t-distribution"><i class="fa fa-check"></i><b>14.2</b> Scenario 2: Yield Estimate (Sampling t-Distribution)</a></li>
<li class="chapter" data-level="14.3" data-path="putting-it-all-together.html"><a href="putting-it-all-together.html#scenario-3-side-by-side-t-test"><i class="fa fa-check"></i><b>14.3</b> Scenario 3: Side-By-Side (t-Test)</a></li>
<li class="chapter" data-level="14.4" data-path="putting-it-all-together.html"><a href="putting-it-all-together.html#scenario-4-fungicide-trial-anova-crd-or-rcbd"><i class="fa fa-check"></i><b>14.4</b> Scenario 4: Fungicide Trial (ANOVA CRD or RCBD)</a></li>
<li class="chapter" data-level="14.5" data-path="putting-it-all-together.html"><a href="putting-it-all-together.html#scenario-5-hybrid-response-to-fungicide-trial-anova-factorial-or-split-plot"><i class="fa fa-check"></i><b>14.5</b> Scenario 5: Hybrid Response to Fungicide Trial (ANOVA Factorial or Split Plot)</a></li>
<li class="chapter" data-level="14.6" data-path="putting-it-all-together.html"><a href="putting-it-all-together.html#scenario-6-foliar-rate-response-trial-linear-or-non-linear-regression"><i class="fa fa-check"></i><b>14.6</b> Scenario 6: Foliar Rate-Response Trial (Linear or Non-Linear Regression)</a></li>
<li class="chapter" data-level="14.7" data-path="putting-it-all-together.html"><a href="putting-it-all-together.html#scenario-7-application-map-shapefiles-and-rasters"><i class="fa fa-check"></i><b>14.7</b> Scenario 7: Application Map (Shapefiles and Rasters)</a></li>
<li class="chapter" data-level="14.8" data-path="putting-it-all-together.html"><a href="putting-it-all-together.html#scenario-8-yield-prediction-multiple-linear-regression-and-other-predictive-models"><i class="fa fa-check"></i><b>14.8</b> Scenario 8: Yield Prediction (Multiple Linear Regression and other Predictive Models)</a></li>
<li class="chapter" data-level="14.9" data-path="putting-it-all-together.html"><a href="putting-it-all-together.html#summary-3"><i class="fa fa-check"></i><b>14.9</b> Summary</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Science for Agricultural Professionals</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="means-separation-and-data-presentation" class="section level1" number="8">
<h1><span class="header-section-number">Chapter 8</span> Means Separation and Data Presentation</h1>
<p>The previous two units focused on the design and analysis of effects in multiple treatment files. Our focus was to determine whether treat ent effects explained more of the variation among individuals in a population than error (or residual) effects, which are based on unexplained differences among individuals.</p>
<p>In the first half of this unit, we will learn three common tools used for testing the differences <em>between</em> treatments. This is often the key purpose of a research trial. We know going in that some treatments will be different. But we don’t how they will rank, and whether the difference between them will be great enough to infer one is better than another.</p>
<p>In the second half, we will learn how to present treatment means in tables and plots. Proper data allows the reader to not only grasp results, but even incorporate some of your findings into their own work</p>
<div id="case-study-5" class="section level2" number="8.1">
<h2><span class="header-section-number">8.1</span> Case Study</h2>
<p>Our sample dataset is inspired by Salas, M.C.; Montero, J.L.; Diaz, J.G.; Berti, F.; Quintero, M.F.; Guzmán, M.; Orsini, F. Defining Optimal Strength of the Nutrient Solution for Soilless Cultivation of Saffron in the Mediterranean. Agronomy 2020, 10, 1311.</p>
<p>Saffron is a spice made from the anthers of the saffron flow. It has a nuanced, sweet, complex flavor, and is used in dishes from Cornish saffron rolls to Spanish paella to Iranian tahdig. It comes from the anthers of the field-grown saffron flower and must be hand picked, making it very expensive.</p>
<p>In this study, saffron was grown hydroponically in 15-L pots filled with perlite, with four nutrient concentrations as defined by electroconductivity (EC): low (EC 2.0), medium (EC 2.5), high (EC 3.0), and very high (EC 4.0). The effect of the solutions on corm production (needed to propagate the crop) was measured.</p>
<div class="sourceCode" id="cb284"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb284-1"><a href="means-separation-and-data-presentation.html#cb284-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
<pre><code>## -- Attaching packages --------------------------------------- tidyverse 1.3.0 --</code></pre>
<pre><code>## v ggplot2 3.3.3     v purrr   0.3.4
## v tibble  3.1.0     v dplyr   1.0.5
## v tidyr   1.1.3     v stringr 1.4.0
## v readr   1.4.0     v forcats 0.5.1</code></pre>
<pre><code>## -- Conflicts ------------------------------------------ tidyverse_conflicts() --
## x dplyr::filter() masks stats::filter()
## x dplyr::lag()    masks stats::lag()</code></pre>
<div class="sourceCode" id="cb288"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb288-1"><a href="means-separation-and-data-presentation.html#cb288-1" aria-hidden="true" tabindex="-1"></a>saffron <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data-unit-8/saffron.csv&quot;</span>)</span>
<span id="cb288-2"><a href="means-separation-and-data-presentation.html#cb288-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(saffron)</span></code></pre></div>
<pre><code>##   plot block rate corm_number
## 1    1     1   1X         246
## 2    2     1   4X         240
## 3    3     1   3X         254
## 4    4     1   2X         285
## 5    5     2   4X         233
## 6    6     2   3X         251</code></pre>
</div>
<div id="least-significant-difference" class="section level2" number="8.2">
<h2><span class="header-section-number">8.2</span> Least Significant Difference</h2>
<p>Perhaps the most straightforward method of means separation is the infamous Least Significant Difference test. Not to be confused with the psychadelic parties in Marin County, California, the LSD test is as simple as this:</p>
<ul>
<li>calculate the least significant difference</li>
<li>any treatment differences that are equal to or greater than the least significant difference are – you guessed it – significant</li>
</ul>
<p>The least significant difference is calculated as follows:</p>
<p><span class="math display">\[LSD_{df, \alpha} = t_{df, \alpha} \cdot SED\]</span></p>
<p>Where <span class="math inline">\(LSD_{df, \alpha}\)</span> is the least significant difference, given the degrees of freedom associated with the error effect and the desired signifance.</p>
<p>Does this formula look vaguely familiar? Does it remind you of what you were doing back in Unit 4? Great, because this is the same formula we used to calculate the distance between confidence limits and the sample mean back when we learned t-tests. Back then, we saw how the confidence interval was used to test the probability our observed difference between treatments was different from zero. Recall if zero fell outside our confidence interval, we inferred the two treatments were different. Similarly, if the difference between two treatments is greater than the least significant difference, we infer the treatments are significantly different.</p>
<p>In R, we use a function, <em>LSD.test()</em>, which is part of the <em>agricolae</em> package, to calculate the LSD. First, however, lets run an analysis of variance on our t
data. The experiment was a randomized complete block design, so our linear additive model is:</p>
<p><span class="math display">\[ Y_{ij} = \mu +B_i + T_j + BT_{ij}\]</span></p>
<p>Where <span class="math inline">\(Y_{ij}\)</span> is the number of corms, <span class="math inline">\(\mu\)</span> is the overall population mean for the trial, <span class="math inline">\(B_i\)</span> is the block effect, <span class="math inline">\(T_j\)</span> is the treatment effect, and <span class="math inline">\(BT_{ij}\)</span> is the block effect.</p>
<p>Our analysis of variance result is below. The effect of fertilization rate is highly significant. And this brings us to an important rule for using the LSD test. We only use the LSD test to separate means <em>if</em> the treatment effect is significant in our analysis of variance. Doing otherwise can lead to errors, as we will discuss below.</p>
<div class="sourceCode" id="cb290"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb290-1"><a href="means-separation-and-data-presentation.html#cb290-1" aria-hidden="true" tabindex="-1"></a>saffron<span class="sc">$</span>block <span class="ot">=</span> <span class="fu">as.factor</span>(saffron<span class="sc">$</span>block)</span>
<span id="cb290-2"><a href="means-separation-and-data-presentation.html#cb290-2" aria-hidden="true" tabindex="-1"></a>saffron_model <span class="ot">=</span> <span class="fu">aov</span>(corm_number <span class="sc">~</span> block <span class="sc">+</span> rate, saffron)</span>
<span id="cb290-3"><a href="means-separation-and-data-presentation.html#cb290-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(saffron_model)</span></code></pre></div>
<pre><code>##             Df Sum Sq Mean Sq F value   Pr(&gt;F)    
## block        3     85    28.4   2.465    0.129    
## rate         3   4473  1490.9 129.333 1.02e-07 ***
## Residuals    9    104    11.5                     
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
<div id="lsd-output-in-r" class="section level2" number="8.3">
<h2><span class="header-section-number">8.3</span> LSD Output in R</h2>
<p>Now that we know the effect of fertilization rate is highly significant, we want to know how the individual treatments rank, and whether they are significantly different from one another. The results of our LSD test are below.</p>
<div class="sourceCode" id="cb292"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb292-1"><a href="means-separation-and-data-presentation.html#cb292-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(agricolae)</span>
<span id="cb292-2"><a href="means-separation-and-data-presentation.html#cb292-2" aria-hidden="true" tabindex="-1"></a>lsd <span class="ot">=</span> <span class="fu">LSD.test</span>(saffron_model, <span class="st">&quot;rate&quot;</span>)</span>
<span id="cb292-3"><a href="means-separation-and-data-presentation.html#cb292-3" aria-hidden="true" tabindex="-1"></a>lsd</span></code></pre></div>
<pre><code>## $statistics
##    MSerror Df    Mean      CV  t.value     LSD
##   11.52778  9 257.125 1.32047 2.262157 5.43101
## 
## $parameters
##         test p.ajusted name.t ntr alpha
##   Fisher-LSD      none   rate   4  0.05
## 
## $means
##    corm_number      std r      LCL      UCL Min Max    Q25   Q50    Q75
## 1X      249.25 4.272002 4 245.4097 253.0903 246 255 246.00 248.0 251.25
## 2X      284.75 2.629956 4 280.9097 288.5903 281 287 284.00 285.5 286.25
## 3X      254.25 2.872281 4 250.4097 258.0903 251 258 253.25 254.0 255.00
## 4X      240.25 5.439056 4 236.4097 244.0903 233 246 238.25 241.0 243.00
## 
## $comparison
## NULL
## 
## $groups
##    corm_number groups
## 2X      284.75      a
## 3X      254.25      b
## 1X      249.25      b
## 4X      240.25      c
## 
## attr(,&quot;class&quot;)
## [1] &quot;group&quot;</code></pre>
<p>Lets unpack this piece by peace. The output from the LSD test is in a list of tables.</p>
<div id="statistics-table" class="section level3" number="8.3.1">
<h3><span class="header-section-number">8.3.1</span> Statistics Table</h3>
<p>Let’s start with the <code>$statistics</code> table. This explains how our LSD was calculated:</p>
<ul>
<li><p>MSerror is the error or residual mean square. It should match that value from the ANOVA table above. Recall that MSerror is an estimate the variance within treatments – that is, the variation among plots unexplained by our model. Therefore, its square root is the standard deviation of the observations within each treatment.</p></li>
<li><p>DF is the degrees of freedom, which is used to calculate our t.value</p></li>
<li><p>Mean is just that – the overall mean of the trial.</p></li>
<li><p>CV is the coefficient of variance. By dividing the standard deviation by the mean, and multiplying by 100, we arrive at this value. Recall from Unit 6 that the CV is a measure of the quality control of the trial: how consistent were our experimental units?</p></li>
<li><p>The t-value is based on the degrees of freedom and <span class="math inline">\(\alpha\)</span>, the desired p-value (often 0.05) to be used to to test significance.</p></li>
<li><p>LSD is the product of the t-value and the standard error of the difference, which can be derived from MSerror and the number of replicates.</p></li>
</ul>
<p>This is a lot to absorb, I realize. The most important two statistics for you to understand from this table are the CV and LSD. The other numbers are intermediate values, although if you list the LSD in a report you should also report the degrees of freedom used to calculate the t-value.</p>
</div>
<div id="means-table" class="section level3" number="8.3.2">
<h3><span class="header-section-number">8.3.2</span> Means Table</h3>
<p>The <code>$means</code> table explains the distribution of observations within a treatment level around their sample mean. Most of these concepts we discussed in Unit 4. The statistics are:</p>
<ul>
<li>corm_number: These are our sample means for each level of treatment</li>
<li>std: the standard error of the sample mean. This is unique to the sample mean for each treatment.</li>
<li>r: the number of replicates per treatment level</li>
<li>LCL and UCL: the lower confidence limit and upper confidence limit for each mean. These are calculated just as we did in Unit 4.</li>
</ul>
<p>The remainder of the statistics show the minimum and maximum values, and the quartiles.</p>
</div>
<div id="groups-table" class="section level3" number="8.3.3">
<h3><span class="header-section-number">8.3.3</span> Groups Table</h3>
<p>Often, the $groups table is the most interesting, for it tests the differences among levels of a treatment. The treatment means among levels are ranked from greatest to least.</p>
<ul>
<li>corm_number: again, the sample mean for each level of treatment</li>
<li>group: this groups treatments that are statistically equivalent. Any means followed by the same letter are considered equal. In the table above, the mean corm numbers associated with the 1X and 3X rates of fertilizer are considered equal. Any means not followed by the same letter are consisdered statistically different at the p-level chosen to test those differences. The 2X rate produced significantly greater corm numbers than the other fertilizer rates. The 4X rate produced statistically lesser corm numbers than the other fertilizer rates.</li>
</ul>
</div>
</div>
<div id="comparisonwise-versus-experimentwise-error" class="section level2" number="8.4">
<h2><span class="header-section-number">8.4</span> Comparisonwise versus Experimentwise Error</h2>
<p>It is important LSD tests not be used indiscriminatly to separate means. The problem is that each time we test the difference between two means, we have a 5% chance of committing a type I error. This is the probability of committing a Type I error in <em>that</em> comparison. This is known as the <em>comparisonwise error rate</em>.</p>
<p>The probability of committing a type I error across all the comparisons is known as the <em>experimentwise error rate</em>. If we only conduct one comparison, our experimentwise error rate is equal to the comparisonwise error rate. But if we conduct two comparisons, our experimentwise error rate is equal to the probability that comparisonwise errors will not occur in both comparison.</p>
<p>We can demonstrate this with some simple calculations. Before we start our comparisons, there is a 100% chance we have not committed an experimentwise Type I error. We cold express this as:</p>
<p><span class="math display">\[ \text{Experimentwise Error} = 1 \]</span></p>
<p>If we once comparison, there is 5% probability of a comparisonwise Type I error – and a 95% chance the error will not occur. We can express this as</p>
<p><span class="math display">\[ \text{Experimentwise Error} = 1 - 0.95 = 0.05 \]</span></p>
<p>If we have two comparisons, there is a 95% probability a comparisonwise Type I error won’t occur in the first comparison – and a 95% probability it won’t occur in the second comparison. But the probability it doesn’t occur in both comparisons is 0.95 * 0.95:</p>
<p><span class="math display">\[ \text{Experimentwise Error} = 1 - 0.95 \times 0.95 = 1 - 0.9025 = 0.0975  \]</span></p>
<p>Now the Experimentwise error rate is 0.0975, or 9.75%.</p>
<p>What about three comparisons?</p>
<p><span class="math display">\[ \text{Experimentwise Error} = 1 - 0.95 \times 0.95 \times 0.95 = 1 - 0.8573 = 0.1427 \]</span></p>
<p>The Experimentwise error rate is now 0.1427, or 14.27%.</p>
<p>Finally, what if we had 10 comparisons?</p>
<p><span class="math display">\[ \text{Experimentwise Error} = 1 - 0.95^{10} = 1 - 0.5987 = 0.4013\]</span>
Our experimentwise error rate is now about 0.40, or 40%.</p>
<p>As the number of our comparisons increases, so does the probability of an experimentwise error. How can we avoid this? The first method, mentioned above, is to not use the LSD test unless the ANOVA shows a significant treatment effect. We call this approach the <em>F-protected LSD test</em>.</p>
<p>The second approach is to use a multiple range test that increases its minimum significant difference for comparing treatments as the number of treatments increases.</p>
</div>
<div id="tukeys-honest-significant-difference" class="section level2" number="8.5">
<h2><span class="header-section-number">8.5</span> Tukey’s Honest Significant Difference</h2>
<p>If we are going to be comparing many treatments, it is better to use a minimum significant difference, like Tukey’s Honest Significant Different (HSD) Test. Tukey’s HSD is calculated very similarly to the least significant difference:</p>
<p><span class="math display">\[ \text{Tukey&#39;s HSD} = Q_{\alpha, df, k} \text{ } \cdot SED \]</span></p>
<p>The difference in Tukey’s HSD is that we use Q, the “studentized range distribution” (you can just call it Q in this course) in place of the t-distribution. Q differs from t in that its value is determined not only by <span class="math inline">\(\alpha\)</span>, the desired probability of a Type I error, and <span class="math inline">\(df\)</span>, the degrees of freedom associated with the error mean square, but also <span class="math inline">\(k\)</span>, the number of treatments.</p>
<p>The plot below includes four “Q” distribution curves, associated with 3, 5, 6, and 10 treatments. Observe how the distributions shift to the right as the number of treatments increases. This means that the minimum difference for significance also increases with the number of treatments.</p>
<div class="sourceCode" id="cb294"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb294-1"><a href="means-separation-and-data-presentation.html#cb294-1" aria-hidden="true" tabindex="-1"></a>tukey_list <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb294-2"><a href="means-separation-and-data-presentation.html#cb294-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(k <span class="cf">in</span> <span class="fu">c</span>(<span class="dv">3</span><span class="sc">:</span><span class="dv">10</span>)){</span>
<span id="cb294-3"><a href="means-separation-and-data-presentation.html#cb294-3" aria-hidden="true" tabindex="-1"></a>  x_ptukey <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">8</span>, <span class="at">by =</span> <span class="fl">0.005</span>)                       <span class="co"># Specify x-values for ptukey function</span></span>
<span id="cb294-4"><a href="means-separation-and-data-presentation.html#cb294-4" aria-hidden="true" tabindex="-1"></a>  y_ptukey <span class="ot">&lt;-</span> <span class="fu">ptukey</span>(x_ptukey, <span class="at">nmeans =</span> k, <span class="at">df =</span> <span class="dv">3</span><span class="sc">*</span>k)     <span class="co"># Apply ptukey function</span></span>
<span id="cb294-5"><a href="means-separation-and-data-presentation.html#cb294-5" aria-hidden="true" tabindex="-1"></a>  tukey_df <span class="ot">=</span> <span class="fu">data.frame</span>(x_ptukey) <span class="sc">%&gt;%</span></span>
<span id="cb294-6"><a href="means-separation-and-data-presentation.html#cb294-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cbind</span>(y_ptukey) <span class="sc">%&gt;%</span></span>
<span id="cb294-7"><a href="means-separation-and-data-presentation.html#cb294-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">change =</span> y_ptukey <span class="sc">-</span> <span class="fu">lag</span>(y_ptukey)) <span class="sc">%&gt;%</span></span>
<span id="cb294-8"><a href="means-separation-and-data-presentation.html#cb294-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">midpoint =</span> (x_ptukey <span class="sc">+</span> <span class="fu">lag</span>(x_ptukey))<span class="sc">/</span><span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb294-9"><a href="means-separation-and-data-presentation.html#cb294-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">no_trt =</span> k) <span class="sc">%&gt;%</span></span>
<span id="cb294-10"><a href="means-separation-and-data-presentation.html#cb294-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb294-11"><a href="means-separation-and-data-presentation.html#cb294-11" aria-hidden="true" tabindex="-1"></a>  i<span class="ot">=</span>k<span class="dv">-2</span></span>
<span id="cb294-12"><a href="means-separation-and-data-presentation.html#cb294-12" aria-hidden="true" tabindex="-1"></a>  tukey_list[[i]] <span class="ot">=</span> tukey_df</span>
<span id="cb294-13"><a href="means-separation-and-data-presentation.html#cb294-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb294-14"><a href="means-separation-and-data-presentation.html#cb294-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb294-15"><a href="means-separation-and-data-presentation.html#cb294-15" aria-hidden="true" tabindex="-1"></a>tukey_df_all <span class="ot">=</span> <span class="fu">bind_rows</span>(tukey_list)</span>
<span id="cb294-16"><a href="means-separation-and-data-presentation.html#cb294-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb294-17"><a href="means-separation-and-data-presentation.html#cb294-17" aria-hidden="true" tabindex="-1"></a>tukey_df_all <span class="sc">%&gt;%</span></span>
<span id="cb294-18"><a href="means-separation-and-data-presentation.html#cb294-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(no_trt <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">5</span>,<span class="dv">8</span>,<span class="dv">10</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb294-19"><a href="means-separation-and-data-presentation.html#cb294-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">no_trt =</span> <span class="fu">as.factor</span>(no_trt)) <span class="sc">%&gt;%</span></span>
<span id="cb294-20"><a href="means-separation-and-data-presentation.html#cb294-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>midpoint, <span class="at">y=</span>change, <span class="at">group=</span>no_trt)) <span class="sc">+</span></span>
<span id="cb294-21"><a href="means-separation-and-data-presentation.html#cb294-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color=</span>no_trt))</span></code></pre></div>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_point).</code></pre>
<p><img src="08-Means-Separation-and-Presentation_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<p>The Tukey test output below is very similar to the LSD test output. The “$statistics” section is identical to that of the LSD output, except it now reports the minimum significant difference instead of the least significant difference. In the “$parameters” section, the “Studentized Range” value (Q) is given in place of the t-value. The “$groups” section can be interpreted the same for the minimum significant difference as for the least significant difference.</p>
<div class="sourceCode" id="cb296"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb296-1"><a href="means-separation-and-data-presentation.html#cb296-1" aria-hidden="true" tabindex="-1"></a>tukey <span class="ot">=</span> <span class="fu">HSD.test</span>(saffron_model, <span class="st">&quot;rate&quot;</span>, <span class="at">group=</span><span class="cn">TRUE</span>)</span>
<span id="cb296-2"><a href="means-separation-and-data-presentation.html#cb296-2" aria-hidden="true" tabindex="-1"></a>tukey</span></code></pre></div>
<pre><code>## $statistics
##    MSerror Df    Mean      CV      MSD
##   11.52778  9 257.125 1.32047 7.494846
## 
## $parameters
##    test name.t ntr StudentizedRange alpha
##   Tukey   rate   4          4.41489  0.05
## 
## $means
##    corm_number      std r Min Max    Q25   Q50    Q75
## 1X      249.25 4.272002 4 246 255 246.00 248.0 251.25
## 2X      284.75 2.629956 4 281 287 284.00 285.5 286.25
## 3X      254.25 2.872281 4 251 258 253.25 254.0 255.00
## 4X      240.25 5.439056 4 233 246 238.25 241.0 243.00
## 
## $comparison
## NULL
## 
## $groups
##    corm_number groups
## 2X      284.75      a
## 3X      254.25      b
## 1X      249.25      b
## 4X      240.25      c
## 
## attr(,&quot;class&quot;)
## [1] &quot;group&quot;</code></pre>
<p>Unlike the LSD test, the Tukey test does not need to be “protected” by first examing whether the Analysis of Variance treatment effect is significant. That said, the Tukey test is unlikely to indicate a significant difference between treatments without the ANOVA treatment effect being significant as well.</p>
</div>
<div id="linear-contrast" class="section level2" number="8.6">
<h2><span class="header-section-number">8.6</span> Linear Contrast</h2>
<p>The last type of means comparison we will learn in this lesson is the linear contrast. Unlike the LSD and Tukey tests, linear contrasts may be used to separate two groups of treatments. While a lot of math may be introduced to the curious, in a linear contrast the statistician defines two groups of treatments through the use of <em>coefficients</em>; R then calculates their means and standard errors, and compares them using a t-test. There are multiple ways we can use a linear contrast, and our saffron dataset is a great way to introduce them.</p>
<div id="coefficients" class="section level3" number="8.6.1">
<h3><span class="header-section-number">8.6.1</span> Coefficients</h3>
<p>Recall how a t-test between two treatments works. We calculate <span class="math inline">\(t\)</span> as:</p>
<p><span class="math display">\[ t = \frac{(\bar{x}_1-\bar{x}_2) - (\mu_1-\mu_2)}{SED}\]</span>
Where <span class="math inline">\(\bar{x}_1 -\bar{x}_2\)</span> is the difference between sample means, <span class="math inline">\((\mu_1-\mu_2)\)</span> is the hypothesized difference between treatments (usually zero), and SED is the standard error of the difference.</p>
<p>For most situations, we could simplify the above equation to:</p>
<p><span class="math display">\[ t = \frac{\bar{x}_1-\bar{x}_2}{SED}\]</span></p>
<p>What about if we are comparing more than two treatments? For our saffron example, what if we wanted to calculate the difference between the mean of the two lower rates (1X and 2X) and the mean of the two higher rates (3X and 4X)? Lets call this difference <span class="math inline">\(L\)</span>. We would calculate this difference as:</p>
<p><span class="math display">\[ L = \frac{\bar{x}_{1X} + \bar{x}_{2X}}{2} - \frac{\bar{x}_{3X} + \bar{x}_{4X}}{2} \]</span></p>
<p>All we are doing above is 1) calculating the means for the two groups and 2) subtracting the mean of the 3X and 4X rates from the mean of the 1X and 2X rates. Now let’s express this same formula a little differently. What we are doing in the above equation is multiplying each number by <span class="math inline">\(1/2\)</span>:</p>
<p><span class="math display">\[ L = \frac{1}{2}(\bar{x}_{1X} + \bar{x}_{2X}) - \frac{1}{2}(\bar{x}_{3X} + \bar{x}_{4X})\]</span></p>
<p>In addition (bear with me!), when we subtract the mean of treatments 3X and 4X, it is the equivalent of adding the negative value of their mean to the mean of treatments 1X and 2X:</p>
<p><span class="math display">\[ L = \frac{1}{2}(\bar{x}_{1X} + \bar{x}_{2X}) + (- \frac{1}{2}(\bar{x}_{3X} + \bar{x}_{4X}))\]</span></p>
<p>Finally, we can arrange the equation above as:</p>
<p><span class="math display">\[ L = \frac{1}{2}\bar{x}_{1X} + \frac{1}{2}\bar{x}_{2X} - \frac{1}{2}\bar{x}_{3X} - \frac{1}{2}\bar{x}_{4X} + \]</span></p>
<p>The reason for this tortuous algebra flashback is to show you where the contrast coefficients come from. Each of the <span class="math inline">\(\frac{1}{2}\)</span>s in the equation above is a contrast coefficient.</p>
<p>Let’s demonstrate this with our saffron data. Our saffron treatment means are:</p>
<div class="sourceCode" id="cb298"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb298-1"><a href="means-separation-and-data-presentation.html#cb298-1" aria-hidden="true" tabindex="-1"></a>saffron_means <span class="ot">=</span> saffron <span class="sc">%&gt;%</span></span>
<span id="cb298-2"><a href="means-separation-and-data-presentation.html#cb298-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(rate) <span class="sc">%&gt;%</span></span>
<span id="cb298-3"><a href="means-separation-and-data-presentation.html#cb298-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">corm_number =</span> <span class="fu">mean</span>(corm_number)) <span class="sc">%&gt;%</span></span>
<span id="cb298-4"><a href="means-separation-and-data-presentation.html#cb298-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb298-5"><a href="means-separation-and-data-presentation.html#cb298-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb298-6"><a href="means-separation-and-data-presentation.html#cb298-6" aria-hidden="true" tabindex="-1"></a>saffron_means</span></code></pre></div>
<pre><code>## # A tibble: 4 x 2
##   rate  corm_number
##   &lt;chr&gt;       &lt;dbl&gt;
## 1 1X           249.
## 2 2X           285.
## 3 3X           254.
## 4 4X           240.</code></pre>
<p>Now let’s add in a column with our coefficients:</p>
<div class="sourceCode" id="cb300"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb300-1"><a href="means-separation-and-data-presentation.html#cb300-1" aria-hidden="true" tabindex="-1"></a>saffron_coefficients <span class="ot">=</span> saffron_means <span class="sc">%&gt;%</span></span>
<span id="cb300-2"><a href="means-separation-and-data-presentation.html#cb300-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">coefficient =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>, <span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>, <span class="sc">-</span><span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>, <span class="sc">-</span><span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>))</span>
<span id="cb300-3"><a href="means-separation-and-data-presentation.html#cb300-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb300-4"><a href="means-separation-and-data-presentation.html#cb300-4" aria-hidden="true" tabindex="-1"></a>saffron_coefficients</span></code></pre></div>
<pre><code>## # A tibble: 4 x 3
##   rate  corm_number coefficient
##   &lt;chr&gt;       &lt;dbl&gt;       &lt;dbl&gt;
## 1 1X           249.         0.5
## 2 2X           285.         0.5
## 3 3X           254.        -0.5
## 4 4X           240.        -0.5</code></pre>
<p>We see that R has converted these to decimals. We then create a new column, “mean_x_coefficient,” that is the product of the original mean and the coefficient. We see these products are approximately half the value of the original sample mean (some may be less than half because of rounding errors).</p>
<div class="sourceCode" id="cb302"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb302-1"><a href="means-separation-and-data-presentation.html#cb302-1" aria-hidden="true" tabindex="-1"></a>saffron_products <span class="ot">=</span> saffron_coefficients <span class="sc">%&gt;%</span></span>
<span id="cb302-2"><a href="means-separation-and-data-presentation.html#cb302-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mean_x_coefficient =</span>  corm_number <span class="sc">*</span> coefficient)</span>
<span id="cb302-3"><a href="means-separation-and-data-presentation.html#cb302-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb302-4"><a href="means-separation-and-data-presentation.html#cb302-4" aria-hidden="true" tabindex="-1"></a>saffron_products</span></code></pre></div>
<pre><code>## # A tibble: 4 x 4
##   rate  corm_number coefficient mean_x_coefficient
##   &lt;chr&gt;       &lt;dbl&gt;       &lt;dbl&gt;              &lt;dbl&gt;
## 1 1X           249.         0.5               125.
## 2 2X           285.         0.5               142.
## 3 3X           254.        -0.5              -127.
## 4 4X           240.        -0.5              -120.</code></pre>
<p>Finally, we can sum the mean_x_coefficient column to get the total difference among treatments.</p>
<div class="sourceCode" id="cb304"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb304-1"><a href="means-separation-and-data-presentation.html#cb304-1" aria-hidden="true" tabindex="-1"></a>total_difference <span class="ot">=</span> <span class="fu">sum</span>(saffron_products<span class="sc">$</span>mean_x_coefficient)</span>
<span id="cb304-2"><a href="means-separation-and-data-presentation.html#cb304-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb304-3"><a href="means-separation-and-data-presentation.html#cb304-3" aria-hidden="true" tabindex="-1"></a><span class="fu">paste</span>(<span class="st">&quot;total difference = &quot;</span>, total_difference, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;total difference = 19.75&quot;</code></pre>
<p>One critical rule about requirements is their sum must always equal zero. Otherwise you are not completely identifying two groups to compare. Using coefficients that do not sum to zero can also suggest you are weighting one group unfairly compared to another.</p>
<p>###Contrast Calculations</p>
<p>To calculate <span class="math inline">\(t\)</span>, of course, we must divide <span class="math inline">\(L\)</span> by the standard error of the difference:</p>
<p><span class="math display">\[ t = \frac{L}{SED}\]</span></p>
<p>So how is the standard error of the difference calculated?</p>
<p>Well, we know that the error mean square from our ANOVA is equal to the mean variance within treatments. And we know that if we take the square root of a variance, we get the standard deviation. And if we divide the standard deviation by the number of observations in each sample mean, we get the standard error.</p>
<p>In a paired t-test, where we are simply calculating a standard error for one set of numbers, the differences between each pair, the standard error of the difference is calculated the same as the standard error:</p>
<p><span class="math display">\[ SED = SE = \frac{s}{\sqrt{n}} \]</span></p>
<p>Where <span class="math inline">\(s\)</span> is the standard deviation and <span class="math inline">\(n\)</span> is the number of observations (reps) per treatment. This formula is equal to that below, where we use the variance, <span class="math inline">\(s^2\)</span>, in place of the standard deviation.</p>
<p><span class="math display">\[ SED = \sqrt{\frac{s^2}{n}} \]</span></p>
<p>When we work with two-treatment trials where the treatments are not paired or blocked, we account for the variance and number replicates individually. For treatment levels “1” and 2", the standard errror of the difference becomes
<span class="math display">\[ SED = \sqrt{\frac{s_1^2}{n_1} + \frac{s_2^2}{n_2}} \]</span></p>
<p>In most trials, however, we assume that the variances and number of replications are equal among treatments. So we can also express the above trial as:</p>
<p><span class="math display">\[ SED = \sqrt\frac{2 \cdot s^2}{n}\]</span></p>
<p>Recall that in the analsis of variance for a multiple treatment trial, the error mean squares <em>is</em> the mean variance within treatments. So the equation above is equivalent to:</p>
<p><span class="math display">\[ SED = \sqrt\frac{2\cdot EMS}{n}\]</span></p>
<p>In a linear contrast, however, our standard error of the difference must be scaled according to the coefficients we use, since we are no longer comparing two treatments, but multiple. So our equation becomes:</p>
<p><span class="math display">\[ SED = c \cdot \sqrt{\frac{EMS}{n}} \]</span></p>
<p>Where <span class="math inline">\(c\)</span> is the square root of the sum of the squared constants is the For our example above, this sum, <span class="math inline">\(c\)</span>, would be:</p>
<p><span class="math display">\[ c = s\sqrt{\sum (\frac{1}{2})^2 + (\frac{1}{2})^2 + (- \frac{1}{2})^2 + (- \frac{1}{2})^2} \]</span></p>
<p>Which is equal to:</p>
<p><span class="math display">\[ c = \sqrt{\frac{1}{4} + \frac{1}{4} +\frac{1}{4} +\frac{1}{4}} = \sqrt{1} = 1\]</span></p>
<p>We can now calculate the standard error of the difference for our contrast. First, lets go back to our analysis of variance.</p>
<div class="sourceCode" id="cb306"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb306-1"><a href="means-separation-and-data-presentation.html#cb306-1" aria-hidden="true" tabindex="-1"></a>saffron_model <span class="ot">=</span> <span class="fu">aov</span>(corm_number <span class="sc">~</span> rate, saffron)</span>
<span id="cb306-2"><a href="means-separation-and-data-presentation.html#cb306-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(saffron_model)</span></code></pre></div>
<pre><code>##             Df Sum Sq Mean Sq F value   Pr(&gt;F)    
## rate         3   4473  1490.9   94.66 1.28e-08 ***
## Residuals   12    189    15.7                     
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>We can see from the ANOVA output thtat our error mean square is 15.75. We know from the design itself that we had 4 replicates</p>
<p>So our final standard error of the difference for our contrast is:</p>
<p><span class="math display">\[ SED = 1 \cdot \sqrt{\frac{15.75}{4}} = 1 \cdot \sqrt{3.9375} = 1.984 \]</span></p>
<p>The t-value for our test, would then be:</p>
<p><span class="math display">\[ t =  \frac{L}{SED} = \frac{19.75}{1.984} = 9.954 \]</span></p>
<p>The probability of a t-value of 10, given the 12 degrees of freedom associated with the error sum of squares, would be:</p>
<div class="sourceCode" id="cb308"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb308-1"><a href="means-separation-and-data-presentation.html#cb308-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pt</span>(<span class="fl">9.954</span>,<span class="dv">12</span>, <span class="at">lower.tail =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## [1] 1.882084e-07</code></pre>
</div>
<div id="linear-contrasts-with-r" class="section level3" number="8.6.2">
<h3><span class="header-section-number">8.6.2</span> Linear Contrasts with R</h3>
<p>We can automate linear contrast testing, of course, using R and the <em>glht()</em> function of the <em>multcomp</em> package. First, we need to define a matrix (table) of contrast coefficients for R. To get the coefficients in the correct order, lets double check the order in which the rate levels are listed in R. It is important to make sure our treatment is classified as a factor. We can do this using the <em>as.factor()</em> function</p>
<div class="sourceCode" id="cb310"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb310-1"><a href="means-separation-and-data-presentation.html#cb310-1" aria-hidden="true" tabindex="-1"></a>saffron<span class="sc">$</span>rate <span class="ot">=</span> <span class="fu">as.factor</span>(saffron<span class="sc">$</span>rate)</span></code></pre></div>
<p>We can then list the order of the levels in treatment rate:</p>
<div class="sourceCode" id="cb311"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb311-1"><a href="means-separation-and-data-presentation.html#cb311-1" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(saffron<span class="sc">$</span>rate)</span></code></pre></div>
<pre><code>## [1] &quot;1X&quot; &quot;2X&quot; &quot;3X&quot; &quot;4X&quot;</code></pre>
<p>We can see they follow the order 1X, 2X, 3X, 4X. We will therefore form a matrix, K, with the appropriate coefficients.</p>
<div class="sourceCode" id="cb313"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb313-1"><a href="means-separation-and-data-presentation.html#cb313-1" aria-hidden="true" tabindex="-1"></a>K <span class="ot">=</span> <span class="fu">matrix</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>, <span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>, <span class="sc">-</span><span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>, <span class="sc">-</span><span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>),<span class="dv">1</span>)</span>
<span id="cb313-2"><a href="means-separation-and-data-presentation.html#cb313-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb313-3"><a href="means-separation-and-data-presentation.html#cb313-3" aria-hidden="true" tabindex="-1"></a>K</span></code></pre></div>
<pre><code>##      [,1] [,2] [,3] [,4]
## [1,]  0.5  0.5 -0.5 -0.5</code></pre>
<p>We are now ready to run our contrast. First, we need to slightly alter our ANOVA model by adding a zero (0) between the tilde (~) and the treatment name. This is one of of those one-off ideosyncrasities of R.</p>
<div class="sourceCode" id="cb315"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb315-1"><a href="means-separation-and-data-presentation.html#cb315-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(multcomp)</span></code></pre></div>
<pre><code>## Loading required package: mvtnorm</code></pre>
<pre><code>## Loading required package: survival</code></pre>
<pre><code>## Loading required package: TH.data</code></pre>
<pre><code>## Loading required package: MASS</code></pre>
<pre><code>## 
## Attaching package: &#39;MASS&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:dplyr&#39;:
## 
##     select</code></pre>
<pre><code>## 
## Attaching package: &#39;TH.data&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:MASS&#39;:
## 
##     geyser</code></pre>
<div class="sourceCode" id="cb324"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb324-1"><a href="means-separation-and-data-presentation.html#cb324-1" aria-hidden="true" tabindex="-1"></a>saffron_model_for_contrast <span class="ot">=</span> <span class="fu">aov</span>(corm_number <span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> rate, saffron)</span>
<span id="cb324-2"><a href="means-separation-and-data-presentation.html#cb324-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb324-3"><a href="means-separation-and-data-presentation.html#cb324-3" aria-hidden="true" tabindex="-1"></a>low_vs_high <span class="ot">=</span> <span class="fu">glht</span>(saffron_model_for_contrast, <span class="at">linfct=</span>K)</span>
<span id="cb324-4"><a href="means-separation-and-data-presentation.html#cb324-4" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(low_vs_high)</span></code></pre></div>
<pre><code>## 
##   Simultaneous Tests for General Linear Hypotheses
## 
## Fit: aov(formula = corm_number ~ 0 + rate, data = saffron)
## 
## Linear Hypotheses:
##        Estimate Std. Error t value Pr(&gt;|t|)    
## 1 == 0   19.750      1.984   9.953 3.77e-07 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## (Adjusted p values reported -- single-step method)</code></pre>
<p>And there we have it, our contrast in seconds. Now let’s interpret this. Estimate is the difference between groups. Since we subtracted the mean of the 3x and 4X rates from the mean of the 1X and 2X rates, the positive estimate value indicates the lower rates produced greater corm numbers than the higher rates.</p>
<p>“Std. Error” is the standard error of the difference, as we calculated above. The t-value is equal to the estimate divided by the standard error of the difference. Finally, “Pr(&gt;|t|)” is the probability of observing the t-value by chance. In this case, it is <span class="math inline">\(3.77 \times 10^{-7}\)</span>, very close to zero. We conclude the lower two rates, as a group, produce greater corms than the upper two rates as a group.</p>
<p>We can quickly as other questions of our data. For example, do the middle two rates (2X and 3X) produce a greater corm number than the lowest (1X and highest (4X) rates? Again, let’s examine the order of our rate levels.</p>
<div class="sourceCode" id="cb326"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb326-1"><a href="means-separation-and-data-presentation.html#cb326-1" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(saffron<span class="sc">$</span>rate)</span></code></pre></div>
<pre><code>## [1] &quot;1X&quot; &quot;2X&quot; &quot;3X&quot; &quot;4X&quot;</code></pre>
<p>In order to subtract the mean corm number of rates 1X and 4X from the mean corm number of rates 2X and 3X, we will need to calculate the difference as:</p>
<p><span class="math display">\[ L = (-\frac{1}{2})\bar{x}_{1X} + (\frac{1}{2})\bar{x}_{2X} + (\frac{1}{2})\bar{x}_{3X} + (- \frac{1}{2})\bar{x}_{4X} + \]</span></p>
<p>So our contrasts coefficients would be <span class="math inline">\(-\frac{1}{2}\)</span>, <span class="math inline">\(\frac{1}{2}\)</span>, <span class="math inline">\(\frac{1}{2}\)</span>, <span class="math inline">\(-\frac{1}{2}\)</span>. Our contrast maxtrix is then:</p>
<div class="sourceCode" id="cb328"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb328-1"><a href="means-separation-and-data-presentation.html#cb328-1" aria-hidden="true" tabindex="-1"></a>K <span class="ot">=</span> <span class="fu">matrix</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>, <span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>, <span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>, <span class="sc">-</span><span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>),<span class="dv">1</span>)</span>
<span id="cb328-2"><a href="means-separation-and-data-presentation.html#cb328-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb328-3"><a href="means-separation-and-data-presentation.html#cb328-3" aria-hidden="true" tabindex="-1"></a>K</span></code></pre></div>
<pre><code>##      [,1] [,2] [,3] [,4]
## [1,] -0.5  0.5  0.5 -0.5</code></pre>
<p>We are now ready to run our contrast. First, we need to slightly alter our ANOVA model by adding a zero (0) between the tilde (~) and the treatment name. This is one of of those one-off ideosyncrasities of R.</p>
<div class="sourceCode" id="cb330"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb330-1"><a href="means-separation-and-data-presentation.html#cb330-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(multcomp)</span>
<span id="cb330-2"><a href="means-separation-and-data-presentation.html#cb330-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb330-3"><a href="means-separation-and-data-presentation.html#cb330-3" aria-hidden="true" tabindex="-1"></a>saffron_model_for_contrast <span class="ot">=</span> <span class="fu">aov</span>(corm_number <span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> rate, saffron)</span>
<span id="cb330-4"><a href="means-separation-and-data-presentation.html#cb330-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb330-5"><a href="means-separation-and-data-presentation.html#cb330-5" aria-hidden="true" tabindex="-1"></a>low_high_vs_middle <span class="ot">=</span> <span class="fu">glht</span>(saffron_model_for_contrast, <span class="at">linfct=</span>K)</span>
<span id="cb330-6"><a href="means-separation-and-data-presentation.html#cb330-6" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(low_high_vs_middle)</span></code></pre></div>
<pre><code>## 
##   Simultaneous Tests for General Linear Hypotheses
## 
## Fit: aov(formula = corm_number ~ 0 + rate, data = saffron)
## 
## Linear Hypotheses:
##        Estimate Std. Error t value Pr(&gt;|t|)    
## 1 == 0   24.750      1.984   12.47 3.14e-08 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## (Adjusted p values reported -- single-step method)</code></pre>
<p>We see this estimated difference is even greater than the previous contrast. The significant t-value suggests there may be a parabolic (curved) response of corm number to nutrient solution rate. Indeed, if we plot the mean, we can see a parabolic response</p>
<div class="sourceCode" id="cb332"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb332-1"><a href="means-separation-and-data-presentation.html#cb332-1" aria-hidden="true" tabindex="-1"></a>saffron_means <span class="sc">%&gt;%</span></span>
<span id="cb332-2"><a href="means-separation-and-data-presentation.html#cb332-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>rate, <span class="at">y=</span>corm_number)) <span class="sc">+</span></span>
<span id="cb332-3"><a href="means-separation-and-data-presentation.html#cb332-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">color=</span><span class="st">&quot;blue&quot;</span>)</span></code></pre></div>
<p><img src="08-Means-Separation-and-Presentation_files/figure-html/unnamed-chunk-19-1.png" width="672" /></p>
</div>
</div>
<div id="means-presentation" class="section level2" number="8.7">
<h2><span class="header-section-number">8.7</span> Means Presentation</h2>
<p>Something I have always overlooked in teaching this course is also something that is most basic: once we have separated our means, how do we present them to others? There are two ways to present means: in a table, or in a plot. Both have advantages and disadvantages. In a table, you provide raw statistics (treatment means, standard errors, and perhaps, means groupings from an LSD or Tukey test). A reader can use these values to recalculate your statistics, say, if they wanted to separate the means at p=0.10, p=0.1, or p=0.001.</p>
<p>A figure, one the otherhand, allows the reader to quickly grasp treatment differences, or patterns among treatments. In addition, they are colorful and – lets face it – more inviting than a dense table of numbers.</p>
<p>Whichever format we use, however, we need to present our treatment means and some reference for the reader to gauge whether those means are statistically equal or different.</p>
<div id="means-tables" class="section level3" number="8.7.1">
<h3><span class="header-section-number">8.7.1</span> Means Tables</h3>
<p>Tables can be tricky, especially when many types of measures are included. If we are reporting results from a factorial experiment, we may be tempted to list the levels of one factor down rows, and the other factor across columns. I would generally discourage this, however, unless required to fit the table neatly into a publication space. Generally, the long form of means presentation is best.</p>
<p>In the long form of data, cases (individual observations or treatment means) are listed down the rows. Measurements from each cases are listed across rows.</p>
<p>For our saffron data, our table would start like this:</p>
<div class="sourceCode" id="cb333"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb333-1"><a href="means-separation-and-data-presentation.html#cb333-1" aria-hidden="true" tabindex="-1"></a>final_table <span class="ot">=</span> saffron_means</span>
<span id="cb333-2"><a href="means-separation-and-data-presentation.html#cb333-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb333-3"><a href="means-separation-and-data-presentation.html#cb333-3" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(final_table)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">rate</th>
<th align="right">corm_number</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1X</td>
<td align="right">249.25</td>
</tr>
<tr class="even">
<td align="left">2X</td>
<td align="right">284.75</td>
</tr>
<tr class="odd">
<td align="left">3X</td>
<td align="right">254.25</td>
</tr>
<tr class="even">
<td align="left">4X</td>
<td align="right">240.25</td>
</tr>
</tbody>
</table>
<p>To this table, we may wish to add the standard error of the difference. Remember, the standard error of the difference for an LSD or Tukey test is equal to:</p>
<p><span class="math display">\[ SED = \sqrt{\frac{(2 \cdot EMS)}{n}} \]</span></p>
<p>So for our saffron trial, where the error mean square is 15.75 (we can get this from either the ANOVA table or the LSD output)and the number of replications is 4, the standard error of the difference is:</p>
<p><span class="math display">\[ SED = \sqrt{\frac{(2 \cdot 15.75)}{4}} = \sqrt{7.875} = 2.80 \]</span></p>
<p>We can add this below the means</p>
<div class="sourceCode" id="cb334"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb334-1"><a href="means-separation-and-data-presentation.html#cb334-1" aria-hidden="true" tabindex="-1"></a>final_table <span class="ot">=</span> saffron_means <span class="sc">%&gt;%</span></span>
<span id="cb334-2"><a href="means-separation-and-data-presentation.html#cb334-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">rate =</span> <span class="fu">as.character</span>(rate)) <span class="sc">%&gt;%</span></span>
<span id="cb334-3"><a href="means-separation-and-data-presentation.html#cb334-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(<span class="fu">c</span>(<span class="st">&quot;SED&quot;</span>, <span class="fl">2.80</span>))</span>
<span id="cb334-4"><a href="means-separation-and-data-presentation.html#cb334-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb334-5"><a href="means-separation-and-data-presentation.html#cb334-5" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(final_table)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">rate</th>
<th align="left">corm_number</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1X</td>
<td align="left">249.25</td>
</tr>
<tr class="even">
<td align="left">2X</td>
<td align="left">284.75</td>
</tr>
<tr class="odd">
<td align="left">3X</td>
<td align="left">254.25</td>
</tr>
<tr class="even">
<td align="left">4X</td>
<td align="left">240.25</td>
</tr>
<tr class="odd">
<td align="left">SED</td>
<td align="left">2.8</td>
</tr>
</tbody>
</table>
<p>It is also important to indicate the number of replications of the treatments. We can add another row to the table with N.</p>
<div class="sourceCode" id="cb335"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb335-1"><a href="means-separation-and-data-presentation.html#cb335-1" aria-hidden="true" tabindex="-1"></a>final_table <span class="ot">=</span> saffron_means <span class="sc">%&gt;%</span></span>
<span id="cb335-2"><a href="means-separation-and-data-presentation.html#cb335-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">rate =</span> <span class="fu">as.character</span>(rate)) <span class="sc">%&gt;%</span></span>
<span id="cb335-3"><a href="means-separation-and-data-presentation.html#cb335-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(<span class="fu">c</span>(<span class="st">&quot;SED&quot;</span>, <span class="fl">2.80</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb335-4"><a href="means-separation-and-data-presentation.html#cb335-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(<span class="fu">c</span>(<span class="st">&quot;N&quot;</span>, <span class="dv">4</span>))</span>
<span id="cb335-5"><a href="means-separation-and-data-presentation.html#cb335-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb335-6"><a href="means-separation-and-data-presentation.html#cb335-6" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(final_table)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">rate</th>
<th align="left">corm_number</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1X</td>
<td align="left">249.25</td>
</tr>
<tr class="even">
<td align="left">2X</td>
<td align="left">284.75</td>
</tr>
<tr class="odd">
<td align="left">3X</td>
<td align="left">254.25</td>
</tr>
<tr class="even">
<td align="left">4X</td>
<td align="left">240.25</td>
</tr>
<tr class="odd">
<td align="left">SED</td>
<td align="left">2.8</td>
</tr>
<tr class="even">
<td align="left">N</td>
<td align="left">4</td>
</tr>
</tbody>
</table>
<p>We should add the LSD, to make it easy for readers to compare treatments. We will want also, to include the <span class="math inline">\(\alpha\)</span> which was used to calcualate the LSD. That way, the reader will know whether the risk of a Type I error – that the LSD will separate treatments that are not truely different – is 5% or some other probability.</p>
<div class="sourceCode" id="cb336"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb336-1"><a href="means-separation-and-data-presentation.html#cb336-1" aria-hidden="true" tabindex="-1"></a>final_table <span class="ot">=</span> saffron_means <span class="sc">%&gt;%</span></span>
<span id="cb336-2"><a href="means-separation-and-data-presentation.html#cb336-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">rate =</span> <span class="fu">as.character</span>(rate)) <span class="sc">%&gt;%</span></span>
<span id="cb336-3"><a href="means-separation-and-data-presentation.html#cb336-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(<span class="fu">c</span>(<span class="st">&quot;SED&quot;</span>, <span class="fl">2.80</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb336-4"><a href="means-separation-and-data-presentation.html#cb336-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(<span class="fu">c</span>(<span class="st">&quot;N&quot;</span>, <span class="dv">4</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb336-5"><a href="means-separation-and-data-presentation.html#cb336-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(<span class="fu">c</span>(<span class="st">&quot;LSD (0.05)&quot;</span>, <span class="fl">6.11</span>)) </span>
<span id="cb336-6"><a href="means-separation-and-data-presentation.html#cb336-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb336-7"><a href="means-separation-and-data-presentation.html#cb336-7" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(final_table)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">rate</th>
<th align="left">corm_number</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1X</td>
<td align="left">249.25</td>
</tr>
<tr class="even">
<td align="left">2X</td>
<td align="left">284.75</td>
</tr>
<tr class="odd">
<td align="left">3X</td>
<td align="left">254.25</td>
</tr>
<tr class="even">
<td align="left">4X</td>
<td align="left">240.25</td>
</tr>
<tr class="odd">
<td align="left">SED</td>
<td align="left">2.8</td>
</tr>
<tr class="even">
<td align="left">N</td>
<td align="left">4</td>
</tr>
<tr class="odd">
<td align="left">LSD (0.05)</td>
<td align="left">6.11</td>
</tr>
</tbody>
</table>
<p>We might want to include the p-value from the ANOVA table, so the reader knows that the LSD is protected. The p-value for the saffron trial is <span class="math inline">\(1.28 \times 10^{-8}\)</span>. This is an extremely small number. It is acceptable for us to simplify this in the table, indicating that the probability of F was less than 0.001.</p>
<div class="sourceCode" id="cb337"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb337-1"><a href="means-separation-and-data-presentation.html#cb337-1" aria-hidden="true" tabindex="-1"></a>final_table <span class="ot">=</span> saffron_means <span class="sc">%&gt;%</span></span>
<span id="cb337-2"><a href="means-separation-and-data-presentation.html#cb337-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">rate =</span> <span class="fu">as.character</span>(rate)) <span class="sc">%&gt;%</span></span>
<span id="cb337-3"><a href="means-separation-and-data-presentation.html#cb337-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(<span class="fu">c</span>(<span class="st">&quot;SED&quot;</span>, <span class="fl">2.80</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb337-4"><a href="means-separation-and-data-presentation.html#cb337-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(<span class="fu">c</span>(<span class="st">&quot;N&quot;</span>, <span class="dv">4</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb337-5"><a href="means-separation-and-data-presentation.html#cb337-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(<span class="fu">c</span>(<span class="st">&quot;LSD (0.05)&quot;</span>, <span class="fl">6.11</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb337-6"><a href="means-separation-and-data-presentation.html#cb337-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(<span class="fu">c</span>(<span class="st">&quot;Pr&lt;F&quot;</span>, <span class="st">&quot;&lt;0.001&quot;</span>))</span>
<span id="cb337-7"><a href="means-separation-and-data-presentation.html#cb337-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb337-8"><a href="means-separation-and-data-presentation.html#cb337-8" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(final_table)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">rate</th>
<th align="left">corm_number</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1X</td>
<td align="left">249.25</td>
</tr>
<tr class="even">
<td align="left">2X</td>
<td align="left">284.75</td>
</tr>
<tr class="odd">
<td align="left">3X</td>
<td align="left">254.25</td>
</tr>
<tr class="even">
<td align="left">4X</td>
<td align="left">240.25</td>
</tr>
<tr class="odd">
<td align="left">SED</td>
<td align="left">2.8</td>
</tr>
<tr class="even">
<td align="left">N</td>
<td align="left">4</td>
</tr>
<tr class="odd">
<td align="left">LSD (0.05)</td>
<td align="left">6.11</td>
</tr>
<tr class="even">
<td align="left">Pr&lt;F</td>
<td align="left">&lt;0.001</td>
</tr>
</tbody>
</table>
<p>Finally, if imperial (pounds, acres, etc) or metric units were used to measure the response variable, it is important to indicate that in the table. Indicating those in parentheses after the variable name is appropriate.</p>
</div>
<div id="plotting-means" class="section level3" number="8.7.2">
<h3><span class="header-section-number">8.7.2</span> Plotting Means</h3>
<p>When we work with categorical treatments (that is, treatments levels that are defined by words), or even numerical variables treated as categorical variables (as in the saffron example) we should use a <em>bar plot</em>, not a line plot, to visualize the data. An easy way to determine whether a bar plot should be used is this: if you are using an LSD or Tukey’s test to separate your means, you should use a bar plot. A line plot, which suggests treatment levels are numerically related to (higher or lower than) each other should be used to fit regression models, where the analysis defines a continuous relationship between Y and X.</p>
<p>A basic bar plot will has a bar representing each treatment mean. The treatment level is indicated along the x (horizontal) axis. The sample mean is indicated along the y (vertical) axis. The bar height indicates the sample mean.</p>
<div class="sourceCode" id="cb338"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb338-1"><a href="means-separation-and-data-presentation.html#cb338-1" aria-hidden="true" tabindex="-1"></a>saffron_means <span class="sc">%&gt;%</span></span>
<span id="cb338-2"><a href="means-separation-and-data-presentation.html#cb338-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>rate, <span class="at">y=</span>corm_number)) <span class="sc">+</span></span>
<span id="cb338-3"><a href="means-separation-and-data-presentation.html#cb338-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="at">color=</span><span class="st">&quot;black&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;darkolivegreen&quot;</span>) <span class="sc">+</span></span>
<span id="cb338-4"><a href="means-separation-and-data-presentation.html#cb338-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">12</span>),</span>
<span id="cb338-5"><a href="means-separation-and-data-presentation.html#cb338-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>))</span></code></pre></div>
<p><img src="08-Means-Separation-and-Presentation_files/figure-html/unnamed-chunk-25-1.png" width="672" /></p>
<p>This plot, however, does not provide the viewer any sense of whether corm number is significantly different among treatments. For that purpose, we can add error bars.</p>
<div class="sourceCode" id="cb339"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb339-1"><a href="means-separation-and-data-presentation.html#cb339-1" aria-hidden="true" tabindex="-1"></a>saffron_means <span class="sc">%&gt;%</span></span>
<span id="cb339-2"><a href="means-separation-and-data-presentation.html#cb339-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>rate, <span class="at">y=</span>corm_number)) <span class="sc">+</span></span>
<span id="cb339-3"><a href="means-separation-and-data-presentation.html#cb339-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="at">color=</span><span class="st">&quot;black&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;darkolivegreen&quot;</span>) <span class="sc">+</span></span>
<span id="cb339-4"><a href="means-separation-and-data-presentation.html#cb339-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>corm_number<span class="fl">-1.81</span>, <span class="at">ymax=</span>corm_number<span class="fl">+1.81</span>), <span class="at">width=</span><span class="fl">0.4</span>, <span class="at">size=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb339-5"><a href="means-separation-and-data-presentation.html#cb339-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">12</span>),</span>
<span id="cb339-6"><a href="means-separation-and-data-presentation.html#cb339-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>))</span></code></pre></div>
<p><img src="08-Means-Separation-and-Presentation_files/figure-html/unnamed-chunk-26-1.png" width="672" /></p>
<p>These error bars stretch from one standard error of the mean below the sample mean to one standard error of the mean above the sample mean. As a general rule of thumb, the least significant difference is approximately 2 standard errors of the mean. If the error bars from two treatment levels don’t overlap, then the two treatments are likely different. We can see in the plot that the corm number for the 1X and 3X fertilizer rates are very similar.</p>
<p>Alternatively, we could set the error bar height equal to the least significant difference itself.</p>
<div class="sourceCode" id="cb340"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb340-1"><a href="means-separation-and-data-presentation.html#cb340-1" aria-hidden="true" tabindex="-1"></a>saffron_means <span class="sc">%&gt;%</span></span>
<span id="cb340-2"><a href="means-separation-and-data-presentation.html#cb340-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>rate, <span class="at">y=</span>corm_number)) <span class="sc">+</span></span>
<span id="cb340-3"><a href="means-separation-and-data-presentation.html#cb340-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="at">color=</span><span class="st">&quot;black&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;darkolivegreen&quot;</span>) <span class="sc">+</span></span>
<span id="cb340-4"><a href="means-separation-and-data-presentation.html#cb340-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>corm_number<span class="fl">-5.63</span>, <span class="at">ymax=</span>corm_number<span class="fl">+5.63</span>), <span class="at">width=</span><span class="fl">0.4</span>, <span class="at">size=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb340-5"><a href="means-separation-and-data-presentation.html#cb340-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">12</span>),</span>
<span id="cb340-6"><a href="means-separation-and-data-presentation.html#cb340-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>))</span></code></pre></div>
<p><img src="08-Means-Separation-and-Presentation_files/figure-html/unnamed-chunk-27-1.png" width="672" /></p>
<p>This allows the viewer to use the actual LSD to separate treatment means. If the range of the error bar from one treatment does not include the mean of another treatment, then the two treatments are not equal.</p>
<p>Significant differences among treatments can also be indicated by including letter groupings from an LSD or Tukey’s test.</p>
<div class="sourceCode" id="cb341"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb341-1"><a href="means-separation-and-data-presentation.html#cb341-1" aria-hidden="true" tabindex="-1"></a>lsd<span class="sc">$</span>groups <span class="sc">%&gt;%</span></span>
<span id="cb341-2"><a href="means-separation-and-data-presentation.html#cb341-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;rate&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb341-3"><a href="means-separation-and-data-presentation.html#cb341-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>rate, <span class="at">y=</span>corm_number)) <span class="sc">+</span></span>
<span id="cb341-4"><a href="means-separation-and-data-presentation.html#cb341-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="at">color=</span><span class="st">&quot;black&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;darkolivegreen&quot;</span>) <span class="sc">+</span></span>
<span id="cb341-5"><a href="means-separation-and-data-presentation.html#cb341-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">x=</span>rate, <span class="at">y=</span>corm_number<span class="sc">+</span><span class="dv">15</span>, <span class="at">label=</span>groups), <span class="at">size=</span><span class="dv">6</span>) <span class="sc">+</span></span>
<span id="cb341-6"><a href="means-separation-and-data-presentation.html#cb341-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">12</span>),</span>
<span id="cb341-7"><a href="means-separation-and-data-presentation.html#cb341-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>))</span></code></pre></div>
<p><img src="08-Means-Separation-and-Presentation_files/figure-html/unnamed-chunk-28-1.png" width="672" /></p>
<p>We will learn more about graphics in the exercises this unit. A very comprehensive resource for creating plots in R is <em>R Graphics Cookbook</em>, by Winston Chang. An online copy is available at <a href="https://r-graphics.org/" class="uri">https://r-graphics.org/</a>. Print copies can also be purchased from common booksellers. This book explains not only how to create plots, but how to adjust labels, legends, axis lables, and so on.</p>
</div>
</div>
<div id="exercise-lsd-and-tukeys-hsd" class="section level2" number="8.8">
<h2><span class="header-section-number">8.8</span> Exercise: LSD and Tukey’s HSD</h2>
<p>In the lecture, we learned two ways of grouping treatment means: by <em>Least Significant Difference" and by </em>Tukey’s HSD (Honest Significant Difference. Both of these tests can be run very quicly in R.</p>
<div id="case-study-common-bean" class="section level3" number="8.8.1">
<h3><span class="header-section-number">8.8.1</span> Case Study: Common Bean</h3>
<p>The yield of organically-grown common bean was assessed following four crops: conventional broccoli, green_manure, fallow, and organic broccoli. Yield is in tons per hectare. We wish to separate the treatment means.</p>
<div class="sourceCode" id="cb342"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb342-1"><a href="means-separation-and-data-presentation.html#cb342-1" aria-hidden="true" tabindex="-1"></a>common_bean <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data-unit-8/exercise_data/common_bean.csv&quot;</span>)</span>
<span id="cb342-2"><a href="means-separation-and-data-presentation.html#cb342-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(common_bean)</span></code></pre></div>
<pre><code>##   block        prev_crop fresh_weight
## 1     1    conv_broccoli     52.15924
## 2     1     green_manure     49.82191
## 3     1 organic_broccoli     36.80926
## 4     1           fallow     54.05342
## 5     2    conv_broccoli     52.79726
## 6     2           fallow     47.68364</code></pre>
<p>First, we need to create our linear model and run our analysis of variance. The trial is a randomized complete block design, so our linear model is:</p>
<p>fresh_weight = mu + block + prev_crop + error</p>
<p>Our model statement in R is:</p>
<div class="sourceCode" id="cb344"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb344-1"><a href="means-separation-and-data-presentation.html#cb344-1" aria-hidden="true" tabindex="-1"></a>common_bean_model <span class="ot">=</span> <span class="fu">aov</span>(fresh_weight <span class="sc">~</span> block <span class="sc">+</span> prev_crop, <span class="at">data=</span>common_bean)</span></code></pre></div>
<p>Our analysis of variance is:</p>
<div class="sourceCode" id="cb345"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb345-1"><a href="means-separation-and-data-presentation.html#cb345-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(common_bean_model)</span></code></pre></div>
<pre><code>##             Df Sum Sq Mean Sq F value Pr(&gt;F)   
## block        1    2.4    2.37   0.179 0.6802   
## prev_crop    3  428.9  142.98  10.835 0.0013 **
## Residuals   11  145.1   13.20                  
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>We see the effect of previous crop is significant, so now we want to separate the treatment means.</p>
</div>
<div id="least-significant-difference-1" class="section level3" number="8.8.2">
<h3><span class="header-section-number">8.8.2</span> Least Significant Difference</h3>
<p>Both the least significant test and Tukey’s Honest Significant Test will be run using functions from the <em>agricolae</em> package. We load the agricolae package to our session using the <em>library()</em> command</p>
<div class="sourceCode" id="cb347"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb347-1"><a href="means-separation-and-data-presentation.html#cb347-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(agricolae)</span></code></pre></div>
<p>After this. We can run the least significant test using tne <em>LSD.test()</em> function. This argument takes two arguments: a linear model, and the treatment we wish to separate (which must be given in quotes). We can simply reference the common_bean_model created above.</p>
<div class="sourceCode" id="cb348"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb348-1"><a href="means-separation-and-data-presentation.html#cb348-1" aria-hidden="true" tabindex="-1"></a>lsd_common_bean <span class="ot">=</span> <span class="fu">LSD.test</span>(common_bean_model, <span class="st">&quot;prev_crop&quot;</span>)</span>
<span id="cb348-2"><a href="means-separation-and-data-presentation.html#cb348-2" aria-hidden="true" tabindex="-1"></a>lsd_common_bean</span></code></pre></div>
<pre><code>## $statistics
##    MSerror Df     Mean       CV  t.value      LSD
##   13.19522 11 48.04057 7.561363 2.200985 5.653408
## 
## $parameters
##         test p.ajusted    name.t ntr alpha
##   Fisher-LSD      none prev_crop   4  0.05
## 
## $means
##                  fresh_weight      std r      LCL      UCL      Min      Max
## conv_broccoli        52.79032 1.242884 4 48.79276 56.78788 51.68091 54.52386
## fallow               50.53382 2.718281 4 46.53625 54.53138 47.68364 54.05342
## green_manure         49.52910 5.135442 4 45.53153 53.52666 44.06679 56.34669
## organic_broccoli     39.30905 3.723475 4 35.31149 43.30662 36.10887 44.26354
##                       Q25      Q50      Q75
## conv_broccoli    52.03966 52.47825 53.22891
## fallow           48.93562 50.19911 51.79730
## green_manure     46.92744 48.85145 51.45311
## organic_broccoli 36.63416 38.43190 41.10679
## 
## $comparison
## NULL
## 
## $groups
##                  fresh_weight groups
## conv_broccoli        52.79032      a
## fallow               50.53382      a
## green_manure         49.52910      a
## organic_broccoli     39.30905      b
## 
## attr(,&quot;class&quot;)
## [1] &quot;group&quot;</code></pre>
<p>The test results are returned in a list, which in R is a collection of objects grouped under one name (“lsd_common_bean” in our code above). We can reference different parts of this list the same way you would reference columns within a data frame: using the dollar sign to separate the list name and the object. For example, if we just wanted to look at our $statistics, we would type:</p>
<div class="sourceCode" id="cb350"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb350-1"><a href="means-separation-and-data-presentation.html#cb350-1" aria-hidden="true" tabindex="-1"></a>lsd_common_bean<span class="sc">$</span>statistics</span></code></pre></div>
<pre><code>##    MSerror Df     Mean       CV  t.value      LSD
##   13.19522 11 48.04057 7.561363 2.200985 5.653408</code></pre>
<p>We can reference the means groupings, as well:</p>
<div class="sourceCode" id="cb352"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb352-1"><a href="means-separation-and-data-presentation.html#cb352-1" aria-hidden="true" tabindex="-1"></a>lsd_common_bean<span class="sc">$</span>groups</span></code></pre></div>
<pre><code>##                  fresh_weight groups
## conv_broccoli        52.79032      a
## fallow               50.53382      a
## green_manure         49.52910      a
## organic_broccoli     39.30905      b</code></pre>
<p>This selectivity is handy if we are writing a report and just want to reference part of the output.</p>
<p>In the grouping above, we see that common bean fresh weight was statistically similar following conventional broccoli, fallow, and green manure. Common bean fresh weight following organic broccoli, however, was statistically less than following the other three treatments.</p>
</div>
<div id="tukey-hsd" class="section level3" number="8.8.3">
<h3><span class="header-section-number">8.8.3</span> Tukey HSD</h3>
<p>We can separate the means, very similarly, using Tukey’s Honest Significant Difference. We use the <em>HSD.test()</em> function. The first two arguments are the same as for the LSD.test(): the linear model and treatment (in quotes) we wish to separate. We add a third argument, “group = TRUE.” This tells the HSD.test to return a letter-grouping of treatment means.</p>
<div class="sourceCode" id="cb354"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb354-1"><a href="means-separation-and-data-presentation.html#cb354-1" aria-hidden="true" tabindex="-1"></a>common_bean_hsd <span class="ot">=</span> <span class="fu">HSD.test</span>(common_bean_model, <span class="st">&quot;prev_crop&quot;</span>, <span class="at">group =</span> <span class="cn">TRUE</span>)</span>
<span id="cb354-2"><a href="means-separation-and-data-presentation.html#cb354-2" aria-hidden="true" tabindex="-1"></a>common_bean_hsd</span></code></pre></div>
<pre><code>## $statistics
##    MSerror Df     Mean       CV      MSD
##   13.19522 11 48.04057 7.561363 7.730267
## 
## $parameters
##    test    name.t ntr StudentizedRange alpha
##   Tukey prev_crop   4         4.256143  0.05
## 
## $means
##                  fresh_weight      std r      Min      Max      Q25      Q50
## conv_broccoli        52.79032 1.242884 4 51.68091 54.52386 52.03966 52.47825
## fallow               50.53382 2.718281 4 47.68364 54.05342 48.93562 50.19911
## green_manure         49.52910 5.135442 4 44.06679 56.34669 46.92744 48.85145
## organic_broccoli     39.30905 3.723475 4 36.10887 44.26354 36.63416 38.43190
##                       Q75
## conv_broccoli    53.22891
## fallow           51.79730
## green_manure     51.45311
## organic_broccoli 41.10679
## 
## $comparison
## NULL
## 
## $groups
##                  fresh_weight groups
## conv_broccoli        52.79032      a
## fallow               50.53382      a
## green_manure         49.52910      a
## organic_broccoli     39.30905      b
## 
## attr(,&quot;class&quot;)
## [1] &quot;group&quot;</code></pre>
<p>The HSD.test output, like the lsd output is a list. We can again reference individual tables within it:</p>
<div class="sourceCode" id="cb356"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb356-1"><a href="means-separation-and-data-presentation.html#cb356-1" aria-hidden="true" tabindex="-1"></a>common_bean_hsd<span class="sc">$</span>groups</span></code></pre></div>
<pre><code>##                  fresh_weight groups
## conv_broccoli        52.79032      a
## fallow               50.53382      a
## green_manure         49.52910      a
## organic_broccoli     39.30905      b</code></pre>
</div>
<div id="practice-apple" class="section level3" number="8.8.4">
<h3><span class="header-section-number">8.8.4</span> Practice: Apple</h3>
<p>Apple yield was measured after treatment with four biostimulants. The design was a randomized complete block. The yield was measured in tons per hectare.</p>
<div class="sourceCode" id="cb358"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb358-1"><a href="means-separation-and-data-presentation.html#cb358-1" aria-hidden="true" tabindex="-1"></a>apple <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data-unit-8/exercise_data/apple.csv&quot;</span>)</span></code></pre></div>
<div id="analysis-of-variance-4" class="section level4" number="8.8.4.1">
<h4><span class="header-section-number">8.8.4.1</span> Analysis of Variance</h4>
<p>Run the analysis of variance. Your results should be:</p>
<pre><code>        Df Sum Sq Mean Sq F value  Pr(&gt;F)   </code></pre>
<p>block 1 0.40 0.40 0.076 0.78800<br />
foliar_trt 3 162.10 54.03 10.345 0.00156 **
Residuals 11 57.45 5.22</p>
<p>Signif. codes: 0 ‘<em><strong>’ 0.001 ‘</strong>’ 0.01 ‘</em>’ 0.05 ‘.’ 0.1 ‘ ’ 1</p>
</div>
<div id="least-significant-difference-2" class="section level4" number="8.8.4.2">
<h4><span class="header-section-number">8.8.4.2</span> Least Significant Difference</h4>
<p>Run the Least Significant Difference test on the treatment means. Your results should look like:</p>
<p>$statistics
MSerror Df Mean CV t.value LSD
5.222995 11 47.96522 4.764676 2.200985 3.55682</p>
<p>$parameters
test p.ajusted name.t ntr alpha
Fisher-LSD none foliar_trt 4 0.05</p>
<p>$means
yield std r LCL UCL Min Max Q25 Q50 Q75
conv 45.70984 1.678378 4 43.19478 48.22489 44.44665 48.15233 44.71817 45.12018 46.11185
conv+calcium 44.08293 1.274706 4 41.56788 46.59798 42.34804 45.30697 43.58313 44.33835 44.83815
conv+seaweed 51.90854 2.073312 4 49.39349 54.42359 50.18903 54.78184 50.51099 51.33165 52.72920
conv+si+zn 50.15958 3.246956 4 47.64452 52.67463 46.88179 54.60709 48.55789 49.57471 51.17639</p>
<p>$comparison
NULL</p>
<p>$groups
yield groups
conv+seaweed 51.90854 a
conv+si+zn 50.15958 a
conv 45.70984 b
conv+calcium 44.08293 b</p>
<p>attr(,“class”)
[1] “group”</p>
</div>
<div id="tukeys-hsd" class="section level4" number="8.8.4.3">
<h4><span class="header-section-number">8.8.4.3</span> Tukey’s HSD</h4>
<p>Run the Honest Significant Difference test on the treatment means. Your results should look like:</p>
<p>$statistics
MSerror Df Mean CV MSD
5.222995 11 47.96522 4.764676 4.863468</p>
<p>$parameters
test name.t ntr StudentizedRange alpha
Tukey foliar_trt 4 4.256143 0.05</p>
<p>$means
yield std r Min Max Q25 Q50 Q75
conv 45.70984 1.678378 4 44.44665 48.15233 44.71817 45.12018 46.11185
conv+calcium 44.08293 1.274706 4 42.34804 45.30697 43.58313 44.33835 44.83815
conv+seaweed 51.90854 2.073312 4 50.18903 54.78184 50.51099 51.33165 52.72920
conv+si+zn 50.15958 3.246956 4 46.88179 54.60709 48.55789 49.57471 51.17639</p>
<p>$comparison
NULL</p>
<p>$groups
yield groups
conv+seaweed 51.90854 a
conv+si+zn 50.15958 ab
conv 45.70984 bc
conv+calcium 44.08293 c</p>
<p>attr(,“class”)
[1] “group”</p>
</div>
</div>
<div id="practice-wheat-treatment-with-mildew" class="section level3" number="8.8.5">
<h3><span class="header-section-number">8.8.5</span> Practice: Wheat Treatment with Mildew</h3>
<p>Four foliar treatments against powdery mildew in wheat were tested in Rothemsted, England, for their effect on grain yield. The same fungicide was tested with four timings: 0 (none), 1 (early), 2 (late), R (repeated).</p>
<p>The treatment was a randomized complete block design. Yield was measured in tons per hectare.</p>
<div class="sourceCode" id="cb360"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb360-1"><a href="means-separation-and-data-presentation.html#cb360-1" aria-hidden="true" tabindex="-1"></a>wheat_mildew <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data-unit-8/exercise_data/mildew.csv&quot;</span>)</span>
<span id="cb360-2"><a href="means-separation-and-data-presentation.html#cb360-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(wheat_mildew)</span></code></pre></div>
<pre><code>##   plot trt block yield
## 1    1  T2    B1  5.73
## 2    2   R    B1  6.08
## 3    3  T0    B1  5.26
## 4    4  T1    B1  5.89
## 5    5  T0    B2  5.37
## 6    6  T2    B2  5.95</code></pre>
<div id="analysis-of-variance-5" class="section level4" number="8.8.5.1">
<h4><span class="header-section-number">8.8.5.1</span> Analysis of Variance</h4>
<p>Run the analysis of variance on the response of wheat yield to fungicide treatment. Your results should look like:</p>
<pre><code>        Df Sum Sq Mean Sq F value   Pr(&gt;F)    </code></pre>
<p>block 8 5.085 0.6356 17.53 2.79e-08 <strong><em>
trt 3 3.129 1.0432 28.77 4.05e-08 </em></strong>
Residuals 24 0.870 0.0363</p>
<p>Signif. codes: 0 ‘<em><strong>’ 0.001 ‘</strong>’ 0.01 ‘</em>’ 0.05 ‘.’ 0.1 ‘ ’ 1</p>
</div>
<div id="least-significant-difference-3" class="section level4" number="8.8.5.2">
<h4><span class="header-section-number">8.8.5.2</span> Least Significant Difference</h4>
<p>Run the Least Significant Difference test on the treatment means. Your results should look like:</p>
<p>$statistics
MSerror Df Mean CV t.value LSD
0.03625532 24 5.801944 3.281802 2.063899 0.1852542</p>
<p>$parameters
test p.ajusted name.t ntr alpha
Fisher-LSD none trt 4 0.05</p>
<p>$means
yield std r LCL UCL Min Max Q25 Q50 Q75
R 5.942222 0.4649403 9 5.811228 6.073217 5.06 6.54 5.76 6.01 6.18
T0 5.310000 0.4519956 9 5.179006 5.440994 4.38 5.82 5.16 5.26 5.73
T1 5.867778 0.4602928 9 5.736783 5.998772 5.04 6.45 5.59 5.89 6.26
T2 6.087778 0.3347304 9 5.956783 6.218772 5.63 6.48 5.76 6.14 6.43</p>
<p>$comparison
NULL</p>
<p>$groups
yield groups
T2 6.087778 a
R 5.942222 ab
T1 5.867778 b
T0 5.310000 c</p>
<p>attr(,“class”)
[1] “group”</p>
</div>
<div id="tukeys-honest-significant-difference-test" class="section level4" number="8.8.5.3">
<h4><span class="header-section-number">8.8.5.3</span> Tukey’s Honest Significant Difference Test</h4>
<p>Run the HSD test on the treatment means. Your results should look like:</p>
<p>$statistics
MSerror Df Mean CV MSD
0.03625532 24 5.801944 3.281802 0.2476109</p>
<p>$parameters
test name.t ntr StudentizedRange alpha
Tukey trt 4 3.901262 0.05</p>
<p>$means
yield std r Min Max Q25 Q50 Q75
R 5.942222 0.4649403 9 5.06 6.54 5.76 6.01 6.18
T0 5.310000 0.4519956 9 4.38 5.82 5.16 5.26 5.73
T1 5.867778 0.4602928 9 5.04 6.45 5.59 5.89 6.26
T2 6.087778 0.3347304 9 5.63 6.48 5.76 6.14 6.43</p>
<p>$comparison
NULL</p>
<p>$groups
yield groups
T2 6.087778 a
R 5.942222 a
T1 5.867778 a
T0 5.310000 b</p>
</div>
</div>
</div>
<div id="exercise-linear-contrasts" class="section level2" number="8.9">
<h2><span class="header-section-number">8.9</span> Exercise: Linear Contrasts</h2>
<p>Linear contrasts are different from least significant difference (LSD) or Tukey’s honest significant difference (HSD) tests in that they can test groups of treatments. As we learned in the lecture, the simplest way to think of a linear constrast is as a t-test between two groups.</p>
<p>We also learned in the lecture that we use contrast coefficients to define these groups. The coefficients for one group should add up to 1; the coefficients of the other group should add up to -1. That way, we subtract the average value for treatments in one group from the average value for treatments in the other group.</p>
<div id="case-study-winter-canola-cultivar-trial." class="section level3" number="8.9.1">
<h3><span class="header-section-number">8.9.1</span> Case Study: Winter Canola Cultivar Trial.</h3>
<p>Six winter canola cultivars were tested in a randomized complete block trial in Manitoba. Yield is in kg / ha.</p>
<div class="sourceCode" id="cb363"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb363-1"><a href="means-separation-and-data-presentation.html#cb363-1" aria-hidden="true" tabindex="-1"></a>canola <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data-unit-8/exercise_data/canola_gd.csv&quot;</span>)</span>
<span id="cb363-2"><a href="means-separation-and-data-presentation.html#cb363-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(canola)</span></code></pre></div>
<pre><code>##   block cultivar    yield
## 1     1      Bob 1508.424
## 2     2      Bob 1962.072
## 3     3      Bob 1364.861
## 4     4      Bob 1832.715
## 5     1    Donna 1212.190
## 6     2    Donna 1629.680</code></pre>
<div id="anova-6" class="section level4" number="8.9.1.1">
<h4><span class="header-section-number">8.9.1.1</span> ANOVA</h4>
<p>First, let’s run our analysis of variance</p>
<div class="sourceCode" id="cb365"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb365-1"><a href="means-separation-and-data-presentation.html#cb365-1" aria-hidden="true" tabindex="-1"></a>canola_model <span class="ot">=</span> <span class="fu">aov</span>(yield <span class="sc">~</span> block <span class="sc">+</span> cultivar, <span class="at">data =</span> canola)</span>
<span id="cb365-2"><a href="means-separation-and-data-presentation.html#cb365-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(canola_model)</span></code></pre></div>
<pre><code>##             Df  Sum Sq Mean Sq F value   Pr(&gt;F)    
## block        1   18600   18600   0.171 0.684822    
## cultivar     5 4551035  910207   8.344 0.000388 ***
## Residuals   17 1854462  109086                     
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
<div id="least-significant-difference-test" class="section level4" number="8.9.1.2">
<h4><span class="header-section-number">8.9.1.2</span> Least Significant Difference Test</h4>
<p>Next, lets run an LSD test.</p>
<div class="sourceCode" id="cb367"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb367-1"><a href="means-separation-and-data-presentation.html#cb367-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(agricolae)</span>
<span id="cb367-2"><a href="means-separation-and-data-presentation.html#cb367-2" aria-hidden="true" tabindex="-1"></a>lsd_canola <span class="ot">=</span> <span class="fu">LSD.test</span>(canola_model, <span class="st">&quot;cultivar&quot;</span>)</span>
<span id="cb367-3"><a href="means-separation-and-data-presentation.html#cb367-3" aria-hidden="true" tabindex="-1"></a>lsd_canola</span></code></pre></div>
<pre><code>## $statistics
##   MSerror Df     Mean       CV  t.value      LSD
##    109086 17 2296.983 14.37894 2.109816 492.7357
## 
## $parameters
##         test p.ajusted   name.t ntr alpha
##   Fisher-LSD      none cultivar   6  0.05
## 
## $means
##           yield      std r      LCL      UCL      Min      Max      Q25
## Bill   2296.268 326.6421 4 1947.851 2644.685 1865.903 2614.081 2142.888
## Bob    1667.018 277.4666 4 1318.601 2015.435 1364.861 1962.072 1472.533
## Donna  1833.282 514.0721 4 1484.865 2181.699 1212.190 2358.635 1525.308
## Jerry  2669.161 176.6376 4 2320.745 3017.578 2545.158 2923.794 2552.218
## Mickey 2406.509 263.5791 4 2058.092 2754.926 2130.413 2656.553 2207.614
## Phil   2909.658 275.1842 4 2561.242 3258.075 2565.683 3218.583 2776.782
##             Q50      Q75
## Bill   2352.545 2505.925
## Bob    1670.570 1865.055
## Donna  1881.152 2189.126
## Jerry  2603.847 2720.790
## Mickey 2419.535 2618.430
## Phil   2927.183 3060.060
## 
## $comparison
## NULL
## 
## $groups
##           yield groups
## Phil   2909.658      a
## Jerry  2669.161     ab
## Mickey 2406.509      b
## Bill   2296.268     bc
## Donna  1833.282     cd
## Bob    1667.018      d
## 
## attr(,&quot;class&quot;)
## [1] &quot;group&quot;</code></pre>
</div>
<div id="contrasts" class="section level4" number="8.9.1.3">
<h4><span class="header-section-number">8.9.1.3</span> Contrasts</h4>
<p>Contrasts should be based on specific questions we want to ask of the data. In this case, there are two questions we would like to answer with contrasts.</p>
<p>First, cultivars “Donna” and “Bob” are from a line of canola called “PITB.” Sometimes they perform well, but other times their performance can be rather discordant. Our first question is, do cultivars from the PITB line perform worse than cultivars from other lines.</p>
<p>Cultivars “Mickey” and “Bill” are from the “FOTM” line of cultivars. They regularly perform better than cultivars with “PITB” pedigrees. But in bad seasons they have a field performance that has been likened to “sneakers in a drier.” Meanwhile, cultivars “Phil” and “Jerry,” from the DKSTR line, under appropriate growing conditions, have a performance potential that is almose unlimited. Our second question, then, is whether cultivars from the DKSTR outperform those from the FOTM line.</p>
</div>
<div id="figuring-out-which-coefficients-to-use" class="section level4" number="8.9.1.4">
<h4><span class="header-section-number">8.9.1.4</span> Figuring Out Which Coefficients to Use</h4>
<p>To answer these questions, we need to define our coefficients. The order of the coefficients must reflect the order of the factor levels in R. To determine this order we use the <em>levels()</em> function.</p>
<div class="sourceCode" id="cb369"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb369-1"><a href="means-separation-and-data-presentation.html#cb369-1" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(canola<span class="sc">$</span>cultivar)</span></code></pre></div>
<pre><code>## NULL</code></pre>
<p>What coefficients do we use, then? Our first hypothesis is that “Bob” and “Donna” perform worse than others.</p>
<ol style="list-style-type: decimal">
<li>So we will assign a -1 to both those cultivars.</li>
</ol>
<p>Coefficients: 0 -1 -1 0 0 0</p>
<ol start="2" style="list-style-type: decimal">
<li>Since we are comparing against the other four cultivars, we assign them +1:</li>
</ol>
<p>Coefficients: +1 -1 -1 +1 +1 +1</p>
<ol start="3" style="list-style-type: decimal">
<li>The coefficients of each group must sum to 1. So we will divide the coefficients for Bob and Donna by 2:</li>
</ol>
<p>Coefficients: +1 -1/2 -1/2 +1 +1 +1</p>
<ol start="4" style="list-style-type: decimal">
<li>And we will divide the coefficients for the other cultivars by 4:</li>
</ol>
<p>Coefficients: +1/4 -1/2 -1/2 +1/4 +1/4 +1/4</p>
<ol start="5" style="list-style-type: decimal">
<li>Now, let’s check our math. The coefficients for Bob and Donna should sum to -1</li>
</ol>
<p>Coefficients: 0 -1/2 -1/2 0 0 0 = -1</p>
<ol start="6" style="list-style-type: decimal">
<li>The coefficients for the other four cultivars should sum to 1.</li>
</ol>
<p>Coefficients: +1/4 0 0 +1/4 +1/4 +1/4 = 1</p>
<ol start="7" style="list-style-type: decimal">
<li>Finally, all the coefficients together should sum to zero:</li>
</ol>
<p>Coefficients: +1/4 -1/2 -1/2 +1/4 +1/4 +1/4 = 0</p>
</div>
<div id="telling-r-the-coefficients-to-use" class="section level4" number="8.9.1.5">
<h4><span class="header-section-number">8.9.1.5</span> Telling R the Coefficients to Use</h4>
<p>Running the linear coefficient in R requires we define a “coefficient matrix.” A matrix is a slightly more primative version of the data.frame – still akin to a table, but its rows and columns tend to be defined by numbers instead of names.</p>
<p>To create a coefficient matrix in R, we need to define it as follows:</p>
<p>K = matrix(c(insert coefficients here),1)</p>
<p>“c(coefficients)” tells R to fill in the coefficients across columns. The “,1” that follows tells R to put those values in the first row. That second part of the matrix, the “,1” is <em>really important</em>. In creating this lesson, I spent at least a half-hour trying to figure out why one of my contrasts would not work. It was because I had written it:</p>
<p>K_bd_vs_others = matrix(c(+1/4, -1/2, -1/2, +1/4, +1/4, +1/4))</p>
<p>Instead of:</p>
<p>K_bd_vs_others = matrix(c(+1/4, -1/2, -1/2, +1/4, +1/4, +1/4), 1)</p>
<div class="sourceCode" id="cb371"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb371-1"><a href="means-separation-and-data-presentation.html#cb371-1" aria-hidden="true" tabindex="-1"></a>K_bd_vs_others <span class="ot">=</span> <span class="fu">matrix</span>(<span class="fu">c</span>(<span class="sc">+</span><span class="dv">1</span><span class="sc">/</span><span class="dv">4</span>, <span class="sc">-</span><span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>, <span class="sc">-</span><span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>, <span class="sc">+</span><span class="dv">1</span><span class="sc">/</span><span class="dv">4</span>, <span class="sc">+</span><span class="dv">1</span><span class="sc">/</span><span class="dv">4</span>, <span class="sc">+</span><span class="dv">1</span><span class="sc">/</span><span class="dv">4</span>),<span class="dv">1</span>)</span>
<span id="cb371-2"><a href="means-separation-and-data-presentation.html#cb371-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb371-3"><a href="means-separation-and-data-presentation.html#cb371-3" aria-hidden="true" tabindex="-1"></a>K_bd_vs_others</span></code></pre></div>
<pre><code>##      [,1] [,2] [,3] [,4] [,5] [,6]
## [1,] 0.25 -0.5 -0.5 0.25 0.25 0.25</code></pre>
<p>Above is our coefficient matrix for “Bob” and “Donna” versus the others.</p>
<p>Now let’s create our second matrix, for “Mickey” and “Bill” versus “Phil” and “Jerry.” Let’s check the order of the factor levels again.</p>
<div class="sourceCode" id="cb373"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb373-1"><a href="means-separation-and-data-presentation.html#cb373-1" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(canola<span class="sc">$</span>cultivar)</span></code></pre></div>
<pre><code>## NULL</code></pre>
<p>Our hypothesis is that “Phil” and “Jerry” perform better than “Mickey” and “Bill,” so we will assign them the initial coefficients of 1</p>
<ol style="list-style-type: decimal">
<li>So we will assign a -1 to both those cultivars.</li>
</ol>
<p>Coefficients: 0 0 0 1 0 1</p>
<ol start="2" style="list-style-type: decimal">
<li>We then assign Mickey and Bill the coefficients -1:</li>
</ol>
<p>Coefficients: -1 0 0 +1 -1 +1</p>
<p>“Donna” and “Bob” are assigned the coefficient zero since they are not involved in this comparison.</p>
<ol start="3" style="list-style-type: decimal">
<li>The coefficients of each group must sum to 1. So we will divide the coefficients for “Jerry” and “Phil” by 2:</li>
</ol>
<p>Coefficients: -1 0 0 +1/2 -1 +1/2</p>
<ol start="4" style="list-style-type: decimal">
<li>We do the same with “Mickey” and “Bill”:</li>
</ol>
<p>Coefficients: -1/2 0 0 +1/2 -1/2 +1/2</p>
<ol start="5" style="list-style-type: decimal">
<li>Now, let’s check our math. The coefficients for Jerry and Phil should sum to 1</li>
</ol>
<p>Coefficients: -0 0 0 +1/2 0 +1/2 = 1</p>
<ol start="6" style="list-style-type: decimal">
<li>The coefficients for Mickey and Bill should sum to -1.</li>
</ol>
<p>Coefficients: -1/2 0 0 0 -1/2 0 = -1</p>
<p>Finally, all the coefficients together should sum to zero:</p>
<p>Coefficients: -1/2 0 0 +1/2 -1/2 +1/2 = 0</p>
<p>Our correlation coefficient for our second contrast, then, is:</p>
<div class="sourceCode" id="cb375"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb375-1"><a href="means-separation-and-data-presentation.html#cb375-1" aria-hidden="true" tabindex="-1"></a>K_jp_vs_mb <span class="ot">=</span> <span class="fu">matrix</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="sc">+</span><span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>, <span class="sc">-</span><span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>, <span class="sc">+</span><span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>),<span class="dv">1</span>)</span>
<span id="cb375-2"><a href="means-separation-and-data-presentation.html#cb375-2" aria-hidden="true" tabindex="-1"></a>K_jp_vs_mb</span></code></pre></div>
<pre><code>##      [,1] [,2] [,3] [,4] [,5] [,6]
## [1,] -0.5    0    0  0.5 -0.5  0.5</code></pre>
</div>
</div>
<div id="running-the-contrast" class="section level3" number="8.9.2">
<h3><span class="header-section-number">8.9.2</span> Running the Contrast</h3>
<p>Defining the contrast coefficients is difficult. Running the contrast afterwards is relatively easy. We use the <em>ghlt()</em> function from the <em>multicomp()</em> package. “glht” stands for “generalized linear hypothesis test.”</p>
<p><em>glht()</em> requires two arguments: a linear model, and the coefficient contrast we created above. The linear model is a little different than what we defined for the ANOVA above. It only contains the terms “0” and the treatment name:</p>
<div class="sourceCode" id="cb377"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb377-1"><a href="means-separation-and-data-presentation.html#cb377-1" aria-hidden="true" tabindex="-1"></a>canola_contrast_model <span class="ot">=</span> <span class="fu">aov</span>(yield <span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> cultivar, <span class="at">data=</span>canola)</span></code></pre></div>
<p>The reason for this is because our contrast is isolating the effect of cultivar. The zero forces our contrast to calculate the actual difference between the two groups’ means (which we may like to report).</p>
<p>Now we can run our contrast</p>
<div class="sourceCode" id="cb378"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb378-1"><a href="means-separation-and-data-presentation.html#cb378-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(multcomp)</span>
<span id="cb378-2"><a href="means-separation-and-data-presentation.html#cb378-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb378-3"><a href="means-separation-and-data-presentation.html#cb378-3" aria-hidden="true" tabindex="-1"></a>bd_vs_others <span class="ot">=</span> <span class="fu">glht</span>(canola_contrast_model, <span class="at">linfct =</span> K_bd_vs_others)</span></code></pre></div>
<p>We will use <em>summary()</em> to summarise these results.</p>
<div class="sourceCode" id="cb379"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb379-1"><a href="means-separation-and-data-presentation.html#cb379-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(bd_vs_others)</span></code></pre></div>
<pre><code>## 
##   Simultaneous Tests for General Linear Hypotheses
## 
## Fit: aov(formula = yield ~ 0 + cultivar, data = canola)
## 
## Linear Hypotheses:
##        Estimate Std. Error t value Pr(&gt;|t|)    
## 1 == 0    820.2      139.7   5.872 1.47e-05 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## (Adjusted p values reported -- single-step method)</code></pre>
<p>Ok, let’s review our results. The Estimate, 820.2 kg / ha, is the difference between the group means. Since we subracted the mean of Bob and Donna from the mean of the other four cultivars, we concluded that Bob and Donna as a group yielded <em>less</em> than the other four cultivars as a group.</p>
<p>The Estimate was divided by Std. Error, the standard error of the difference, to produce the t value of 5.872. The probability of observing a t-value of this size or greater, if in fact the true difference between groups was zero, is very small, 1.47 x 10^-5. The difference between groups is significant.</p>
<p>We conclude that cultivars from the PITB line yielded worse than other cultivars as a group.</p>
<p>We can now run the second contrast, Jerry and Phil versus Mickey and Bill. We again use the canola_contrast_model as our linear model, and this time, the coefficient matrix K_jp_vs_mb</p>
<div class="sourceCode" id="cb381"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb381-1"><a href="means-separation-and-data-presentation.html#cb381-1" aria-hidden="true" tabindex="-1"></a>jp_vs_mb <span class="ot">=</span> <span class="fu">glht</span>(canola_contrast_model, <span class="at">linfct=</span>K_jp_vs_mb)</span>
<span id="cb381-2"><a href="means-separation-and-data-presentation.html#cb381-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(jp_vs_mb)</span></code></pre></div>
<pre><code>## 
##   Simultaneous Tests for General Linear Hypotheses
## 
## Fit: aov(formula = yield ~ 0 + cultivar, data = canola)
## 
## Linear Hypotheses:
##        Estimate Std. Error t value Pr(&gt;|t|)  
## 1 == 0    438.0      161.3   2.716   0.0142 *
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## (Adjusted p values reported -- single-step method)</code></pre>
<p>The mean of Jerry and Phill was 438.0 kg/ha greater than the mean of Mickey and Bill. (Note: when you look at the Estimate, always ask yourself whether the number, based on treatment means, makes sense. The first time I ran this contrast, for example, the estimate was greater than 2000. This caused me to review my contrast coefficients, where I discovered one of the coefficients was positive when it should have been negative.)</p>
<p>Our t value again has a probability of occuring by chance of less than P=0.05, so the difference between the mean of Jerry and the mean of Phil and Mickey and Bill is significant. We conclude that cultivars from the DKSTR line yielded better than cultivars from the FOTM line.</p>
</div>
<div id="practice-corn-nitrogen-source-and-timing" class="section level3" number="8.9.3">
<h3><span class="header-section-number">8.9.3</span> Practice: Corn Nitrogen Source and Timing</h3>
<p>This is a really powerful example of how contrasts can be used, not just to test significances, but generate much broader understandings than either the ANOVA or LSD/HSD alone would allow. Our practice data set is from a nitrogen management trial was conducted near Whitehouse, Ohio, to compare 7 nitrogen management strategies in corn. All treatments (other than the control) provided 200 units (lbs) total N during the growing season.</p>
<div class="sourceCode" id="cb383"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb383-1"><a href="means-separation-and-data-presentation.html#cb383-1" aria-hidden="true" tabindex="-1"></a>n_source_timing <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data-unit-8/exercise_data/corn_nitrogen_source_timing.csv&quot;</span>)</span>
<span id="cb383-2"><a href="means-separation-and-data-presentation.html#cb383-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(n_source_timing)</span></code></pre></div>
<pre><code>##           trt    yield block  B      Error
## 1     control 147.9109    B1 -4  3.9108674
## 2      aa_pre 178.1029    B1 -4 -1.8970790
## 3     aa_post 173.5400    B1 -4  5.5400054
## 4 aa_pre_post 184.2516    B1 -4 -2.7483991
## 5     uan_pre 170.4471    B1 -4 -0.5529385
## 6    uan_post 165.4751    B1 -4 -1.5249071</code></pre>
<p>Lets look at the seven levels of nitrogen treatment:</p>
<div class="sourceCode" id="cb385"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb385-1"><a href="means-separation-and-data-presentation.html#cb385-1" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(n_source_timing<span class="sc">$</span>trt)</span></code></pre></div>
<pre><code>## NULL</code></pre>
<p>Treatment abbreviations are: “aa” = anhydrous amonia, “uan” = urea ammonium nitrate, “pre” = pre-plant, and “post” = post-emergence (sidedress).</p>
<div id="anova-7" class="section level4" number="8.9.3.1">
<h4><span class="header-section-number">8.9.3.1</span> ANOVA</h4>
<p>First, let’s look at our analysis of variance:</p>
<div class="sourceCode" id="cb387"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb387-1"><a href="means-separation-and-data-presentation.html#cb387-1" aria-hidden="true" tabindex="-1"></a>corn_nitrogen_model  <span class="ot">=</span> <span class="fu">aov</span>(yield <span class="sc">~</span> block <span class="sc">+</span> trt, <span class="at">data=</span>n_source_timing)</span>
<span id="cb387-2"><a href="means-separation-and-data-presentation.html#cb387-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(corn_nitrogen_model)</span></code></pre></div>
<pre><code>##             Df Sum Sq Mean Sq F value   Pr(&gt;F)    
## block        3    283    94.3   2.414      0.1    
## trt          6   3421   570.1  14.591 4.81e-06 ***
## Residuals   18    703    39.1                     
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>Our nitrogen treatments are significantly different. What does the least significant test tell us differences among treatment means?</p>
<div class="sourceCode" id="cb389"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb389-1"><a href="means-separation-and-data-presentation.html#cb389-1" aria-hidden="true" tabindex="-1"></a>lsd <span class="ot">=</span> <span class="fu">LSD.test</span>(corn_nitrogen_model, <span class="st">&quot;trt&quot;</span>)</span>
<span id="cb389-2"><a href="means-separation-and-data-presentation.html#cb389-2" aria-hidden="true" tabindex="-1"></a>lsd<span class="sc">$</span>groups</span></code></pre></div>
<pre><code>##                 yield groups
## aa_pre_post  191.2750      a
## aa_pre       181.7686      b
## uan_pre_post 179.3731      b
## uan_pre      175.9014     bc
## aa_post      174.7789     bc
## uan_post     167.9677      c
## control      153.2038      d</code></pre>
<p>We have four questions:
1) is the mean yield of the six nitrogen treatments significantly different from the untreated control?</p>
<ol start="2" style="list-style-type: decimal">
<li><p>is the mean yield of preplant nitrogen treatments significantly different from the mean of postemergence nitrogen treatments?</p></li>
<li><p>is the mean yield of anhydrous ammonia treatments significantly different from the mean of urea ammonium nitrate treatments?</p></li>
<li><p>is the mean yield of split application treatments significantly different from the mean of single application treatments?</p></li>
</ol>
</div>
<div id="nitrogen-treatments-vs-control" class="section level4" number="8.9.3.2">
<h4><span class="header-section-number">8.9.3.2</span> Nitrogen Treatments vs Control</h4>
<p>Lets walk through this one together. What contrast coefficients do we use? Lets first double-check the order of factor levels (treatments):</p>
<div class="sourceCode" id="cb391"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb391-1"><a href="means-separation-and-data-presentation.html#cb391-1" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(n_source_timing<span class="sc">$</span>trt)</span></code></pre></div>
<pre><code>## NULL</code></pre>
<p>So we will assign -1 to the control treatment and 1 to all other treatments:</p>
<p>coefficients: 1 1 1 -1 1 1 1</p>
<p>The coefficients assigned to each group should add up to an absolute value. Since there are six nitrogen treatments, we need to divide each of their coefficients by six.</p>
<p>coefficients: 1/6 1/6 1/6 -1 1/6 1/6 1/6</p>
<p>Now, if we check, we will see the coefficients assigned to the six nitrogen treatments sum to 1, the control coefficient is zero, and the sum of all coefficients is 0.</p>
<p>Now let’s define our first contrast coefficient matrix by filling in the matrix below:</p>
<div class="sourceCode" id="cb393"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb393-1"><a href="means-separation-and-data-presentation.html#cb393-1" aria-hidden="true" tabindex="-1"></a>K_n_vs_control <span class="ot">=</span> <span class="fu">matrix</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">6</span>, <span class="dv">1</span><span class="sc">/</span><span class="dv">6</span>, <span class="dv">1</span><span class="sc">/</span><span class="dv">6</span>, <span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span><span class="sc">/</span><span class="dv">6</span>, <span class="dv">1</span><span class="sc">/</span><span class="dv">6</span>, <span class="dv">1</span><span class="sc">/</span><span class="dv">6</span>), <span class="dv">1</span>)  <span class="co"># don&#39;t forget the &quot;, 1&quot; after the coefficients!</span></span></code></pre></div>
<p>We need to define our nitrogen contrast model. Remember, the model only has 0 and the factor name on the right side of the equation. Complete our equation below.</p>
<div class="sourceCode" id="cb394"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb394-1"><a href="means-separation-and-data-presentation.html#cb394-1" aria-hidden="true" tabindex="-1"></a>n_contrast_model <span class="ot">=</span> <span class="fu">aov</span>(yield <span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> trt, <span class="at">data=</span>n_source_timing)</span></code></pre></div>
<p>Finally, run the contrast using the glht() and summary() functions below:</p>
<div class="sourceCode" id="cb395"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb395-1"><a href="means-separation-and-data-presentation.html#cb395-1" aria-hidden="true" tabindex="-1"></a>n_vs_control <span class="ot">=</span> <span class="fu">glht</span>(n_contrast_model, <span class="at">linfct=</span>K_n_vs_control)</span>
<span id="cb395-2"><a href="means-separation-and-data-presentation.html#cb395-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(n_vs_control)</span></code></pre></div>
<pre><code>## 
##   Simultaneous Tests for General Linear Hypotheses
## 
## Fit: aov(formula = yield ~ 0 + trt, data = n_source_timing)
## 
## Linear Hypotheses:
##        Estimate Std. Error t value Pr(&gt;|t|)    
## 1 == 0   25.307      3.701   6.838 9.26e-07 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## (Adjusted p values reported -- single-step method)</code></pre>
<p>What do you conclude?</p>
</div>
<div id="preplant-vs-postemergence-treatments" class="section level4" number="8.9.3.3">
<h4><span class="header-section-number">8.9.3.3</span> Preplant vs Postemergence Treatments</h4>
<p>Compare the two pre-plant treatments ("_pre“) to the two post-emergence (”_post") treatments. Your results should look like:</p>
<div class="sourceCode" id="cb397"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb397-1"><a href="means-separation-and-data-presentation.html#cb397-1" aria-hidden="true" tabindex="-1"></a>K_pre_vs_post <span class="ot">=</span> <span class="fu">matrix</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>,<span class="sc">-</span><span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>,<span class="sc">-</span><span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>,<span class="dv">0</span>),<span class="dv">1</span>)</span>
<span id="cb397-2"><a href="means-separation-and-data-presentation.html#cb397-2" aria-hidden="true" tabindex="-1"></a>pre_vs_post <span class="ot">=</span> <span class="fu">glht</span>(n_contrast_model, K_pre_vs_post)</span>
<span id="cb397-3"><a href="means-separation-and-data-presentation.html#cb397-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(pre_vs_post)</span></code></pre></div>
<pre><code>## 
##   Simultaneous Tests for General Linear Hypotheses
## 
## Fit: aov(formula = yield ~ 0 + trt, data = n_source_timing)
## 
## Linear Hypotheses:
##        Estimate Std. Error t value Pr(&gt;|t|)  
## 1 == 0   -7.462      3.427  -2.178    0.041 *
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## (Adjusted p values reported -- single-step method)</code></pre>
<pre><code> Simultaneous Tests for General Linear Hypotheses</code></pre>
<p>Fit: aov(formula = yield ~ 0 + trt, data = n_source_timing)</p>
<p>Linear Hypotheses:
Estimate Std. Error t value Pr(&gt;|t|)<br />
1 == 0 -7.462 3.427 -2.178 0.041 *
Signif. codes: 0 ‘<em><strong>’ 0.001 ‘</strong>’ 0.01 ‘</em>’ 0.05 ‘.’ 0.1 ‘ ’ 1
(Adjusted p values reported – single-step method)</p>
</div>
<div id="anhydrous-ammonian-vs-urea-ammonium-nitrate" class="section level4" number="8.9.3.4">
<h4><span class="header-section-number">8.9.3.4</span> Anhydrous Ammonian vs Urea Ammonium Nitrate</h4>
<p>Compare the three anhydrous ammonia (“aa_”) treatments to the three urea ammonium nitrate ("_uan") treatments. Your results should look like:</p>
<div class="sourceCode" id="cb400"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb400-1"><a href="means-separation-and-data-presentation.html#cb400-1" aria-hidden="true" tabindex="-1"></a>K_aa_vs_uan <span class="ot">=</span> <span class="fu">matrix</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>,<span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>,<span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>,<span class="dv">0</span>,<span class="sc">-</span><span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>,<span class="sc">-</span><span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>,<span class="sc">-</span><span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>),<span class="dv">1</span>)</span>
<span id="cb400-2"><a href="means-separation-and-data-presentation.html#cb400-2" aria-hidden="true" tabindex="-1"></a>aa_vs_uan <span class="ot">=</span> <span class="fu">glht</span>(n_contrast_model, K_aa_vs_uan)</span>
<span id="cb400-3"><a href="means-separation-and-data-presentation.html#cb400-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(aa_vs_uan)</span></code></pre></div>
<pre><code>## 
##   Simultaneous Tests for General Linear Hypotheses
## 
## Fit: aov(formula = yield ~ 0 + trt, data = n_source_timing)
## 
## Linear Hypotheses:
##        Estimate Std. Error t value Pr(&gt;|t|)   
## 1 == 0    8.193      2.798   2.929  0.00803 **
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## (Adjusted p values reported -- single-step method)</code></pre>
<pre><code> Simultaneous Tests for General Linear Hypotheses</code></pre>
<p>Fit: aov(formula = yield ~ 0 + trt, data = n_source_timing)</p>
<p>Linear Hypotheses:
Estimate Std. Error t value Pr(&gt;|t|)<br />
1 == 0 8.193 2.798 2.929 0.00803 **
Signif. codes: 0 ‘<em><strong>’ 0.001 ‘</strong>’ 0.01 ‘</em>’ 0.05 ‘.’ 0.1 ‘ ’ 1
(Adjusted p values reported – single-step method)</p>
</div>
<div id="split-vs-single-applications" class="section level4" number="8.9.3.5">
<h4><span class="header-section-number">8.9.3.5</span> Split vs Single Applications</h4>
<p>Compare the two split application ("_pre_post“) treatments to the four single-application (”_pre" or "_post") treatments. Your results should look like:</p>
<div class="sourceCode" id="cb403"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb403-1"><a href="means-separation-and-data-presentation.html#cb403-1" aria-hidden="true" tabindex="-1"></a>K_split_vs_single <span class="ot">=</span> <span class="fu">matrix</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span><span class="sc">/</span><span class="dv">4</span>, <span class="sc">-</span><span class="dv">1</span><span class="sc">/</span><span class="dv">4</span>, <span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>,<span class="dv">0</span>,<span class="sc">-</span><span class="dv">1</span><span class="sc">/</span><span class="dv">4</span>,<span class="sc">-</span><span class="dv">1</span><span class="sc">/</span><span class="dv">4</span>,<span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>),<span class="dv">1</span>)</span>
<span id="cb403-2"><a href="means-separation-and-data-presentation.html#cb403-2" aria-hidden="true" tabindex="-1"></a>split_vs_single <span class="ot">=</span> <span class="fu">glht</span>(n_contrast_model, K_split_vs_single)</span>
<span id="cb403-3"><a href="means-separation-and-data-presentation.html#cb403-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(split_vs_single)</span></code></pre></div>
<pre><code>## 
##   Simultaneous Tests for General Linear Hypotheses
## 
## Fit: aov(formula = yield ~ 0 + trt, data = n_source_timing)
## 
## Linear Hypotheses:
##        Estimate Std. Error t value Pr(&gt;|t|)   
## 1 == 0   10.220      2.967   3.444  0.00243 **
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## (Adjusted p values reported -- single-step method)</code></pre>
<pre><code> Simultaneous Tests for General Linear Hypotheses</code></pre>
<p>Fit: aov(formula = yield ~ 0 + trt, data = n_source_timing)</p>
<p>Linear Hypotheses:
Estimate Std. Error t value Pr(&gt;|t|)<br />
1 == 0 10.220 2.967 3.444 0.00243 **</p>
<p>Signif. codes: 0 ‘<em><strong>’ 0.001 ‘</strong>’ 0.01 ‘</em>’ 0.05 ‘.’ 0.1 ‘ ’ 1
(Adjusted p values reported – single-step method)</p>
<p>attr(,“class”)
[1] “group”</p>
</div>
</div>
</div>
<div id="exercise-means-tables" class="section level2" number="8.10">
<h2><span class="header-section-number">8.10</span> Exercise: Means Tables</h2>
<p>Rarely have I seen a statistics course that addresses how to present your summarise your data for others to use. Sure, you learn about tests and P-values. But how do you summarise your data for a report publication or presentation? I don’t want to go too deep in the weeds, but I want you to be aware how to organize your data, and how R can be used to present it.</p>
<div id="case-study-corn-nitrogen-source-and-timing" class="section level3" number="8.10.1">
<h3><span class="header-section-number">8.10.1</span> Case Study: Corn Nitrogen Source and Timing</h3>
<p>This is the same dataset we used for practice in the linear contrasts exercise. Seven N fertilizer treatments (6 combinations of source and timing, plus one unfertilized control) were tested in a randomized complete block trial near Whitehouse, Ohio.</p>
<div class="sourceCode" id="cb406"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb406-1"><a href="means-separation-and-data-presentation.html#cb406-1" aria-hidden="true" tabindex="-1"></a>corn_n <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data-unit-8/exercise_data/corn_nitrogen_source_timing.csv&quot;</span>)</span>
<span id="cb406-2"><a href="means-separation-and-data-presentation.html#cb406-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(corn_n)</span></code></pre></div>
<pre><code>##           trt    yield block  B      Error
## 1     control 147.9109    B1 -4  3.9108674
## 2      aa_pre 178.1029    B1 -4 -1.8970790
## 3     aa_post 173.5400    B1 -4  5.5400054
## 4 aa_pre_post 184.2516    B1 -4 -2.7483991
## 5     uan_pre 170.4471    B1 -4 -0.5529385
## 6    uan_post 165.4751    B1 -4 -1.5249071</code></pre>
<div id="calculating-means" class="section level4" number="8.10.1.1">
<h4><span class="header-section-number">8.10.1.1</span> Calculating Means</h4>
<p>We can quicly calculate means using the LSD.test() function from the <em>agricolae</em> package.</p>
<div class="sourceCode" id="cb408"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb408-1"><a href="means-separation-and-data-presentation.html#cb408-1" aria-hidden="true" tabindex="-1"></a>corn_n_model <span class="ot">=</span> <span class="fu">aov</span>(yield <span class="sc">~</span> block <span class="sc">+</span> trt, <span class="at">data=</span>corn_n)</span>
<span id="cb408-2"><a href="means-separation-and-data-presentation.html#cb408-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(corn_n_model)</span></code></pre></div>
<pre><code>##             Df Sum Sq Mean Sq F value   Pr(&gt;F)    
## block        3    283    94.3   2.414      0.1    
## trt          6   3421   570.1  14.591 4.81e-06 ***
## Residuals   18    703    39.1                     
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb410"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb410-1"><a href="means-separation-and-data-presentation.html#cb410-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(agricolae)</span>
<span id="cb410-2"><a href="means-separation-and-data-presentation.html#cb410-2" aria-hidden="true" tabindex="-1"></a>lsd <span class="ot">=</span> <span class="fu">LSD.test</span>(corn_n_model, <span class="st">&quot;trt&quot;</span>)</span>
<span id="cb410-3"><a href="means-separation-and-data-presentation.html#cb410-3" aria-hidden="true" tabindex="-1"></a>lsd</span></code></pre></div>
<pre><code>## $statistics
##    MSerror Df     Mean       CV  t.value      LSD
##   39.07036 18 174.8955 3.573922 2.100922 9.285785
## 
## $parameters
##         test p.ajusted name.t ntr alpha
##   Fisher-LSD      none    trt   7  0.05
## 
## $means
##                 yield       std r      LCL      UCL      Min      Max      Q25
## aa_post      174.7789  3.277009 4 168.2129 181.3449 170.7412 177.8955 172.8403
## aa_pre       181.7686  7.217879 4 175.2026 188.3347 174.6550 191.2908 177.2409
## aa_pre_post  191.2750  5.197261 4 184.7089 197.8410 184.2516 196.7370 189.6756
## control      153.2038  7.242153 4 146.6378 159.7699 147.6271 163.0634 147.8399
## uan_post     167.9677  1.734666 4 161.4016 174.5337 165.4751 169.4693 167.5775
## uan_pre      175.9014 12.448641 4 169.3354 182.4675 162.5066 191.5746 168.4619
## uan_pre_post 179.3731  5.336127 4 172.8070 185.9391 171.7472 184.1562 178.2194
##                   Q50      Q75
## aa_post      175.2394 177.1780
## aa_pre       180.5644 185.0921
## aa_pre_post  192.0556 193.6550
## control      151.0624 156.4262
## uan_post     168.4631 168.8533
## uan_pre      174.7623 182.2018
## uan_pre_post 180.7945 181.9482
## 
## $comparison
## NULL
## 
## $groups
##                 yield groups
## aa_pre_post  191.2750      a
## aa_pre       181.7686      b
## uan_pre_post 179.3731      b
## uan_pre      175.9014     bc
## aa_post      174.7789     bc
## uan_post     167.9677      c
## control      153.2038      d
## 
## attr(,&quot;class&quot;)
## [1] &quot;group&quot;</code></pre>
<p>We can create a separate table with just the $groups section:</p>
<div class="sourceCode" id="cb412"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb412-1"><a href="means-separation-and-data-presentation.html#cb412-1" aria-hidden="true" tabindex="-1"></a>trt_means<span class="ot">=</span>lsd<span class="sc">$</span>groups</span>
<span id="cb412-2"><a href="means-separation-and-data-presentation.html#cb412-2" aria-hidden="true" tabindex="-1"></a>trt_means</span></code></pre></div>
<pre><code>##                 yield groups
## aa_pre_post  191.2750      a
## aa_pre       181.7686      b
## uan_pre_post 179.3731      b
## uan_pre      175.9014     bc
## aa_post      174.7789     bc
## uan_post     167.9677      c
## control      153.2038      d</code></pre>
<p>One thing we need to fix are the row_names. These are our treatment names and we want them in a column for our final table. The <em>rownames_to_column()</em> argument quickly fixes this (and is a trick it took me quite a while to figure out!). The rownames_to_column function takes one argument, <em>vars = ""</em>, which tells R what to name the new column.</p>
<div class="sourceCode" id="cb414"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb414-1"><a href="means-separation-and-data-presentation.html#cb414-1" aria-hidden="true" tabindex="-1"></a>table_w_trt_column <span class="ot">=</span> trt_means <span class="sc">%&gt;%</span></span>
<span id="cb414-2"><a href="means-separation-and-data-presentation.html#cb414-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">&quot;trt&quot;</span>)</span>
<span id="cb414-3"><a href="means-separation-and-data-presentation.html#cb414-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb414-4"><a href="means-separation-and-data-presentation.html#cb414-4" aria-hidden="true" tabindex="-1"></a>table_w_trt_column</span></code></pre></div>
<pre><code>##            trt    yield groups
## 1  aa_pre_post 191.2750      a
## 2       aa_pre 181.7686      b
## 3 uan_pre_post 179.3731      b
## 4      uan_pre 175.9014     bc
## 5      aa_post 174.7789     bc
## 6     uan_post 167.9677      c
## 7      control 153.2038      d</code></pre>
</div>
<div id="additional-statistics-to-include" class="section level4" number="8.10.1.2">
<h4><span class="header-section-number">8.10.1.2</span> Additional Statistics to Include</h4>
<p>It would be good to add some additional statistics to this, particularly the standard error of the difference and the number of replicatons. We can calculate the standard error of the difference from the Error Mean Square given in the LSD output. First, let’s start by retrieving the error mean square from the $statistics section of our lsd output. This is a case where we can use multiple dollar signs to drill down through the list and dataframe within the list, to get to the individual statistic.</p>
<div class="sourceCode" id="cb416"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb416-1"><a href="means-separation-and-data-presentation.html#cb416-1" aria-hidden="true" tabindex="-1"></a>ems <span class="ot">=</span> lsd<span class="sc">$</span>statistics<span class="sc">$</span>MSerror</span>
<span id="cb416-2"><a href="means-separation-and-data-presentation.html#cb416-2" aria-hidden="true" tabindex="-1"></a>ems</span></code></pre></div>
<pre><code>## [1] 39.07036</code></pre>
<p>Here is another important trick. Analyses of variance and LSD outputs rarely report the standard error of the difference. But we can quickly calculate it if we know the error mean square and the number of replicates:</p>
<p>SED = sqrt(2 *EMS / r)</p>
<p>Where r is the number of replicates. We will set r as a variable and call it into the equation (that way we can call it into our table later as well.)</p>
<div class="sourceCode" id="cb418"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb418-1"><a href="means-separation-and-data-presentation.html#cb418-1" aria-hidden="true" tabindex="-1"></a>r <span class="ot">=</span> <span class="dv">4</span></span>
<span id="cb418-2"><a href="means-separation-and-data-presentation.html#cb418-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb418-3"><a href="means-separation-and-data-presentation.html#cb418-3" aria-hidden="true" tabindex="-1"></a>sed <span class="ot">=</span> <span class="fu">sqrt</span>(<span class="dv">2</span> <span class="sc">*</span> ems<span class="sc">/</span>r)</span>
<span id="cb418-4"><a href="means-separation-and-data-presentation.html#cb418-4" aria-hidden="true" tabindex="-1"></a>sed</span></code></pre></div>
<pre><code>## [1] 4.419862</code></pre>
<p>Now we can add the standard error of the difference and the number of replicates to the bottom of the column yield. We will put the name of the statistic in the “treatment” column and the value of the statistic in the “yield” column. All we need to do is use the <em>add_row()</em> function and tell R what the value of treatment and yield are in this new row.</p>
<div class="sourceCode" id="cb420"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb420-1"><a href="means-separation-and-data-presentation.html#cb420-1" aria-hidden="true" tabindex="-1"></a>table_w_stats <span class="ot">=</span> table_w_trt_column <span class="sc">%&gt;%</span></span>
<span id="cb420-2"><a href="means-separation-and-data-presentation.html#cb420-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_row</span>(<span class="at">trt=</span><span class="st">&quot;SED&quot;</span>, <span class="at">yield=</span>sed) <span class="sc">%&gt;%</span></span>
<span id="cb420-3"><a href="means-separation-and-data-presentation.html#cb420-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_row</span>(<span class="at">trt=</span><span class="st">&quot;Number of Reps&quot;</span>, <span class="at">yield=</span>r)</span>
<span id="cb420-4"><a href="means-separation-and-data-presentation.html#cb420-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb420-5"><a href="means-separation-and-data-presentation.html#cb420-5" aria-hidden="true" tabindex="-1"></a>table_w_stats</span></code></pre></div>
<pre><code>##              trt      yield groups
## 1    aa_pre_post 191.274962      a
## 2         aa_pre 181.768650      b
## 3   uan_pre_post 179.373088      b
## 4        uan_pre 175.901428     bc
## 5        aa_post 174.778906     bc
## 6       uan_post 167.967676      c
## 7        control 153.203813      d
## 8            SED   4.419862   &lt;NA&gt;
## 9 Number of Reps   4.000000   &lt;NA&gt;</code></pre>
</div>
<div id="making-a-presentation-quality-table" class="section level4" number="8.10.1.3">
<h4><span class="header-section-number">8.10.1.3</span> Making a Presentation-Quality Table</h4>
<p>At this point, our table content is complete. If we are going to include this in a document or presentation, however, we need to spruce it up a little. While there are multiple packages to create tables in R, I think that the <em>gt</em> table is one you are most likely to use.</p>
<p>We’ll start loading the gt package, then using the <em>gt()</em> function to flow in the means table we have created above. This will create a preliminary table.</p>
<div class="sourceCode" id="cb422"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb422-1"><a href="means-separation-and-data-presentation.html#cb422-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gt)</span>
<span id="cb422-2"><a href="means-separation-and-data-presentation.html#cb422-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb422-3"><a href="means-separation-and-data-presentation.html#cb422-3" aria-hidden="true" tabindex="-1"></a>gt_tbl <span class="ot">=</span> <span class="fu">gt</span>(table_w_stats)</span>
<span id="cb422-4"><a href="means-separation-and-data-presentation.html#cb422-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb422-5"><a href="means-separation-and-data-presentation.html#cb422-5" aria-hidden="true" tabindex="-1"></a>gt_tbl</span></code></pre></div>
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#fympmylzdb .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#fympmylzdb .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#fympmylzdb .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#fympmylzdb .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 4px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#fympmylzdb .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#fympmylzdb .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#fympmylzdb .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#fympmylzdb .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#fympmylzdb .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#fympmylzdb .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#fympmylzdb .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#fympmylzdb .gt_group_heading {
  padding: 8px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#fympmylzdb .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#fympmylzdb .gt_from_md > :first-child {
  margin-top: 0;
}

#fympmylzdb .gt_from_md > :last-child {
  margin-bottom: 0;
}

#fympmylzdb .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#fympmylzdb .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 12px;
}

#fympmylzdb .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#fympmylzdb .gt_first_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
}

#fympmylzdb .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#fympmylzdb .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#fympmylzdb .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#fympmylzdb .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#fympmylzdb .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#fympmylzdb .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding: 4px;
}

#fympmylzdb .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#fympmylzdb .gt_sourcenote {
  font-size: 90%;
  padding: 4px;
}

#fympmylzdb .gt_left {
  text-align: left;
}

#fympmylzdb .gt_center {
  text-align: center;
}

#fympmylzdb .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#fympmylzdb .gt_font_normal {
  font-weight: normal;
}

#fympmylzdb .gt_font_bold {
  font-weight: bold;
}

#fympmylzdb .gt_font_italic {
  font-style: italic;
}

#fympmylzdb .gt_super {
  font-size: 65%;
}

#fympmylzdb .gt_footnote_marks {
  font-style: italic;
  font-size: 65%;
}
</style>
<div id="fympmylzdb" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;"><table class="gt_table">
  
  <thead class="gt_col_headings">
    <tr>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">trt</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1">yield</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">groups</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr>
      <td class="gt_row gt_left">aa_pre_post</td>
      <td class="gt_row gt_right">191.274962</td>
      <td class="gt_row gt_left">a</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">aa_pre</td>
      <td class="gt_row gt_right">181.768650</td>
      <td class="gt_row gt_left">b</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">uan_pre_post</td>
      <td class="gt_row gt_right">179.373088</td>
      <td class="gt_row gt_left">b</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">uan_pre</td>
      <td class="gt_row gt_right">175.901428</td>
      <td class="gt_row gt_left">bc</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">aa_post</td>
      <td class="gt_row gt_right">174.778906</td>
      <td class="gt_row gt_left">bc</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">uan_post</td>
      <td class="gt_row gt_right">167.967676</td>
      <td class="gt_row gt_left">c</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">control</td>
      <td class="gt_row gt_right">153.203813</td>
      <td class="gt_row gt_left">d</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">SED</td>
      <td class="gt_row gt_right">4.419862</td>
      <td class="gt_row gt_left">NA</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">Number of Reps</td>
      <td class="gt_row gt_right">4.000000</td>
      <td class="gt_row gt_left">NA</td>
    </tr>
  </tbody>
  
  
</table></div>
<div id="choose-decimal-places" class="section level5" number="8.10.1.3.1">
<h5><span class="header-section-number">8.10.1.3.1</span> Choose Decimal Places</h5>
<p>One of the first things we notice is that the means in our middle column, yield, has have way too many decimal places. We can correct this with the <em>fmt_number()</em> function. This function takes two arguments. First, “columns - vars(yield) tells R which column to format. The second argument,”decimals=1", tells the number of decimal places to round to.</p>
<div class="sourceCode" id="cb423"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb423-1"><a href="means-separation-and-data-presentation.html#cb423-1" aria-hidden="true" tabindex="-1"></a>gt_tbl <span class="sc">%&gt;%</span></span>
<span id="cb423-2"><a href="means-separation-and-data-presentation.html#cb423-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fmt_number</span>(<span class="at">columns =</span> <span class="fu">vars</span>(yield), <span class="at">decimals =</span> <span class="dv">1</span>) </span></code></pre></div>
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#vusjmjfsqx .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#vusjmjfsqx .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#vusjmjfsqx .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#vusjmjfsqx .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 4px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#vusjmjfsqx .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#vusjmjfsqx .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#vusjmjfsqx .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#vusjmjfsqx .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#vusjmjfsqx .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#vusjmjfsqx .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#vusjmjfsqx .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#vusjmjfsqx .gt_group_heading {
  padding: 8px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#vusjmjfsqx .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#vusjmjfsqx .gt_from_md > :first-child {
  margin-top: 0;
}

#vusjmjfsqx .gt_from_md > :last-child {
  margin-bottom: 0;
}

#vusjmjfsqx .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#vusjmjfsqx .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 12px;
}

#vusjmjfsqx .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#vusjmjfsqx .gt_first_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
}

#vusjmjfsqx .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#vusjmjfsqx .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#vusjmjfsqx .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#vusjmjfsqx .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#vusjmjfsqx .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#vusjmjfsqx .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding: 4px;
}

#vusjmjfsqx .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#vusjmjfsqx .gt_sourcenote {
  font-size: 90%;
  padding: 4px;
}

#vusjmjfsqx .gt_left {
  text-align: left;
}

#vusjmjfsqx .gt_center {
  text-align: center;
}

#vusjmjfsqx .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#vusjmjfsqx .gt_font_normal {
  font-weight: normal;
}

#vusjmjfsqx .gt_font_bold {
  font-weight: bold;
}

#vusjmjfsqx .gt_font_italic {
  font-style: italic;
}

#vusjmjfsqx .gt_super {
  font-size: 65%;
}

#vusjmjfsqx .gt_footnote_marks {
  font-style: italic;
  font-size: 65%;
}
</style>
<div id="vusjmjfsqx" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;"><table class="gt_table">
  
  <thead class="gt_col_headings">
    <tr>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">trt</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1">yield</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">groups</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr>
      <td class="gt_row gt_left">aa_pre_post</td>
      <td class="gt_row gt_right">191.3</td>
      <td class="gt_row gt_left">a</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">aa_pre</td>
      <td class="gt_row gt_right">181.8</td>
      <td class="gt_row gt_left">b</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">uan_pre_post</td>
      <td class="gt_row gt_right">179.4</td>
      <td class="gt_row gt_left">b</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">uan_pre</td>
      <td class="gt_row gt_right">175.9</td>
      <td class="gt_row gt_left">bc</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">aa_post</td>
      <td class="gt_row gt_right">174.8</td>
      <td class="gt_row gt_left">bc</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">uan_post</td>
      <td class="gt_row gt_right">168.0</td>
      <td class="gt_row gt_left">c</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">control</td>
      <td class="gt_row gt_right">153.2</td>
      <td class="gt_row gt_left">d</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">SED</td>
      <td class="gt_row gt_right">4.4</td>
      <td class="gt_row gt_left">NA</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">Number of Reps</td>
      <td class="gt_row gt_right">4.0</td>
      <td class="gt_row gt_left">NA</td>
    </tr>
  </tbody>
  
  
</table></div>
</div>
<div id="adjust-table-width" class="section level5" number="8.10.1.3.2">
<h5><span class="header-section-number">8.10.1.3.2</span> Adjust Table Width</h5>
<p>Our table looks a little skinny, so let’s adjust the width. We can do this with the cols_width function. The “everything() argument tells R to set each of the columns to the same width.”~200px" sets the width at 200px.</p>
<div class="sourceCode" id="cb424"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb424-1"><a href="means-separation-and-data-presentation.html#cb424-1" aria-hidden="true" tabindex="-1"></a>gt_tbl <span class="sc">%&gt;%</span></span>
<span id="cb424-2"><a href="means-separation-and-data-presentation.html#cb424-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fmt_number</span>(<span class="at">columns =</span> <span class="fu">vars</span>(yield), <span class="at">decimals =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb424-3"><a href="means-separation-and-data-presentation.html#cb424-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_width</span>(<span class="fu">everything</span>() <span class="sc">~</span> <span class="st">&quot;200px&quot;</span>)  </span></code></pre></div>
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#cvlwthkjkt .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#cvlwthkjkt .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#cvlwthkjkt .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#cvlwthkjkt .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 4px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#cvlwthkjkt .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#cvlwthkjkt .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#cvlwthkjkt .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#cvlwthkjkt .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#cvlwthkjkt .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#cvlwthkjkt .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#cvlwthkjkt .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#cvlwthkjkt .gt_group_heading {
  padding: 8px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#cvlwthkjkt .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#cvlwthkjkt .gt_from_md > :first-child {
  margin-top: 0;
}

#cvlwthkjkt .gt_from_md > :last-child {
  margin-bottom: 0;
}

#cvlwthkjkt .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#cvlwthkjkt .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 12px;
}

#cvlwthkjkt .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#cvlwthkjkt .gt_first_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
}

#cvlwthkjkt .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#cvlwthkjkt .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#cvlwthkjkt .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#cvlwthkjkt .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#cvlwthkjkt .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#cvlwthkjkt .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding: 4px;
}

#cvlwthkjkt .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#cvlwthkjkt .gt_sourcenote {
  font-size: 90%;
  padding: 4px;
}

#cvlwthkjkt .gt_left {
  text-align: left;
}

#cvlwthkjkt .gt_center {
  text-align: center;
}

#cvlwthkjkt .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#cvlwthkjkt .gt_font_normal {
  font-weight: normal;
}

#cvlwthkjkt .gt_font_bold {
  font-weight: bold;
}

#cvlwthkjkt .gt_font_italic {
  font-style: italic;
}

#cvlwthkjkt .gt_super {
  font-size: 65%;
}

#cvlwthkjkt .gt_footnote_marks {
  font-style: italic;
  font-size: 65%;
}
</style>
<div id="cvlwthkjkt" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;"><table class="gt_table" style="table-layout: fixed;; width: 0px">
  <colgroup>
    <col style="width:200px;"/>
    <col style="width:200px;"/>
    <col style="width:200px;"/>
  </colgroup>
  
  <thead class="gt_col_headings">
    <tr>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">trt</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1">yield</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">groups</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr>
      <td class="gt_row gt_left">aa_pre_post</td>
      <td class="gt_row gt_right">191.3</td>
      <td class="gt_row gt_left">a</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">aa_pre</td>
      <td class="gt_row gt_right">181.8</td>
      <td class="gt_row gt_left">b</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">uan_pre_post</td>
      <td class="gt_row gt_right">179.4</td>
      <td class="gt_row gt_left">b</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">uan_pre</td>
      <td class="gt_row gt_right">175.9</td>
      <td class="gt_row gt_left">bc</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">aa_post</td>
      <td class="gt_row gt_right">174.8</td>
      <td class="gt_row gt_left">bc</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">uan_post</td>
      <td class="gt_row gt_right">168.0</td>
      <td class="gt_row gt_left">c</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">control</td>
      <td class="gt_row gt_right">153.2</td>
      <td class="gt_row gt_left">d</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">SED</td>
      <td class="gt_row gt_right">4.4</td>
      <td class="gt_row gt_left">NA</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">Number of Reps</td>
      <td class="gt_row gt_right">4.0</td>
      <td class="gt_row gt_left">NA</td>
    </tr>
  </tbody>
  
  
</table></div>
</div>
<div id="align-columns" class="section level5" number="8.10.1.3.3">
<h5><span class="header-section-number">8.10.1.3.3</span> Align Columns</h5>
<p>We use the <em>cols_align()</em> function to align our columns. The first argument to this function, “cols_align =,” specifies whether the column(s) should be aligned to the left, center, or right. The second argument, “columns = vars()” tells R which columns to which the alignment should be applied.</p>
<div class="sourceCode" id="cb425"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb425-1"><a href="means-separation-and-data-presentation.html#cb425-1" aria-hidden="true" tabindex="-1"></a>gt_tbl <span class="sc">%&gt;%</span></span>
<span id="cb425-2"><a href="means-separation-and-data-presentation.html#cb425-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fmt_number</span>(<span class="at">columns =</span> <span class="fu">vars</span>(yield), <span class="at">decimals =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb425-3"><a href="means-separation-and-data-presentation.html#cb425-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_width</span>(<span class="fu">everything</span>() <span class="sc">~</span> <span class="st">&quot;200px&quot;</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb425-4"><a href="means-separation-and-data-presentation.html#cb425-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_align</span>(<span class="at">align =</span> <span class="st">&quot;left&quot;</span>, <span class="at">columns =</span> <span class="fu">vars</span>(trt)) <span class="sc">%&gt;%</span></span>
<span id="cb425-5"><a href="means-separation-and-data-presentation.html#cb425-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_align</span>(<span class="at">align =</span> <span class="st">&quot;center&quot;</span>, <span class="at">columns =</span> <span class="fu">vars</span>(yield, groups)) </span></code></pre></div>
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#vuecqqgzrj .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#vuecqqgzrj .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#vuecqqgzrj .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#vuecqqgzrj .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 4px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#vuecqqgzrj .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#vuecqqgzrj .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#vuecqqgzrj .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#vuecqqgzrj .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#vuecqqgzrj .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#vuecqqgzrj .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#vuecqqgzrj .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#vuecqqgzrj .gt_group_heading {
  padding: 8px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#vuecqqgzrj .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#vuecqqgzrj .gt_from_md > :first-child {
  margin-top: 0;
}

#vuecqqgzrj .gt_from_md > :last-child {
  margin-bottom: 0;
}

#vuecqqgzrj .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#vuecqqgzrj .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 12px;
}

#vuecqqgzrj .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#vuecqqgzrj .gt_first_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
}

#vuecqqgzrj .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#vuecqqgzrj .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#vuecqqgzrj .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#vuecqqgzrj .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#vuecqqgzrj .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#vuecqqgzrj .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding: 4px;
}

#vuecqqgzrj .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#vuecqqgzrj .gt_sourcenote {
  font-size: 90%;
  padding: 4px;
}

#vuecqqgzrj .gt_left {
  text-align: left;
}

#vuecqqgzrj .gt_center {
  text-align: center;
}

#vuecqqgzrj .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#vuecqqgzrj .gt_font_normal {
  font-weight: normal;
}

#vuecqqgzrj .gt_font_bold {
  font-weight: bold;
}

#vuecqqgzrj .gt_font_italic {
  font-style: italic;
}

#vuecqqgzrj .gt_super {
  font-size: 65%;
}

#vuecqqgzrj .gt_footnote_marks {
  font-style: italic;
  font-size: 65%;
}
</style>
<div id="vuecqqgzrj" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;"><table class="gt_table" style="table-layout: fixed;; width: 0px">
  <colgroup>
    <col style="width:200px;"/>
    <col style="width:200px;"/>
    <col style="width:200px;"/>
  </colgroup>
  
  <thead class="gt_col_headings">
    <tr>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">trt</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">yield</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">groups</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr>
      <td class="gt_row gt_left">aa_pre_post</td>
      <td class="gt_row gt_center">191.3</td>
      <td class="gt_row gt_center">a</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">aa_pre</td>
      <td class="gt_row gt_center">181.8</td>
      <td class="gt_row gt_center">b</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">uan_pre_post</td>
      <td class="gt_row gt_center">179.4</td>
      <td class="gt_row gt_center">b</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">uan_pre</td>
      <td class="gt_row gt_center">175.9</td>
      <td class="gt_row gt_center">bc</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">aa_post</td>
      <td class="gt_row gt_center">174.8</td>
      <td class="gt_row gt_center">bc</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">uan_post</td>
      <td class="gt_row gt_center">168.0</td>
      <td class="gt_row gt_center">c</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">control</td>
      <td class="gt_row gt_center">153.2</td>
      <td class="gt_row gt_center">d</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">SED</td>
      <td class="gt_row gt_center">4.4</td>
      <td class="gt_row gt_center">NA</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">Number of Reps</td>
      <td class="gt_row gt_center">4.0</td>
      <td class="gt_row gt_center">NA</td>
    </tr>
  </tbody>
  
  
</table></div>
</div>
<div id="assign-formal-column-headers" class="section level5" number="8.10.1.3.4">
<h5><span class="header-section-number">8.10.1.3.4</span> Assign Formal Column Headers</h5>
<p>While it is great to use simple column names when we are processing our data, when we go to present a table we should use more formal column headings. We use the function <em>cols_label()</em>. We specify the current column name and desired name in our argument. For example, the current name of the left column is trt, but we want to rename it “Treatment.”</p>
<p>The second argument is a little fancier. We want to rename our column so that it includes “Yield” and the measurement units, “bu/acre.” But we want to use the more scientifically accepted practice of foregoing the forward-slash “/” and instead write “bu acre-1” where the “-1” is in superscript.</p>
<p>To get a superscript “-1,” we need to use a couple of html arguments: “<sup>” specifices the following text is superscript; </sup> ends the superscript text.</p>
<p>So our text argument is “Yield (bu acre<sup>-1</sup>).” To have R render it in html, however, we need to place our text string inside the <em>html()</em> function.</p>
<div class="sourceCode" id="cb426"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb426-1"><a href="means-separation-and-data-presentation.html#cb426-1" aria-hidden="true" tabindex="-1"></a>gt_tbl <span class="sc">%&gt;%</span></span>
<span id="cb426-2"><a href="means-separation-and-data-presentation.html#cb426-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fmt_number</span>(<span class="at">columns =</span> <span class="fu">vars</span>(yield), <span class="at">decimals =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb426-3"><a href="means-separation-and-data-presentation.html#cb426-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_width</span>(<span class="fu">everything</span>() <span class="sc">~</span> <span class="st">&quot;200px&quot;</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb426-4"><a href="means-separation-and-data-presentation.html#cb426-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_align</span>(<span class="at">align =</span> <span class="st">&quot;left&quot;</span>, <span class="at">columns =</span> <span class="fu">vars</span>(trt)) <span class="sc">%&gt;%</span></span>
<span id="cb426-5"><a href="means-separation-and-data-presentation.html#cb426-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_align</span>(<span class="at">align =</span> <span class="st">&quot;center&quot;</span>, <span class="at">columns =</span> <span class="fu">vars</span>(yield, groups)) <span class="sc">%&gt;%</span></span>
<span id="cb426-6"><a href="means-separation-and-data-presentation.html#cb426-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_label</span>(<span class="at">trt =</span> <span class="st">&quot;Treatment&quot;</span>,</span>
<span id="cb426-7"><a href="means-separation-and-data-presentation.html#cb426-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">yield =</span> <span class="fu">html</span>(<span class="st">&quot;Yield (bu acre&lt;sup&gt;-1&lt;/sup&gt;)&quot;</span>),</span>
<span id="cb426-8"><a href="means-separation-and-data-presentation.html#cb426-8" aria-hidden="true" tabindex="-1"></a>             <span class="at">groups =</span> <span class="st">&quot;LSD Groupings&quot;</span>) </span></code></pre></div>
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#dpeqtuvclb .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#dpeqtuvclb .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#dpeqtuvclb .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#dpeqtuvclb .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 4px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#dpeqtuvclb .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#dpeqtuvclb .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#dpeqtuvclb .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#dpeqtuvclb .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#dpeqtuvclb .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#dpeqtuvclb .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#dpeqtuvclb .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#dpeqtuvclb .gt_group_heading {
  padding: 8px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#dpeqtuvclb .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#dpeqtuvclb .gt_from_md > :first-child {
  margin-top: 0;
}

#dpeqtuvclb .gt_from_md > :last-child {
  margin-bottom: 0;
}

#dpeqtuvclb .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#dpeqtuvclb .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 12px;
}

#dpeqtuvclb .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#dpeqtuvclb .gt_first_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
}

#dpeqtuvclb .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#dpeqtuvclb .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#dpeqtuvclb .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#dpeqtuvclb .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#dpeqtuvclb .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#dpeqtuvclb .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding: 4px;
}

#dpeqtuvclb .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#dpeqtuvclb .gt_sourcenote {
  font-size: 90%;
  padding: 4px;
}

#dpeqtuvclb .gt_left {
  text-align: left;
}

#dpeqtuvclb .gt_center {
  text-align: center;
}

#dpeqtuvclb .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#dpeqtuvclb .gt_font_normal {
  font-weight: normal;
}

#dpeqtuvclb .gt_font_bold {
  font-weight: bold;
}

#dpeqtuvclb .gt_font_italic {
  font-style: italic;
}

#dpeqtuvclb .gt_super {
  font-size: 65%;
}

#dpeqtuvclb .gt_footnote_marks {
  font-style: italic;
  font-size: 65%;
}
</style>
<div id="dpeqtuvclb" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;"><table class="gt_table" style="table-layout: fixed;; width: 0px">
  <colgroup>
    <col style="width:200px;"/>
    <col style="width:200px;"/>
    <col style="width:200px;"/>
  </colgroup>
  
  <thead class="gt_col_headings">
    <tr>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">Treatment</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">Yield (bu acre<sup>-1</sup>)</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">LSD Groupings</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr>
      <td class="gt_row gt_left">aa_pre_post</td>
      <td class="gt_row gt_center">191.3</td>
      <td class="gt_row gt_center">a</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">aa_pre</td>
      <td class="gt_row gt_center">181.8</td>
      <td class="gt_row gt_center">b</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">uan_pre_post</td>
      <td class="gt_row gt_center">179.4</td>
      <td class="gt_row gt_center">b</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">uan_pre</td>
      <td class="gt_row gt_center">175.9</td>
      <td class="gt_row gt_center">bc</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">aa_post</td>
      <td class="gt_row gt_center">174.8</td>
      <td class="gt_row gt_center">bc</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">uan_post</td>
      <td class="gt_row gt_center">168.0</td>
      <td class="gt_row gt_center">c</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">control</td>
      <td class="gt_row gt_center">153.2</td>
      <td class="gt_row gt_center">d</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">SED</td>
      <td class="gt_row gt_center">4.4</td>
      <td class="gt_row gt_center">NA</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">Number of Reps</td>
      <td class="gt_row gt_center">4.0</td>
      <td class="gt_row gt_center">NA</td>
    </tr>
  </tbody>
  
  
</table></div>
</div>
<div id="table-aesthetics" class="section level5" number="8.10.1.3.5">
<h5><span class="header-section-number">8.10.1.3.5</span> Table Aesthetics</h5>
<p>At this point, our table is essentially complete. But we would like tweak a couple of things. The first thing we want to do is remove those “NA”s and just leave those cells blank. The <em>fmt_missing()</em> function will take care of that for us. The first argument, “columns = TRUE,” tells R to search the entire table for occurrences of “NA.” The second argument, missing_text = "“, specifies they be replaced with a blank (”").</p>
<div class="sourceCode" id="cb427"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb427-1"><a href="means-separation-and-data-presentation.html#cb427-1" aria-hidden="true" tabindex="-1"></a>gt_tbl <span class="sc">%&gt;%</span></span>
<span id="cb427-2"><a href="means-separation-and-data-presentation.html#cb427-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fmt_number</span>(<span class="at">columns =</span> <span class="fu">vars</span>(yield), <span class="at">decimals =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb427-3"><a href="means-separation-and-data-presentation.html#cb427-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_width</span>(<span class="fu">everything</span>() <span class="sc">~</span> <span class="st">&quot;200px&quot;</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb427-4"><a href="means-separation-and-data-presentation.html#cb427-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_align</span>(<span class="at">align =</span> <span class="st">&quot;left&quot;</span>, <span class="at">columns =</span> <span class="fu">vars</span>(trt)) <span class="sc">%&gt;%</span></span>
<span id="cb427-5"><a href="means-separation-and-data-presentation.html#cb427-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_align</span>(<span class="at">align =</span> <span class="st">&quot;center&quot;</span>, <span class="at">columns =</span> <span class="fu">vars</span>(yield, groups)) <span class="sc">%&gt;%</span></span>
<span id="cb427-6"><a href="means-separation-and-data-presentation.html#cb427-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_label</span>(<span class="at">trt =</span> <span class="st">&quot;Treatment&quot;</span>,</span>
<span id="cb427-7"><a href="means-separation-and-data-presentation.html#cb427-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">yield =</span> <span class="fu">html</span>(<span class="st">&quot;Yield (bu acre&lt;sup&gt;-1&lt;/sup&gt;)&quot;</span>),</span>
<span id="cb427-8"><a href="means-separation-and-data-presentation.html#cb427-8" aria-hidden="true" tabindex="-1"></a>             <span class="at">groups =</span> <span class="st">&quot;LSD Groupings&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb427-9"><a href="means-separation-and-data-presentation.html#cb427-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fmt_missing</span>(<span class="at">columns =</span> <span class="cn">TRUE</span>, <span class="at">missing_text =</span> <span class="st">&quot;&quot;</span>)</span></code></pre></div>
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#oibypgeqxs .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#oibypgeqxs .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#oibypgeqxs .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#oibypgeqxs .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 4px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#oibypgeqxs .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#oibypgeqxs .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#oibypgeqxs .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#oibypgeqxs .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#oibypgeqxs .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#oibypgeqxs .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#oibypgeqxs .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#oibypgeqxs .gt_group_heading {
  padding: 8px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#oibypgeqxs .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#oibypgeqxs .gt_from_md > :first-child {
  margin-top: 0;
}

#oibypgeqxs .gt_from_md > :last-child {
  margin-bottom: 0;
}

#oibypgeqxs .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#oibypgeqxs .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 12px;
}

#oibypgeqxs .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#oibypgeqxs .gt_first_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
}

#oibypgeqxs .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#oibypgeqxs .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#oibypgeqxs .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#oibypgeqxs .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#oibypgeqxs .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#oibypgeqxs .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding: 4px;
}

#oibypgeqxs .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#oibypgeqxs .gt_sourcenote {
  font-size: 90%;
  padding: 4px;
}

#oibypgeqxs .gt_left {
  text-align: left;
}

#oibypgeqxs .gt_center {
  text-align: center;
}

#oibypgeqxs .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#oibypgeqxs .gt_font_normal {
  font-weight: normal;
}

#oibypgeqxs .gt_font_bold {
  font-weight: bold;
}

#oibypgeqxs .gt_font_italic {
  font-style: italic;
}

#oibypgeqxs .gt_super {
  font-size: 65%;
}

#oibypgeqxs .gt_footnote_marks {
  font-style: italic;
  font-size: 65%;
}
</style>
<div id="oibypgeqxs" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;"><table class="gt_table" style="table-layout: fixed;; width: 0px">
  <colgroup>
    <col style="width:200px;"/>
    <col style="width:200px;"/>
    <col style="width:200px;"/>
  </colgroup>
  
  <thead class="gt_col_headings">
    <tr>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">Treatment</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">Yield (bu acre<sup>-1</sup>)</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">LSD Groupings</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr>
      <td class="gt_row gt_left">aa_pre_post</td>
      <td class="gt_row gt_center">191.3</td>
      <td class="gt_row gt_center">a</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">aa_pre</td>
      <td class="gt_row gt_center">181.8</td>
      <td class="gt_row gt_center">b</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">uan_pre_post</td>
      <td class="gt_row gt_center">179.4</td>
      <td class="gt_row gt_center">b</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">uan_pre</td>
      <td class="gt_row gt_center">175.9</td>
      <td class="gt_row gt_center">bc</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">aa_post</td>
      <td class="gt_row gt_center">174.8</td>
      <td class="gt_row gt_center">bc</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">uan_post</td>
      <td class="gt_row gt_center">168.0</td>
      <td class="gt_row gt_center">c</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">control</td>
      <td class="gt_row gt_center">153.2</td>
      <td class="gt_row gt_center">d</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">SED</td>
      <td class="gt_row gt_center">4.4</td>
      <td class="gt_row gt_center"></td>
    </tr>
    <tr>
      <td class="gt_row gt_left">Number of Reps</td>
      <td class="gt_row gt_center">4.0</td>
      <td class="gt_row gt_center"></td>
    </tr>
  </tbody>
  
  
</table></div>
<p>Finally, we would like to make a couple of the lines a little wider: the top and bottom lines of the table, as well as the line under the column labels. We can do this with the <em>tab_options()</em> function, which then allows us to format dozens of table aspects: lines, fonts, background colors, etc. We will use a table of “trio.border” arguments to widen those three lines.</p>
<div class="sourceCode" id="cb428"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb428-1"><a href="means-separation-and-data-presentation.html#cb428-1" aria-hidden="true" tabindex="-1"></a>gt_tbl <span class="sc">%&gt;%</span></span>
<span id="cb428-2"><a href="means-separation-and-data-presentation.html#cb428-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fmt_number</span>(<span class="at">columns =</span> <span class="fu">vars</span>(yield), <span class="at">decimals =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb428-3"><a href="means-separation-and-data-presentation.html#cb428-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_width</span>(<span class="fu">everything</span>() <span class="sc">~</span> <span class="st">&quot;200px&quot;</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb428-4"><a href="means-separation-and-data-presentation.html#cb428-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_align</span>(<span class="at">align =</span> <span class="st">&quot;left&quot;</span>, <span class="at">columns =</span> <span class="fu">vars</span>(trt)) <span class="sc">%&gt;%</span></span>
<span id="cb428-5"><a href="means-separation-and-data-presentation.html#cb428-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_align</span>(<span class="at">align =</span> <span class="st">&quot;center&quot;</span>, <span class="at">columns =</span> <span class="fu">vars</span>(yield, groups)) <span class="sc">%&gt;%</span></span>
<span id="cb428-6"><a href="means-separation-and-data-presentation.html#cb428-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_label</span>(<span class="at">trt =</span> <span class="st">&quot;Treatment&quot;</span>,</span>
<span id="cb428-7"><a href="means-separation-and-data-presentation.html#cb428-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">yield =</span> <span class="fu">html</span>(<span class="st">&quot;Yield (bu acre&lt;sup&gt;-1&lt;/sup&gt;)&quot;</span>),</span>
<span id="cb428-8"><a href="means-separation-and-data-presentation.html#cb428-8" aria-hidden="true" tabindex="-1"></a>             <span class="at">groups =</span> <span class="st">&quot;LSD Groupings&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb428-9"><a href="means-separation-and-data-presentation.html#cb428-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fmt_missing</span>(<span class="at">columns =</span> <span class="cn">TRUE</span>, <span class="at">missing_text =</span> <span class="st">&quot;&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb428-10"><a href="means-separation-and-data-presentation.html#cb428-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tab_options</span>(<span class="at">table.border.top.width =</span> <span class="dv">4</span>,</span>
<span id="cb428-11"><a href="means-separation-and-data-presentation.html#cb428-11" aria-hidden="true" tabindex="-1"></a>              <span class="at">table.border.bottom.width =</span> <span class="dv">4</span>,</span>
<span id="cb428-12"><a href="means-separation-and-data-presentation.html#cb428-12" aria-hidden="true" tabindex="-1"></a>              <span class="at">column_labels.border.bottom.width =</span> <span class="dv">4</span>)</span></code></pre></div>
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#aswquwzxlu .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 4px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 4px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#aswquwzxlu .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#aswquwzxlu .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#aswquwzxlu .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 4px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#aswquwzxlu .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#aswquwzxlu .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 4px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#aswquwzxlu .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#aswquwzxlu .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#aswquwzxlu .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#aswquwzxlu .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#aswquwzxlu .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 4px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#aswquwzxlu .gt_group_heading {
  padding: 8px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#aswquwzxlu .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#aswquwzxlu .gt_from_md > :first-child {
  margin-top: 0;
}

#aswquwzxlu .gt_from_md > :last-child {
  margin-bottom: 0;
}

#aswquwzxlu .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#aswquwzxlu .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 12px;
}

#aswquwzxlu .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#aswquwzxlu .gt_first_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
}

#aswquwzxlu .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#aswquwzxlu .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#aswquwzxlu .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#aswquwzxlu .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#aswquwzxlu .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#aswquwzxlu .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding: 4px;
}

#aswquwzxlu .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#aswquwzxlu .gt_sourcenote {
  font-size: 90%;
  padding: 4px;
}

#aswquwzxlu .gt_left {
  text-align: left;
}

#aswquwzxlu .gt_center {
  text-align: center;
}

#aswquwzxlu .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#aswquwzxlu .gt_font_normal {
  font-weight: normal;
}

#aswquwzxlu .gt_font_bold {
  font-weight: bold;
}

#aswquwzxlu .gt_font_italic {
  font-style: italic;
}

#aswquwzxlu .gt_super {
  font-size: 65%;
}

#aswquwzxlu .gt_footnote_marks {
  font-style: italic;
  font-size: 65%;
}
</style>
<div id="aswquwzxlu" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;"><table class="gt_table" style="table-layout: fixed;; width: 0px">
  <colgroup>
    <col style="width:200px;"/>
    <col style="width:200px;"/>
    <col style="width:200px;"/>
  </colgroup>
  
  <thead class="gt_col_headings">
    <tr>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">Treatment</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">Yield (bu acre<sup>-1</sup>)</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">LSD Groupings</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr>
      <td class="gt_row gt_left">aa_pre_post</td>
      <td class="gt_row gt_center">191.3</td>
      <td class="gt_row gt_center">a</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">aa_pre</td>
      <td class="gt_row gt_center">181.8</td>
      <td class="gt_row gt_center">b</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">uan_pre_post</td>
      <td class="gt_row gt_center">179.4</td>
      <td class="gt_row gt_center">b</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">uan_pre</td>
      <td class="gt_row gt_center">175.9</td>
      <td class="gt_row gt_center">bc</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">aa_post</td>
      <td class="gt_row gt_center">174.8</td>
      <td class="gt_row gt_center">bc</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">uan_post</td>
      <td class="gt_row gt_center">168.0</td>
      <td class="gt_row gt_center">c</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">control</td>
      <td class="gt_row gt_center">153.2</td>
      <td class="gt_row gt_center">d</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">SED</td>
      <td class="gt_row gt_center">4.4</td>
      <td class="gt_row gt_center"></td>
    </tr>
    <tr>
      <td class="gt_row gt_left">Number of Reps</td>
      <td class="gt_row gt_center">4.0</td>
      <td class="gt_row gt_center"></td>
    </tr>
  </tbody>
  
  
</table></div>
</div>
<div id="saving-your-table" class="section level5" number="8.10.1.3.6">
<h5><span class="header-section-number">8.10.1.3.6</span> Saving Your Table</h5>
<p>Finally, we can assign our table to an object, “corn_n_gt,” and save is as an .html file to use it in other documents.</p>
<div class="sourceCode" id="cb429"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb429-1"><a href="means-separation-and-data-presentation.html#cb429-1" aria-hidden="true" tabindex="-1"></a>corn_n_gt <span class="ot">=</span>  gt_tbl <span class="sc">%&gt;%</span></span>
<span id="cb429-2"><a href="means-separation-and-data-presentation.html#cb429-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fmt_number</span>(<span class="at">columns =</span> <span class="fu">vars</span>(yield), <span class="at">decimals =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb429-3"><a href="means-separation-and-data-presentation.html#cb429-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_width</span>(<span class="fu">everything</span>() <span class="sc">~</span> <span class="st">&quot;200px&quot;</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb429-4"><a href="means-separation-and-data-presentation.html#cb429-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_align</span>(<span class="at">align =</span> <span class="st">&quot;left&quot;</span>, <span class="at">columns =</span> <span class="fu">vars</span>(trt)) <span class="sc">%&gt;%</span></span>
<span id="cb429-5"><a href="means-separation-and-data-presentation.html#cb429-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_align</span>(<span class="at">align =</span> <span class="st">&quot;center&quot;</span>, <span class="at">columns =</span> <span class="fu">vars</span>(yield, groups)) <span class="sc">%&gt;%</span></span>
<span id="cb429-6"><a href="means-separation-and-data-presentation.html#cb429-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_label</span>(<span class="at">trt =</span> <span class="st">&quot;Treatment&quot;</span>,</span>
<span id="cb429-7"><a href="means-separation-and-data-presentation.html#cb429-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">yield =</span> <span class="fu">html</span>(<span class="st">&quot;Yield (bu acre&lt;sup&gt;-1&lt;/sup&gt;)&quot;</span>),</span>
<span id="cb429-8"><a href="means-separation-and-data-presentation.html#cb429-8" aria-hidden="true" tabindex="-1"></a>             <span class="at">groups =</span> <span class="st">&quot;LSD Groupings&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb429-9"><a href="means-separation-and-data-presentation.html#cb429-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fmt_missing</span>(<span class="at">columns =</span> <span class="cn">TRUE</span>, <span class="at">missing_text =</span> <span class="st">&quot;&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb429-10"><a href="means-separation-and-data-presentation.html#cb429-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tab_options</span>(<span class="at">table.border.top.width =</span> <span class="dv">4</span>,</span>
<span id="cb429-11"><a href="means-separation-and-data-presentation.html#cb429-11" aria-hidden="true" tabindex="-1"></a>              <span class="at">table.border.bottom.width =</span> <span class="dv">4</span>,</span>
<span id="cb429-12"><a href="means-separation-and-data-presentation.html#cb429-12" aria-hidden="true" tabindex="-1"></a>              <span class="at">column_labels.border.bottom.width =</span> <span class="dv">4</span>)</span>
<span id="cb429-13"><a href="means-separation-and-data-presentation.html#cb429-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb429-14"><a href="means-separation-and-data-presentation.html#cb429-14" aria-hidden="true" tabindex="-1"></a><span class="fu">gtsave</span>(corn_n_gt, <span class="st">&quot;corn_n_means_table.html&quot;</span>)</span></code></pre></div>
<p>We could go on and on, but that is enough for now, and should get you pretty far. More information on the gt() package can be found at: <a href="https://gt.rstudio.com/index.html" class="uri">https://gt.rstudio.com/index.html</a>.</p>
</div>
</div>
</div>
<div id="practice-canola" class="section level3" number="8.10.2">
<h3><span class="header-section-number">8.10.2</span> Practice: Canola</h3>
<div class="sourceCode" id="cb430"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb430-1"><a href="means-separation-and-data-presentation.html#cb430-1" aria-hidden="true" tabindex="-1"></a>canola <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data-unit-8/exercise_data/canola_gd.csv&quot;</span>)</span>
<span id="cb430-2"><a href="means-separation-and-data-presentation.html#cb430-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(canola)</span></code></pre></div>
<pre><code>##   block cultivar    yield
## 1     1      Bob 1508.424
## 2     2      Bob 1962.072
## 3     3      Bob 1364.861
## 4     4      Bob 1832.715
## 5     1    Donna 1212.190
## 6     2    Donna 1629.680</code></pre>
<p>Since the focus of this exercise is table-building, I will generate the initial lsd means table for you.</p>
<div class="sourceCode" id="cb432"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb432-1"><a href="means-separation-and-data-presentation.html#cb432-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(agricolae)</span>
<span id="cb432-2"><a href="means-separation-and-data-presentation.html#cb432-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb432-3"><a href="means-separation-and-data-presentation.html#cb432-3" aria-hidden="true" tabindex="-1"></a>canola_model <span class="ot">=</span> <span class="fu">aov</span>(yield <span class="sc">~</span> block <span class="sc">+</span> cultivar, <span class="at">data =</span> canola)</span>
<span id="cb432-4"><a href="means-separation-and-data-presentation.html#cb432-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb432-5"><a href="means-separation-and-data-presentation.html#cb432-5" aria-hidden="true" tabindex="-1"></a>lsd_canola <span class="ot">=</span> <span class="fu">LSD.test</span>(canola_model, <span class="st">&quot;cultivar&quot;</span>)</span>
<span id="cb432-6"><a href="means-separation-and-data-presentation.html#cb432-6" aria-hidden="true" tabindex="-1"></a>lsd_canola</span></code></pre></div>
<pre><code>## $statistics
##   MSerror Df     Mean       CV  t.value      LSD
##    109086 17 2296.983 14.37894 2.109816 492.7357
## 
## $parameters
##         test p.ajusted   name.t ntr alpha
##   Fisher-LSD      none cultivar   6  0.05
## 
## $means
##           yield      std r      LCL      UCL      Min      Max      Q25
## Bill   2296.268 326.6421 4 1947.851 2644.685 1865.903 2614.081 2142.888
## Bob    1667.018 277.4666 4 1318.601 2015.435 1364.861 1962.072 1472.533
## Donna  1833.282 514.0721 4 1484.865 2181.699 1212.190 2358.635 1525.308
## Jerry  2669.161 176.6376 4 2320.745 3017.578 2545.158 2923.794 2552.218
## Mickey 2406.509 263.5791 4 2058.092 2754.926 2130.413 2656.553 2207.614
## Phil   2909.658 275.1842 4 2561.242 3258.075 2565.683 3218.583 2776.782
##             Q50      Q75
## Bill   2352.545 2505.925
## Bob    1670.570 1865.055
## Donna  1881.152 2189.126
## Jerry  2603.847 2720.790
## Mickey 2419.535 2618.430
## Phil   2927.183 3060.060
## 
## $comparison
## NULL
## 
## $groups
##           yield groups
## Phil   2909.658      a
## Jerry  2669.161     ab
## Mickey 2406.509      b
## Bill   2296.268     bc
## Donna  1833.282     cd
## Bob    1667.018      d
## 
## attr(,&quot;class&quot;)
## [1] &quot;group&quot;</code></pre>
<p>First, extract the $groups table from the canola_lsd output.</p>
<div class="sourceCode" id="cb434"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb434-1"><a href="means-separation-and-data-presentation.html#cb434-1" aria-hidden="true" tabindex="-1"></a>canola_means <span class="ot">=</span> lsd_canola<span class="sc">$</span>groups</span></code></pre></div>
<p>Next, we need to convert our row names to a column</p>
<div class="sourceCode" id="cb435"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb435-1"><a href="means-separation-and-data-presentation.html#cb435-1" aria-hidden="true" tabindex="-1"></a>canola_table_w_trt_column <span class="ot">=</span> canola_means <span class="sc">%&gt;%</span></span>
<span id="cb435-2"><a href="means-separation-and-data-presentation.html#cb435-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;cultivar&quot;</span>)</span>
<span id="cb435-3"><a href="means-separation-and-data-presentation.html#cb435-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb435-4"><a href="means-separation-and-data-presentation.html#cb435-4" aria-hidden="true" tabindex="-1"></a>canola_table_w_trt_column</span></code></pre></div>
<pre><code>##   cultivar    yield groups
## 1     Phil 2909.658      a
## 2    Jerry 2669.161     ab
## 3   Mickey 2406.509      b
## 4     Bill 2296.268     bc
## 5    Donna 1833.282     cd
## 6      Bob 1667.018      d</code></pre>
<p>Finally, we need to calculate the SED and number of reps and add those to the table</p>
<div class="sourceCode" id="cb437"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb437-1"><a href="means-separation-and-data-presentation.html#cb437-1" aria-hidden="true" tabindex="-1"></a>r <span class="ot">=</span> <span class="dv">4</span></span>
<span id="cb437-2"><a href="means-separation-and-data-presentation.html#cb437-2" aria-hidden="true" tabindex="-1"></a>MSE <span class="ot">=</span> lsd_canola<span class="sc">$</span>statistics<span class="sc">$</span>MSerror</span>
<span id="cb437-3"><a href="means-separation-and-data-presentation.html#cb437-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb437-4"><a href="means-separation-and-data-presentation.html#cb437-4" aria-hidden="true" tabindex="-1"></a>SED <span class="ot">=</span> <span class="fu">sqrt</span>(<span class="dv">2</span><span class="sc">*</span>MSE<span class="sc">/</span>r)</span>
<span id="cb437-5"><a href="means-separation-and-data-presentation.html#cb437-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb437-6"><a href="means-separation-and-data-presentation.html#cb437-6" aria-hidden="true" tabindex="-1"></a>canola_table_w_stats <span class="ot">=</span> canola_table_w_trt_column <span class="sc">%&gt;%</span></span>
<span id="cb437-7"><a href="means-separation-and-data-presentation.html#cb437-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_row</span>(<span class="at">cultivar=</span><span class="st">&quot;SED&quot;</span>, <span class="at">yield=</span>SED, <span class="at">groups=</span><span class="st">&quot;&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb437-8"><a href="means-separation-and-data-presentation.html#cb437-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_row</span>(<span class="at">cultivar=</span><span class="st">&quot;Number of Reps&quot;</span>, <span class="at">yield=</span>r, <span class="at">groups=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb437-9"><a href="means-separation-and-data-presentation.html#cb437-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb437-10"><a href="means-separation-and-data-presentation.html#cb437-10" aria-hidden="true" tabindex="-1"></a>canola_table_w_stats</span></code></pre></div>
<pre><code>##         cultivar     yield groups
## 1           Phil 2909.6584      a
## 2          Jerry 2669.1615     ab
## 3         Mickey 2406.5090      b
## 4           Bill 2296.2682     bc
## 5          Donna 1833.2820     cd
## 6            Bob 1667.0181      d
## 7            SED  233.5444       
## 8 Number of Reps    4.0000</code></pre>
<p>At this point, the easiest way for you to make your table is to tweak the code below (this is why coding is so powerful).</p>
<p>You will need to:</p>
<ul>
<li>change the name of the data.frame from which you are building the table (line 1)</li>
<li>change the number of decimals in the fmt_number table to zero (line 2)</li>
<li>substitute “cultivar” for treatment wherever it occurs in the code (lines 3 and 5)</li>
<li>change the formal heading for the first column from “Treatment” to “Cultivar”</li>
</ul>
<div class="sourceCode" id="cb439"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb439-1"><a href="means-separation-and-data-presentation.html#cb439-1" aria-hidden="true" tabindex="-1"></a>canola_means_table <span class="ot">=</span> <span class="fu">gt</span>(canola_table_w_trt_column) <span class="sc">%&gt;%</span></span>
<span id="cb439-2"><a href="means-separation-and-data-presentation.html#cb439-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fmt_number</span>(<span class="at">columns =</span> <span class="fu">vars</span>(yield), <span class="at">decimals =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb439-3"><a href="means-separation-and-data-presentation.html#cb439-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_width</span>(<span class="fu">everything</span>() <span class="sc">~</span> <span class="st">&quot;200px&quot;</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb439-4"><a href="means-separation-and-data-presentation.html#cb439-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_align</span>(<span class="at">align =</span> <span class="st">&quot;left&quot;</span>, <span class="at">columns =</span> <span class="fu">vars</span>(cultivar)) <span class="sc">%&gt;%</span></span>
<span id="cb439-5"><a href="means-separation-and-data-presentation.html#cb439-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_align</span>(<span class="at">align =</span> <span class="st">&quot;center&quot;</span>, <span class="at">columns =</span> <span class="fu">vars</span>(yield, groups)) <span class="sc">%&gt;%</span></span>
<span id="cb439-6"><a href="means-separation-and-data-presentation.html#cb439-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_label</span>(<span class="at">cultivar =</span> <span class="st">&quot;Cultivar&quot;</span>,</span>
<span id="cb439-7"><a href="means-separation-and-data-presentation.html#cb439-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">yield =</span> <span class="fu">html</span>(<span class="st">&quot;Yield (bu acre&lt;sup&gt;-1&lt;/sup&gt;)&quot;</span>),</span>
<span id="cb439-8"><a href="means-separation-and-data-presentation.html#cb439-8" aria-hidden="true" tabindex="-1"></a>             <span class="at">groups =</span> <span class="st">&quot;LSD Groupings&quot;</span>) </span>
<span id="cb439-9"><a href="means-separation-and-data-presentation.html#cb439-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb439-10"><a href="means-separation-and-data-presentation.html#cb439-10" aria-hidden="true" tabindex="-1"></a>canola_means_table</span></code></pre></div>
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#olbrwxgqaz .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#olbrwxgqaz .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#olbrwxgqaz .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#olbrwxgqaz .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 4px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#olbrwxgqaz .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#olbrwxgqaz .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#olbrwxgqaz .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#olbrwxgqaz .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#olbrwxgqaz .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#olbrwxgqaz .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#olbrwxgqaz .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#olbrwxgqaz .gt_group_heading {
  padding: 8px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#olbrwxgqaz .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#olbrwxgqaz .gt_from_md > :first-child {
  margin-top: 0;
}

#olbrwxgqaz .gt_from_md > :last-child {
  margin-bottom: 0;
}

#olbrwxgqaz .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#olbrwxgqaz .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 12px;
}

#olbrwxgqaz .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#olbrwxgqaz .gt_first_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
}

#olbrwxgqaz .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#olbrwxgqaz .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#olbrwxgqaz .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#olbrwxgqaz .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#olbrwxgqaz .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#olbrwxgqaz .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding: 4px;
}

#olbrwxgqaz .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#olbrwxgqaz .gt_sourcenote {
  font-size: 90%;
  padding: 4px;
}

#olbrwxgqaz .gt_left {
  text-align: left;
}

#olbrwxgqaz .gt_center {
  text-align: center;
}

#olbrwxgqaz .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#olbrwxgqaz .gt_font_normal {
  font-weight: normal;
}

#olbrwxgqaz .gt_font_bold {
  font-weight: bold;
}

#olbrwxgqaz .gt_font_italic {
  font-style: italic;
}

#olbrwxgqaz .gt_super {
  font-size: 65%;
}

#olbrwxgqaz .gt_footnote_marks {
  font-style: italic;
  font-size: 65%;
}
</style>
<div id="olbrwxgqaz" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;"><table class="gt_table" style="table-layout: fixed;; width: 0px">
  <colgroup>
    <col style="width:200px;"/>
    <col style="width:200px;"/>
    <col style="width:200px;"/>
  </colgroup>
  
  <thead class="gt_col_headings">
    <tr>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">Cultivar</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">Yield (bu acre<sup>-1</sup>)</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">LSD Groupings</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr>
      <td class="gt_row gt_left">Phil</td>
      <td class="gt_row gt_center">2,909.7</td>
      <td class="gt_row gt_center">a</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">Jerry</td>
      <td class="gt_row gt_center">2,669.2</td>
      <td class="gt_row gt_center">ab</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">Mickey</td>
      <td class="gt_row gt_center">2,406.5</td>
      <td class="gt_row gt_center">b</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">Bill</td>
      <td class="gt_row gt_center">2,296.3</td>
      <td class="gt_row gt_center">bc</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">Donna</td>
      <td class="gt_row gt_center">1,833.3</td>
      <td class="gt_row gt_center">cd</td>
    </tr>
    <tr>
      <td class="gt_row gt_left">Bob</td>
      <td class="gt_row gt_center">1,667.0</td>
      <td class="gt_row gt_center">d</td>
    </tr>
  </tbody>
  
  
</table></div>
</div>
<div id="practice-broccoli-lsd" class="section level3" number="8.10.3">
<h3><span class="header-section-number">8.10.3</span> Practice: Broccoli LSD</h3>
<p>Common bean was grown with organic practices following four previous crops.</p>
<div class="sourceCode" id="cb440"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb440-1"><a href="means-separation-and-data-presentation.html#cb440-1" aria-hidden="true" tabindex="-1"></a>bean <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&quot;data-unit-8/exercise_data/common_bean.csv&quot;</span>)</span>
<span id="cb440-2"><a href="means-separation-and-data-presentation.html#cb440-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(bean)</span></code></pre></div>
<pre><code>##   block        prev_crop fresh_weight
## 1     1    conv_broccoli     52.15924
## 2     1     green_manure     49.82191
## 3     1 organic_broccoli     36.80926
## 4     1           fallow     54.05342
## 5     2    conv_broccoli     52.79726
## 6     2           fallow     47.68364</code></pre>
<div id="create-the-lsd-groups-table" class="section level4" number="8.10.3.1">
<h4><span class="header-section-number">8.10.3.1</span> Create the LSD Groups table</h4>
<div class="sourceCode" id="cb442"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb442-1"><a href="means-separation-and-data-presentation.html#cb442-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(agricolae)</span>
<span id="cb442-2"><a href="means-separation-and-data-presentation.html#cb442-2" aria-hidden="true" tabindex="-1"></a><span class="co"># model statement</span></span>
<span id="cb442-3"><a href="means-separation-and-data-presentation.html#cb442-3" aria-hidden="true" tabindex="-1"></a><span class="co"># LSD.test</span></span>
<span id="cb442-4"><a href="means-separation-and-data-presentation.html#cb442-4" aria-hidden="true" tabindex="-1"></a><span class="co"># extract groups table from lsd output</span></span>
<span id="cb442-5"><a href="means-separation-and-data-presentation.html#cb442-5" aria-hidden="true" tabindex="-1"></a><span class="co"># use rownames_to_column to create new column with rownames</span></span></code></pre></div>
</div>
<div id="prepare-the-table-contents" class="section level4" number="8.10.3.2">
<h4><span class="header-section-number">8.10.3.2</span> Prepare the table contents</h4>
<div class="sourceCode" id="cb443"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb443-1"><a href="means-separation-and-data-presentation.html#cb443-1" aria-hidden="true" tabindex="-1"></a><span class="co"># r = number of reps</span></span>
<span id="cb443-2"><a href="means-separation-and-data-presentation.html#cb443-2" aria-hidden="true" tabindex="-1"></a><span class="co"># extract MSE from lsd output</span></span>
<span id="cb443-3"><a href="means-separation-and-data-presentation.html#cb443-3" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate SED</span></span>
<span id="cb443-4"><a href="means-separation-and-data-presentation.html#cb443-4" aria-hidden="true" tabindex="-1"></a><span class="co"># use add_rows to add SED and r to lsd groups table</span></span></code></pre></div>
</div>
<div id="create-the-gt-table" class="section level4" number="8.10.3.3">
<h4><span class="header-section-number">8.10.3.3</span> Create the gt() table</h4>
<div class="sourceCode" id="cb444"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb444-1"><a href="means-separation-and-data-presentation.html#cb444-1" aria-hidden="true" tabindex="-1"></a><span class="co"># recycle table from above: </span></span>
<span id="cb444-2"><a href="means-separation-and-data-presentation.html#cb444-2" aria-hidden="true" tabindex="-1"></a><span class="co"># change data.frame you feed to gt (line 1)</span></span>
<span id="cb444-3"><a href="means-separation-and-data-presentation.html#cb444-3" aria-hidden="true" tabindex="-1"></a><span class="co"># adjust decimals as appropriate (line 2)</span></span>
<span id="cb444-4"><a href="means-separation-and-data-presentation.html#cb444-4" aria-hidden="true" tabindex="-1"></a><span class="co"># change variable references for treatment and, if appropriate, yield (lines 3-5)</span></span>
<span id="cb444-5"><a href="means-separation-and-data-presentation.html#cb444-5" aria-hidden="true" tabindex="-1"></a><span class="co"># change column labels (lines 5-7)</span></span>
<span id="cb444-6"><a href="means-separation-and-data-presentation.html#cb444-6" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span></code></pre></div>

</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="multiple-treatment-designs.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="messy-and-missing-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-start.pdf", "bookdown-start.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
